From d590b5b6953304edb000de90335a3b4838b2482c Mon Sep 17 00:00:00 2001
From: San Mehat <san@google.com>
Date: Wed, 16 Sep 2009 12:39:10 -0700
Subject: [PATCH 0073/5965] proc: smaps: Allow smaps access for
 CAP_SYS_RESOURCE

Signed-off-by: San Mehat <san@google.com>
---
 kernel/fork.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/kernel/fork.c b/kernel/fork.c
index 987b28a1f01b..5733ad5d5856 100644
--- a/kernel/fork.c
+++ b/kernel/fork.c
@@ -696,7 +696,8 @@ struct mm_struct *mm_access(struct task_struct *task, unsigned int mode)
 
 	mm = get_task_mm(task);
 	if (mm && mm != current->mm &&
-			!ptrace_may_access(task, mode)) {
+			!ptrace_may_access(task, mode) &&
+			!capable(CAP_SYS_RESOURCE)) {
 		mmput(mm);
 		mm = ERR_PTR(-EACCES);
 	}
-- 
2.19.0

