From f3ea075005e43f0a5bb7d963843b7473ae1c0678 Mon Sep 17 00:00:00 2001
From: Dmitry Shmidt <dimitrysh@google.com>
Date: Mon, 30 Aug 2010 10:29:37 -0700
Subject: [PATCH 0079/5965] net: Fix CONFIG_RPS option to be turned off

Signed-off-by: Dmitry Shmidt <dimitrysh@google.com>
---
 net/Kconfig | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/net/Kconfig b/net/Kconfig
index 6a7c30a7d7f0..c5e34cf623c4 100644
--- a/net/Kconfig
+++ b/net/Kconfig
@@ -234,7 +234,7 @@ source "net/vmw_vsock/Kconfig"
 source "net/netlink/Kconfig"
 
 config RPS
-	boolean
+	boolean "RPS"
 	depends on SMP && SYSFS && USE_GENERIC_SMP_HELPERS
 	default y
 
-- 
2.19.0

