From 519be4566e2e60293d55bcfec71490af8e61b9e7 Mon Sep 17 00:00:00 2001
From: Greg Kroah-Hartman <gregkh@linuxfoundation.org>
Date: Wed, 14 Aug 2013 22:59:42 -0700
Subject: [PATCH 0926/5965] Linux 3.10.7

---
 Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index fd92ffba51c3..33e36aba5fa2 100644
--- a/Makefile
+++ b/Makefile
@@ -1,6 +1,6 @@
 VERSION = 3
 PATCHLEVEL = 10
-SUBLEVEL = 6
+SUBLEVEL = 7
 EXTRAVERSION =
 NAME = TOSSUG Baby Fish
 
-- 
2.19.0

