From 85cdabba08d484bdcc4b25f0bbc23ac60c75aa5b Mon Sep 17 00:00:00 2001
From: Greg Kroah-Hartman <gregkh@linuxfoundation.org>
Date: Sat, 7 Sep 2013 22:10:14 -0700
Subject: [PATCH 1086/5965] Linux 3.10.11

---
 Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index b119684ddb2d..595076dc14a0 100644
--- a/Makefile
+++ b/Makefile
@@ -1,6 +1,6 @@
 VERSION = 3
 PATCHLEVEL = 10
-SUBLEVEL = 10
+SUBLEVEL = 11
 EXTRAVERSION =
 NAME = TOSSUG Baby Fish
 
-- 
2.19.0

