From cff43fc8785eb4f8b3fa5a1030890fd8cad0cbdd Mon Sep 17 00:00:00 2001
From: Greg Kroah-Hartman <gregkh@linuxfoundation.org>
Date: Thu, 26 Sep 2013 17:18:49 -0700
Subject: [PATCH 1244/5965] Linux 3.10.13

---
 Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index afe001e3b2d6..25d38b79b3e0 100644
--- a/Makefile
+++ b/Makefile
@@ -1,6 +1,6 @@
 VERSION = 3
 PATCHLEVEL = 10
-SUBLEVEL = 12
+SUBLEVEL = 13
 EXTRAVERSION =
 NAME = TOSSUG Baby Fish
 
-- 
2.19.0

