From 8c15abc94c737f9120d3d4a550abbcbb9be121f6 Mon Sep 17 00:00:00 2001
From: Greg Kroah-Hartman <gregkh@linuxfoundation.org>
Date: Tue, 1 Oct 2013 09:18:05 -0700
Subject: [PATCH 1294/5965] Linux 3.10.14

---
 Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index 25d38b79b3e0..129c49f2dcd1 100644
--- a/Makefile
+++ b/Makefile
@@ -1,6 +1,6 @@
 VERSION = 3
 PATCHLEVEL = 10
-SUBLEVEL = 13
+SUBLEVEL = 14
 EXTRAVERSION =
 NAME = TOSSUG Baby Fish
 
-- 
2.19.0

