From 7fc878164dd8bc7812eb774e728f2cc1ffe905ed Mon Sep 17 00:00:00 2001
From: Greg Kroah-Hartman <gregkh@linuxfoundation.org>
Date: Sun, 13 Oct 2013 16:08:56 -0700
Subject: [PATCH 1457/5965] Linux 3.10.16

---
 Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index 9a77179b0b4f..e9528d242111 100644
--- a/Makefile
+++ b/Makefile
@@ -1,6 +1,6 @@
 VERSION = 3
 PATCHLEVEL = 10
-SUBLEVEL = 15
+SUBLEVEL = 16
 EXTRAVERSION =
 NAME = TOSSUG Baby Fish
 
-- 
2.19.0

