From 55a1068d3b342f89ccb4933995d3584b507bd7ba Mon Sep 17 00:00:00 2001
From: "yun.cai" <yun.cai@amlogic.com>
Date: Tue, 22 Oct 2013 10:15:09 +0800
Subject: [PATCH 1536/5965] add ion kconfig

---
 drivers/amlogic/Kconfig | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/amlogic/Kconfig b/drivers/amlogic/Kconfig
index 209c68417500..1ef2e136bae5 100755
--- a/drivers/amlogic/Kconfig
+++ b/drivers/amlogic/Kconfig
@@ -93,7 +93,7 @@ source "drivers/amlogic/hdmi/Kconfig"
 #comment "GPU (ARM Mali (external))"
 source "../hardware/arm/gpu/ump/Kconfig"
 source "../hardware/arm/gpu/mali/Kconfig"
-#source "drivers/amlogic/ion_dev/Kconfig"
+source "drivers/amlogic/ion_dev/Kconfig"
 
 #
 # crypto
-- 
2.19.0

