From b56dfd82ce98b57351d602a580cca2064a452e31 Mon Sep 17 00:00:00 2001
From: "brian.zhu" <brian.zhu@amlogic.com>
Date: Mon, 21 Oct 2013 13:06:48 +0800
Subject: [PATCH 1574/5965] rm debug print in amlvideo2

---
 drivers/amlogic/video_dev/amlvideo2.c | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/drivers/amlogic/video_dev/amlvideo2.c b/drivers/amlogic/video_dev/amlvideo2.c
index 368234e17f43..cffd59a888bb 100755
--- a/drivers/amlogic/video_dev/amlvideo2.c
+++ b/drivers/amlogic/video_dev/amlvideo2.c
@@ -1348,9 +1348,8 @@ static int vidioc_querybuf(struct file *file, void *priv, struct v4l2_buffer *p)
 	    output.height = fh->height;
 	    output.canvas_id = -1;
 	    p->reserved  = convert_canvas_index(&output, VM_RES0_CANVAS_INDEX+p->index*3);
-	    printk("Brian test: vidioc_querybuf: canvas:0x%x, id: %d\n",p->reserved,p->index);
 	}else{
-           p->reserved = 0;
+	    p->reserved = 0;
 	}
 #endif
 	return ret;
-- 
2.19.0

