From b24008e54217ed72addfedf92709d0fe5b6a0d81 Mon Sep 17 00:00:00 2001
From: "chunhai.guo" <chunhai.guo@amlogic.com>
Date: Tue, 15 Oct 2013 19:43:36 +0800
Subject: [PATCH 1602/5965] remove some debug print messages

---
 drivers/mmc/card/emmc_partitions.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/mmc/card/emmc_partitions.c b/drivers/mmc/card/emmc_partitions.c
index 142bc420a691..e99d1b374095 100644
--- a/drivers/mmc/card/emmc_partitions.c
+++ b/drivers/mmc/card/emmc_partitions.c
@@ -469,10 +469,10 @@ int aml_emmc_partition_ops (struct mmc_card *card, struct gendisk *disk)
     struct disk_part_iter piter;
     struct hd_struct *part;
 
-    printk("Enter %s\n", __FUNCTION__);
+    // printk("Enter %s\n", __FUNCTION__);
 
     if (!is_card_emmc(card)) { // not emmc, nothing to do
-        printk("[%s] %s is not eMMC!\n", __FUNCTION__, disk->disk_name);
+        // printk("[%s] %s is not eMMC!\n", __FUNCTION__, disk->disk_name);
         return 0;
     }
     
-- 
2.19.0

