From 6347384aaf96d87ad2e42489a6a0edcddc42a05b Mon Sep 17 00:00:00 2001
From: "chunhai.guo" <chunhai.guo@amlogic.com>
Date: Mon, 21 Oct 2013 17:23:55 +0800
Subject: [PATCH 1603/5965] PD81458: solve sdio timeout problem while kernel
 resumes

---
 drivers/amlogic/mmc/aml_sdio.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/drivers/amlogic/mmc/aml_sdio.c b/drivers/amlogic/mmc/aml_sdio.c
index 4e2d46610465..0c1b9b346520 100755
--- a/drivers/amlogic/mmc/aml_sdio.c
+++ b/drivers/amlogic/mmc/aml_sdio.c
@@ -31,6 +31,7 @@
 #include "amlsd.h"
 
 #include <linux/amlogic/aml_gpio_consumer.h>
+#include <linux/amlogic/wifi_dt.h>
 
 static struct mmc_claim aml_sdio_claim;
 
@@ -830,6 +831,9 @@ static int aml_sdio_resume(struct platform_device *pdev)
 
 	printk("***Entered %s:%s\n", __FILE__,__func__);
 	list_for_each_entry(pdata, &host->sibling, sibling){
+        if(pdata->port == MESON_SDIO_PORT_A) { // sdio_wifi
+            wifi_setup_dt();
+        }
 		mmc = pdata->mmc;
 		//mmc_power_restore_host(mmc);
 		mmc_resume_host(mmc);
-- 
2.19.0

