From 3de12c6d1a7aa3f0ae750b91004fd395f8570b3c Mon Sep 17 00:00:00 2001
From: "qi.duan" <qi.duan@amlogic.com>
Date: Fri, 1 Nov 2013 15:58:22 +0800
Subject: [PATCH 1609/5965] remove custom directory

---
 customer | 1 -
 1 file changed, 1 deletion(-)
 delete mode 160000 customer

diff --git a/customer b/customer
deleted file mode 160000
index c6ccfb45b672..000000000000
--- a/customer
+++ /dev/null
@@ -1 +0,0 @@
-Subproject commit c6ccfb45b6728ffcdebfd3ed4ce1dfde47b24dc0
-- 
2.19.0

