From 1c9756c7c985a056e5144b64a91d55c766b7c76f Mon Sep 17 00:00:00 2001
From: Mingliang Yang <mingliang.yang@amlogic.com>
Date: Wed, 30 Oct 2013 16:40:49 +0800
Subject: [PATCH 1616/5965] #PD 81459 fixed NULL P will crash

---
 drivers/amlogic/amports/ptsserv.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/drivers/amlogic/amports/ptsserv.c b/drivers/amlogic/amports/ptsserv.c
index 38267b016d63..31dfc331252a 100755
--- a/drivers/amlogic/amports/ptsserv.c
+++ b/drivers/amlogic/amports/ptsserv.c
@@ -624,7 +624,8 @@ static int pts_lookup_offset_inline(
     return -1;
 }
 int pts_lookup_offset(u8 type, u32 offset, u32 *val, u32 pts_margin){
-	return pts_lookup_offset_inline(type,offset,val,pts_margin,0);
+	u64 pts_us;
+	return pts_lookup_offset_inline(type,offset,val,pts_margin,&pts_us);
 }
 int pts_lookup_offset_us64(u8 type, u32 offset, u32 *val ,u32 pts_margin, u64 *uS64){
 	return pts_lookup_offset_inline(type,offset,val,pts_margin,uS64);
-- 
2.19.0

