From ef603694dd742df4abcba611cd161f353a6aa2b0 Mon Sep 17 00:00:00 2001
From: "jian.xu" <jian.xu@amlogic.com>
Date: Fri, 1 Nov 2013 17:14:29 +0800
Subject: [PATCH 1622/5965] audio dai code merged

---
 sound/soc/aml/aml_dai.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/sound/soc/aml/aml_dai.c b/sound/soc/aml/aml_dai.c
index 4da31a1c6248..02b88b0b9896 100755
--- a/sound/soc/aml/aml_dai.c
+++ b/sound/soc/aml/aml_dai.c
@@ -15,7 +15,7 @@
 
 #include "aml_dai.h"
 #include "aml_pcm.h"
-#include <aml_audio_hw.h>
+#include "aml_audio_hw.h"
 #include <linux/of.h>
 
 static aml_dai_info_t dai_info[3] = {{0}};
-- 
2.19.0

