From 4e77f7f1261f65cff06918bc5e66d02a418fc842 Mon Sep 17 00:00:00 2001
From: Greg Kroah-Hartman <gregkh@linuxfoundation.org>
Date: Mon, 4 Nov 2013 04:31:29 -0800
Subject: [PATCH 1698/5965] Linux 3.10.18

---
 Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index 5c7d3d63d376..5fb14e503fe3 100644
--- a/Makefile
+++ b/Makefile
@@ -1,6 +1,6 @@
 VERSION = 3
 PATCHLEVEL = 10
-SUBLEVEL = 17
+SUBLEVEL = 18
 EXTRAVERSION =
 NAME = TOSSUG Baby Fish
 
-- 
2.19.0

