From d505b469ea6c7415cf0dbbb0e57240e83fc45d52 Mon Sep 17 00:00:00 2001
From: "shi.liu" <shi.liu@amlogic.com>
Date: Tue, 5 Nov 2013 14:16:54 +0800
Subject: [PATCH 1707/5965] cpufreq:work run on policy->cpu

---
 drivers/cpufreq/cpufreq_hotplug.c | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/drivers/cpufreq/cpufreq_hotplug.c b/drivers/cpufreq/cpufreq_hotplug.c
index 6ff9d3b04d80..98a9b6d1018c 100755
--- a/drivers/cpufreq/cpufreq_hotplug.c
+++ b/drivers/cpufreq/cpufreq_hotplug.c
@@ -491,7 +491,6 @@ static void cpu_hotplug_thread(int *hotplug_flag)
 	while(1){
 		if (kthread_should_stop())
 			break;
-
 		mutex_lock(&dbs_info->cdbs.timer_mutex);
 		if(*hotplug_flag == CPU_HOTPLUG_PLUG){
 			*hotplug_flag = CPU_HOTPLUG_NONE;
@@ -677,9 +676,9 @@ static void hg_dbs_timer(struct work_struct *work)
 {
 	struct hg_cpu_dbs_info_s *dbs_info =
 		container_of(work, struct hg_cpu_dbs_info_s, cdbs.work.work);
-	unsigned int cpu = dbs_info->cdbs.cpu;
 	unsigned long flags;
 	struct cpufreq_policy *policy = dbs_info->cdbs.cur_policy;
+	unsigned int cpu = policy->cpu;
 	struct dbs_data *dbs_data = policy->governor_data;
 	struct hg_dbs_tuners *hg_tuners = dbs_data->tuners;
 
-- 
2.19.0

