From ef8242484a50a6aeb425efbf723fb822f7335f5a Mon Sep 17 00:00:00 2001
From: "shuai.cao" <shuai.cao@amlogic.com>
Date: Tue, 5 Nov 2013 14:48:16 +0800
Subject: [PATCH 1708/5965] fix video prot compile problem

---
 drivers/amlogic/amports/video_prot.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/amlogic/amports/video_prot.c b/drivers/amlogic/amports/video_prot.c
index 3f49ab403048..bb26c787fbce 100755
--- a/drivers/amlogic/amports/video_prot.c
+++ b/drivers/amlogic/amports/video_prot.c
@@ -7,7 +7,7 @@
  * Author:  Amlogic R&D Group
  *
  */
-#include "video_prot.h"
+#include <linux/amlogic/amports/video_prot.h>
 
 static int set_prot_NV21(u32 x_start, u32 x_end, u32 y_start, u32 y_end, u32 y_step, u32 angle, u32 pat_val) {
 
-- 
2.19.0

