From 2784a2c5cb84f101c4cf8449625d62574b974c2e Mon Sep 17 00:00:00 2001
From: "zhenfei.li" <zhenfei.li@amlogic>
Date: Tue, 29 Oct 2013 13:26:16 +0800
Subject: [PATCH 1730/5965] PD #81021: for compile error.

---
 drivers/amlogic/display/vout2/vout2_serve.c | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/drivers/amlogic/display/vout2/vout2_serve.c b/drivers/amlogic/display/vout2/vout2_serve.c
index 7de95c503057..4334fd98441c 100755
--- a/drivers/amlogic/display/vout2/vout2_serve.c
+++ b/drivers/amlogic/display/vout2/vout2_serve.c
@@ -277,6 +277,14 @@ static void meson_vout_late_resume(struct early_suspend *h)
 **	vout driver interface  
 **
 ******************************************************************/
+#if MESON_CPU_TYPE >= MESON_CPU_TYPE_MESON8
+#define VPP_OFIFO_SIZE_WID          13
+#else
+#define VPP_OFIFO_SIZE_WID          12
+#endif // MESON_CPU_TYPE >= MESON_CPU_TYPE_MESON8
+#define VPP_OFIFO_SIZE_MASK         0xfff
+#define VPP_OFIFO_SIZE_BIT          0
+
 static int 
  meson_vout_probe(struct platform_device *pdev)
 {
-- 
2.19.0

