From 4306056939e212253bd1638d8326b6d67c6a2bd7 Mon Sep 17 00:00:00 2001
From: xu xing <xing.xu@amlogic.com>
Date: Wed, 6 Nov 2013 19:05:15 +0800
Subject: [PATCH 1744/5965] Pinctrl:Free memory when pinctrl register errror.

---
 drivers/amlogic/pinctrl/pinctrl-amlogic.c | 8 ++------
 1 file changed, 2 insertions(+), 6 deletions(-)

diff --git a/drivers/amlogic/pinctrl/pinctrl-amlogic.c b/drivers/amlogic/pinctrl/pinctrl-amlogic.c
index 7342502949e8..919978af203b 100755
--- a/drivers/amlogic/pinctrl/pinctrl-amlogic.c
+++ b/drivers/amlogic/pinctrl/pinctrl-amlogic.c
@@ -732,14 +732,10 @@ int  amlogic_pmx_probe(struct platform_device *pdev,struct amlogic_pinctrl_soc_d
 	amlogic_pmx_desc.npins = apmx->soc->npins;
 
 	apmx->pctl = pinctrl_register(&amlogic_pmx_desc, &pdev->dev, apmx);
-	if (ret) {
-		dev_err(&pdev->dev, "dt probe failed: %d\n", ret);
-		goto err;
-	}
-	
+
 	if (!apmx->pctl) {
 		dev_err(&pdev->dev, "Couldn't register pinctrl driver\n");
-		return -ENODEV;
+		goto err;
 	}
 	pinctrl_add_gpio_range(apmx->pctl, &amlogic_gpio_ranges);
 	pctdev_name=dev_name(&pdev->dev);
-- 
2.19.0

