From f7f4783f01bf60f7ea04d60abf959f64a41b4518 Mon Sep 17 00:00:00 2001
From: "stark.li" <stark.li@amlogic.com>
Date: Thu, 7 Nov 2013 14:31:22 +0800
Subject: [PATCH 1756/5965] Fix cts testNoExecutePermission

---
 arch/arm/kernel/process.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/arch/arm/kernel/process.c b/arch/arm/kernel/process.c
index 34ee7807d019..9c0a9c919e97 100755
--- a/arch/arm/kernel/process.c
+++ b/arch/arm/kernel/process.c
@@ -661,7 +661,7 @@ int in_gate_area_no_mm(unsigned long addr)
 {
 	return in_gate_area(NULL, addr);
 }
-#define is_gate_vma(vma)	((vma) = &gate_vma)
+#define is_gate_vma(vma)	((vma) == &gate_vma)
 #else
 #define is_gate_vma(vma)	0
 #endif
-- 
2.19.0

