From 0d171a5e8d34b9d6c431cd21a2f1906f7fc38400 Mon Sep 17 00:00:00 2001
From: "stark.li" <stark.li@amlogic.com>
Date: Thu, 7 Nov 2013 14:33:13 +0800
Subject: [PATCH 1757/5965] fix CTS testTryToCrashVold

---
 net/netlink/af_netlink.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/net/netlink/af_netlink.c b/net/netlink/af_netlink.c
index 57ee84d21470..daf7c576b24b 100644
--- a/net/netlink/af_netlink.c
+++ b/net/netlink/af_netlink.c
@@ -2762,7 +2762,7 @@ static int netlink_seq_show(struct seq_file *seq, void *v)
 		struct sock *s = v;
 		struct netlink_sock *nlk = nlk_sk(s);
 
-		seq_printf(seq, "%pK %-3d %-6u %08x %-8d %-8d %pK %-8d %-8d %-8lu\n",
+		seq_printf(seq, "%pK %-3d %-6d %08x %-8d %-8d %pK %-8d %-8d %-8lu\n",
 			   s,
 			   s->sk_protocol,
 			   nlk->portid,
-- 
2.19.0

