From 90d9bebbef0ed058fb717355a80d2b8de8dd4cf6 Mon Sep 17 00:00:00 2001
From: Sandy lUo <sandy.luo@amlogic.com>
Date: Fri, 8 Nov 2013 15:13:05 +0800
Subject: [PATCH 1777/5965] fix interactive can't be used

---
 drivers/cpufreq/Kconfig | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/cpufreq/Kconfig b/drivers/cpufreq/Kconfig
index f50d04be7966..259668186fa1 100755
--- a/drivers/cpufreq/Kconfig
+++ b/drivers/cpufreq/Kconfig
@@ -181,6 +181,7 @@ config CPU_FREQ_GOV_HOTPLUG
 	depends on SMP
 	default y
 	help
+
 config CPU_FREQ_GOV_INTERACTIVE
 	tristate "'interactive' cpufreq policy governor"
 	help
-- 
2.19.0

