From 2f6f26542b2a0dffc32595fe6a12f5e4e508898b Mon Sep 17 00:00:00 2001
From: "frank.zhao" <frank.zhao@amlogic.com>
Date: Fri, 8 Nov 2013 15:51:02 +0800
Subject: [PATCH 1780/5965] disable RAW data will store to DDR in bt656.

---
 drivers/amlogic/tvin/bt656/bt656_601_in.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/amlogic/tvin/bt656/bt656_601_in.c b/drivers/amlogic/tvin/bt656/bt656_601_in.c
index d20e32eb5605..f62a76a03361 100755
--- a/drivers/amlogic/tvin/bt656/bt656_601_in.c
+++ b/drivers/amlogic/tvin/bt656/bt656_601_in.c
@@ -394,6 +394,7 @@ static void reinit_camera_dec(struct am656in_dev_s *devp)
 			|(1 << BT_CAMERA_MODE)     // enable camera mode
 			|(1 << BT_656CLOCK_RESET) 
 			|(1 << BT_SYSCLOCK_RESET) 
+			|(1<<27)				//enable raw data to isp
 			;
 	}
 	if(devp->para.bt_path == BT_PATH_GPIO) {
-- 
2.19.0

