From e60151efc7919295d6068e7c21fd2e83b4d865fe Mon Sep 17 00:00:00 2001
From: jiwei chen <jiwei.chen@amlogic.com>
Date: Fri, 8 Nov 2013 16:48:42 +0800
Subject: [PATCH 1782/5965] isp,change the order for structure.

---
 include/linux/amlogic/tvin/tvin_v4l2.h | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/include/linux/amlogic/tvin/tvin_v4l2.h b/include/linux/amlogic/tvin/tvin_v4l2.h
index 0fc15b5062a7..b31dcc92ffc7 100755
--- a/include/linux/amlogic/tvin/tvin_v4l2.h
+++ b/include/linux/amlogic/tvin/tvin_v4l2.h
@@ -335,10 +335,10 @@ typedef struct xml_algorithm_ae_s {
         unsigned int  targetmid;  // 0 ~ 255
         unsigned int  targetlow;  // 0 ~ 255
         unsigned int  radium_inner_h;  // 0 ~ 255
-        unsigned int  radium_outer_h;  // 0 ~ 255
         unsigned int  radium_inner_m;  // 0 ~ 255
-        unsigned int  radium_outer_m;  // 0 ~ 255
         unsigned int  radium_inner_l;  // 0 ~ 255
+        unsigned int  radium_outer_h;  // 0 ~ 255
+        unsigned int  radium_outer_m;  // 0 ~ 255
         unsigned int  radium_outer_l;  // 0 ~ 255
         unsigned int  flash_thr;       // 0 ~ 255 
 
-- 
2.19.0

