From 66114410d076fa93187d3d4022cac98b069295b9 Mon Sep 17 00:00:00 2001
From: "kele.bai" <kele.bai@amlogic.com>
Date: Sun, 10 Nov 2013 18:22:09 +0800
Subject: [PATCH 1804/5965] fix compile error.

---
 drivers/amlogic/tvin/isp/isp_sm.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/amlogic/tvin/isp/isp_sm.c b/drivers/amlogic/tvin/isp/isp_sm.c
index 85168faa21d0..3e56111d514f 100755
--- a/drivers/amlogic/tvin/isp/isp_sm.c
+++ b/drivers/amlogic/tvin/isp/isp_sm.c
@@ -972,6 +972,7 @@ static bool is_lost_focus(isp_af_info_t *af_info,xml_algorithm_af_t *af_alg)
 			} else{
                                 if(af_sm_dg&0x1)
 				        pr_info("static ave_vdc=%llu.\n",ave_vdc);
+			}
 			break;
 		}else if(tmp_vdc2 < ave_vdc){
 			if(++static_cnt >= af_alg->detect_step_cnt)
-- 
2.19.0

