From 5ef41177808c867a45441aeb161b93cbe873e68f Mon Sep 17 00:00:00 2001
From: Evoke Zhang <evoke.zhang@amlogic.com>
Date: Mon, 11 Nov 2013 16:04:45 +0800
Subject: [PATCH 1815/5965] update vpu driver for uboot compatible

---
 arch/arm/mach-meson8/vpu.c | 2 --
 1 file changed, 2 deletions(-)

diff --git a/arch/arm/mach-meson8/vpu.c b/arch/arm/mach-meson8/vpu.c
index 946a126e113e..7fc2921c7878 100755
--- a/arch/arm/mach-meson8/vpu.c
+++ b/arch/arm/mach-meson8/vpu.c
@@ -677,9 +677,7 @@ static int vpu_probe(struct platform_device *pdev)
 	
 	printk("VPU driver version: %s\n", VPU_VERION);
 	get_vpu_config(pdev);
-#ifndef CONFIG_VPU_DYNAMIC_ADJ
 	set_vpu_clk(vpu_config.clk_level);
-#endif
 	
 	ret = class_register(&aml_vpu_debug_class);
 	if(ret){
-- 
2.19.0

