From 8ec267ac48a00f644e52e76cdea7645e374af002 Mon Sep 17 00:00:00 2001
From: "zhenfei.li" <zhenfei.li@amlogic>
Date: Mon, 11 Nov 2013 22:05:26 +0800
Subject: [PATCH 1827/5965] adjust ov5640 display effect.

---
 drivers/amlogic/camera/ov5640.c | 23 +++++++++++------------
 1 file changed, 11 insertions(+), 12 deletions(-)

diff --git a/drivers/amlogic/camera/ov5640.c b/drivers/amlogic/camera/ov5640.c
index b36ae9471539..338492582169 100755
--- a/drivers/amlogic/camera/ov5640.c
+++ b/drivers/amlogic/camera/ov5640.c
@@ -73,7 +73,7 @@ static unsigned debug;
 //module_param(debug, uint, 0644);
 //MODULE_PARM_DESC(debug, "activates debug info");
 
-static unsigned int vid_limit = 16;
+static unsigned int vid_limit = 32;
 //module_param(vid_limit, uint, 0644);
 //MODULE_PARM_DESC(vid_limit, "capture memory limit in megabytes");
 
@@ -745,7 +745,7 @@ static struct aml_camera_i2c_fig_s OV5640_script[] = {
 	{0x530b, 0x04},  
 	{0x530c, 0x06},     
 	{0x5580, 0x06},  
-	{0x5583, 0x10}, 
+	{0x5583, 0x15}, 
 	{0x5584, 0x10}, 
 	{0x5589, 0x10}, 
 	{0x558a, 0x00}, 
@@ -1136,7 +1136,7 @@ static struct aml_camera_i2c_fig_s OV5640_preview_1080P_script[] = {
 	{0x548f, 0xea},
 	{0x5490, 0x1d},
 	{0x5580, 0x02},
-	{0x5583, 0x10},
+	{0x5583, 0x15},
 	{0x5584, 0x10},
 	{0x5589, 0x10},
 	{0x558a, 0x00},
@@ -1406,7 +1406,7 @@ static struct aml_camera_i2c_fig_s OV5640_preview_960P_script[] = {
 	{0x530b, 0x04},  
 	{0x530c, 0x06},     
 	{0x5580, 0x06},  
-	{0x5583, 0x10}, 
+	{0x5583, 0x15}, 
 	{0x5584, 0x10}, 
 	{0x5589, 0x10}, 
 	{0x558a, 0x00}, 
@@ -1619,7 +1619,7 @@ static struct aml_camera_i2c_fig_s OV5640_preview_960P_script[] = {
 	{0x3a0e, 0x06},
 	{0x3a0d, 0x08}, 
 #endif
-	
+
 	{0xffff, 0xff}
 };
 
@@ -1855,7 +1855,7 @@ static struct aml_camera_i2c_fig_s OV5640_preview_QVGA_script[] = {
 	
 	{0x3820, 0x41}, // #3 ck. origin: 0x47,
 	{0x3821, 0x07}, // #3 ck. origin: 0x01, 
-	{0x3031, 0x08},   
+	{0x3031, 0x08},  
 	{0xffff, 0xff}
 };
 
@@ -2019,10 +2019,10 @@ static resolution_param_t  prev_resolution_array[] = {
 	},
 	#endif
 	{
-		.frmsize			= {1024, 768},
+		.frmsize			= {1280, 960},
 		.active_frmsize			= {1280, 958},
 		.active_fps			= 30,
-		.size_type			= SIZE_1024X768,
+		.size_type			= SIZE_1280X960,
 		.reg_script			= OV5640_preview_960P_script,
 	},{
 		.frmsize			= {1280, 720},
@@ -2031,14 +2031,13 @@ static resolution_param_t  prev_resolution_array[] = {
 		.size_type			= SIZE_1280X720,
 		.reg_script			= OV5640_preview_720P_script,
 	},
-	/*{
+	{
 		.frmsize			= {1024, 768},
 		.active_frmsize			= {1280, 958},
 		.active_fps			= 30,
 		.size_type			= SIZE_1024X768,
-		//.reg_script			= OV5640_preview_960P_script,
-		.reg_script			= OV5640_preview_720P_script,
-	},*/{
+		.reg_script			= OV5640_preview_960P_script,
+	},{
 		.frmsize			= {640, 480},
 		.active_frmsize			= {640, 478},
 		.active_fps			= 30,
-- 
2.19.0

