From e6276322481cc91b11eca3b634fec495ae537994 Mon Sep 17 00:00:00 2001
From: Evoke Zhang <evoke.zhang@amlogic.com>
Date: Tue, 12 Nov 2013 16:13:15 +0800
Subject: [PATCH 1836/5965] optimize M8 lvds phy setting in lcd driver

---
 drivers/amlogic/display/vout/lcdoutc.c | 14 +++++++-------
 1 file changed, 7 insertions(+), 7 deletions(-)

diff --git a/drivers/amlogic/display/vout/lcdoutc.c b/drivers/amlogic/display/vout/lcdoutc.c
index d331adbe5bc2..17f74b47ee06 100755
--- a/drivers/amlogic/display/vout/lcdoutc.c
+++ b/drivers/amlogic/display/vout/lcdoutc.c
@@ -1982,26 +1982,26 @@ static void init_phy_lvds(Lcd_Config_t *pConf)
 	//WRITE_LCD_CBUS_REG(HHI_DIF_CSI_PHY_CNTL1, 0x00000348);
 	switch (pConf->lcd_control.lvds_config->lvds_vswing) {
 		case 0:
-			swing_ctrl = 0x328;
+			swing_ctrl = 0x028;
 			break;
 		case 1:
-			swing_ctrl = 0x348;
+			swing_ctrl = 0x048;
 			break;
 		case 2:
-			swing_ctrl = 0x388;
+			swing_ctrl = 0x088;
 			break;
 		case 3:
-			swing_ctrl = 0x3c8;
+			swing_ctrl = 0x0c8;
 			break;
 		case 4:
-			swing_ctrl = 0x3f8;
+			swing_ctrl = 0x0f8;
 			break;
 		default:
-			swing_ctrl = 0x348;
+			swing_ctrl = 0x048;
 			break;
 	}
 	WRITE_LCD_CBUS_REG(HHI_DIF_CSI_PHY_CNTL1, swing_ctrl);
-	WRITE_LCD_CBUS_REG(HHI_DIF_CSI_PHY_CNTL2, 0x004665b7);
+	WRITE_LCD_CBUS_REG(HHI_DIF_CSI_PHY_CNTL2, 0x000665b7);
 	WRITE_LCD_CBUS_REG(HHI_DIF_CSI_PHY_CNTL3, 0x84070000);
 #endif
 }
-- 
2.19.0

