From 3729ed7c6aa8c5b9eee8f832e4a246b8fa1d56b5 Mon Sep 17 00:00:00 2001
From: Greg Kroah-Hartman <gregkh@linuxfoundation.org>
Date: Wed, 13 Nov 2013 12:05:59 +0900
Subject: [PATCH 1925/5965] Linux 3.10.19

---
 Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index 5fb14e503fe3..83a02f5b2c00 100644
--- a/Makefile
+++ b/Makefile
@@ -1,6 +1,6 @@
 VERSION = 3
 PATCHLEVEL = 10
-SUBLEVEL = 18
+SUBLEVEL = 19
 EXTRAVERSION =
 NAME = TOSSUG Baby Fish
 
-- 
2.19.0

