From be358b4e50d614a137282a1f8a27a7fd715cee97 Mon Sep 17 00:00:00 2001
From: "chunhai.guo" <chunhai.guo@amlogic.com>
Date: Wed, 13 Nov 2013 17:45:04 +0800
Subject: [PATCH 1934/5965] temporary fix for emmc resume timeout issue

---
 drivers/mmc/core/mmc_ops.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/mmc/core/mmc_ops.c b/drivers/mmc/core/mmc_ops.c
index 49f04bc9d0eb..bd2cc701dbb6 100755
--- a/drivers/mmc/core/mmc_ops.c
+++ b/drivers/mmc/core/mmc_ops.c
@@ -142,7 +142,7 @@ int mmc_send_op_cond(struct mmc_host *host, u32 ocr, u32 *rocr)
 	cmd.flags = MMC_RSP_SPI_R1 | MMC_RSP_R3 | MMC_CMD_BCR;
 
 	for (i = 100; i; i--) {
-		err = mmc_wait_for_cmd(host, &cmd, 0);
+		err = mmc_wait_for_cmd(host, &cmd, 3); // fix: temporary add for emmc resume timeout issue
 		if (err)
 			break;
 
-- 
2.19.0

