From 0cbec9ebf43ac2ea28c7af012f7f1b4c2db28a1a Mon Sep 17 00:00:00 2001
From: "frank.zhao" <frank.zhao@amlogic.com>
Date: Thu, 14 Nov 2013 10:11:52 +0800
Subject: [PATCH 1940/5965] update lens-shading debug function.

---
 drivers/amlogic/tvin/isp/isp_drv.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/drivers/amlogic/tvin/isp/isp_drv.c b/drivers/amlogic/tvin/isp/isp_drv.c
index 260847d92dbe..2c7b2ac3c0a2 100755
--- a/drivers/amlogic/tvin/isp/isp_drv.c
+++ b/drivers/amlogic/tvin/isp/isp_drv.c
@@ -1188,6 +1188,7 @@ static int isp_probe(struct platform_device *pdev)
 	ret = device_create_file(devp->dev,&dev_attr_wave_param);
 	ret = device_create_file(devp->dev,&dev_attr_gamma_debug);
 	ret = device_create_file(devp->dev,&dev_attr_gamma);
+	ret = device_create_file(devp->dev,&dev_attr_lens);
 	if(ret < 0)
 		goto err;
 	
@@ -1219,6 +1220,7 @@ static int isp_remove(struct platform_device *pdev)
 	device_remove_file(devp->dev,&dev_attr_af_param);
 	device_remove_file(devp->dev,&dev_attr_cap_param);
 	device_remove_file(devp->dev,&dev_attr_wave_param);
+	device_remove_file(devp->dev,&dev_attr_lens);
 	
 	isp_delete_device(devp->index);
         tvin_unreg_frontend(&devp->frontend);
-- 
2.19.0

