From 059399069cf763046040e85d490c0f8a057ef9d1 Mon Sep 17 00:00:00 2001
From: Evoke Zhang <evoke.zhang@amlogic.com>
Date: Fri, 15 Nov 2013 11:32:26 +0800
Subject: [PATCH 1948/5965] update include file for CONFIG_SMP control in vpu
 driver

---
 arch/arm/mach-meson8/vpu.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/arch/arm/mach-meson8/vpu.c b/arch/arm/mach-meson8/vpu.c
index 732bf2391daf..86c1d820ad8b 100755
--- a/arch/arm/mach-meson8/vpu.c
+++ b/arch/arm/mach-meson8/vpu.c
@@ -14,7 +14,9 @@
 #include <plat/io.h>
 #include <mach/am_regs.h>
 #include <mach/vpu.h>
+#ifdef CONFIG_SMP
 #include <mach/smp.h>
+#endif
 #include <linux/amlogic/vout/vinfo.h>
 
 #define VPU_VERION	"v03"
-- 
2.19.0

