From f7804c91b3f880953780de801925793960ad4f1f Mon Sep 17 00:00:00 2001
From: "zhenfei.li" <zhenfei.li@amlogic>
Date: Fri, 15 Nov 2013 19:46:16 +0800
Subject: [PATCH 1955/5965] increase ov5640 saturation.

---
 drivers/amlogic/camera/ov5640.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/drivers/amlogic/camera/ov5640.c b/drivers/amlogic/camera/ov5640.c
index 192a29f97b05..4c4411e827df 100755
--- a/drivers/amlogic/camera/ov5640.c
+++ b/drivers/amlogic/camera/ov5640.c
@@ -764,7 +764,7 @@ static struct aml_camera_i2c_fig_s OV5640_script[] = {
 	{0x530b, 0x04},  
 	{0x530c, 0x06},     
 	{0x5580, 0x06},  
-	{0x5583, 0x15}, 
+	{0x5583, 0x30}, 
 	{0x5584, 0x10}, 
 	{0x5589, 0x10}, 
 	{0x558a, 0x00}, 
@@ -1155,7 +1155,7 @@ static struct aml_camera_i2c_fig_s OV5640_preview_1080P_script[] = {
 	{0x548f, 0xea},
 	{0x5490, 0x1d},
 	{0x5580, 0x02},
-	{0x5583, 0x15},
+	{0x5583, 0x30},
 	{0x5584, 0x10},
 	{0x5589, 0x10},
 	{0x558a, 0x00},
@@ -1425,7 +1425,7 @@ static struct aml_camera_i2c_fig_s OV5640_preview_960P_script[] = {
 	{0x530b, 0x04},  
 	{0x530c, 0x06},     
 	{0x5580, 0x06},  
-	{0x5583, 0x15}, 
+	{0x5583, 0x30}, 
 	{0x5584, 0x10}, 
 	{0x5589, 0x10}, 
 	{0x558a, 0x00}, 
-- 
2.19.0

