From a840b660701dd6a312a3cb1219f9b593c1583c25 Mon Sep 17 00:00:00 2001
From: jiwei chen <jiwei.chen@amlogic.com>
Date: Fri, 15 Nov 2013 20:09:46 +0800
Subject: [PATCH 1956/5965] isp,fix a compile error.

---
 drivers/amlogic/tvin/isp/isp_sm.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/amlogic/tvin/isp/isp_sm.c b/drivers/amlogic/tvin/isp/isp_sm.c
index c1971f04f7e7..1e70a587c722 100755
--- a/drivers/amlogic/tvin/isp/isp_sm.c
+++ b/drivers/amlogic/tvin/isp/isp_sm.c
@@ -195,6 +195,7 @@ void isp_ae_base_sm(isp_dev_t *devp)
 	int i;
 	static int k = 0;
 	static int h = 0;
+	static int sum = 0;
 	int step = 0;
 	unsigned short targrate;
 	unsigned int targstep, newstep;
-- 
2.19.0

