From c25f0146978d7cbd4ebce4d1e0f9c6c8461e0568 Mon Sep 17 00:00:00 2001
From: "dezhi.kong" <dezhi.kong@amlogic.com>
Date: Tue, 19 Nov 2013 20:18:13 +0800
Subject: [PATCH 1984/5965] isp,set fullscan done flag when af success

---
 drivers/amlogic/tvin/isp/isp_sm.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/drivers/amlogic/tvin/isp/isp_sm.c b/drivers/amlogic/tvin/isp/isp_sm.c
index 623e99ab9eaa..ba5520a54b6f 100755
--- a/drivers/amlogic/tvin/isp/isp_sm.c
+++ b/drivers/amlogic/tvin/isp/isp_sm.c
@@ -1332,6 +1332,7 @@ void isp_af_sm(isp_dev_t *devp)
 						sm_state.af_state = AF_DETECT_INIT;
 					else
 						sm_state.af_state = AF_NULL;
+					devp->cmd_state = CAM_STATE_SUCCESS;
 					isp_set_blenr_stat(af_info->x0,af_info->y0,af_info->x1,af_info->y1);
 					isp_set_af_scan_stat(af_info->x0,af_info->y0,af_info->x1,af_info->y1);
 				} else {/*af success*/
@@ -1350,6 +1351,7 @@ void isp_af_sm(isp_dev_t *devp)
 						sm_state.af_state = AF_DETECT_INIT;
 					else
 						sm_state.af_state = AF_NULL;
+					devp->cmd_state = CAM_STATE_SUCCESS;
 					isp_set_blenr_stat(af_info->x0,af_info->y0,af_info->x1,af_info->y1);
 					isp_set_af_scan_stat(af_info->x0,af_info->y0,af_info->x1,af_info->y1);
 				}
-- 
2.19.0

