From d3d047894756ee9b1595e35d097231826883f264 Mon Sep 17 00:00:00 2001
From: "jian.xu" <jian.xu@amlogic.com>
Date: Tue, 19 Nov 2013 16:11:02 +0800
Subject: [PATCH 2039/5965] pd # 82486 :enable hrtimer for M8 driver

---
 sound/soc/aml/aml_pcm.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/sound/soc/aml/aml_pcm.c b/sound/soc/aml/aml_pcm.c
index b10d3f6a2536..336a7cae3f02 100755
--- a/sound/soc/aml/aml_pcm.c
+++ b/sound/soc/aml/aml_pcm.c
@@ -32,8 +32,8 @@
 #include "aml_audio_hw.h"
 #include "aml_platform.h"
 
-#define USE_HRTIMER 0
-#define HRTIMER_PERIOD (1000000000UL/48000)
+#define USE_HRTIMER 1
+#define HRTIMER_PERIOD (1000000000UL/1000)
 #define DEBUG_ALSA_PLATFRORM
 
 #define ALSA_PRINT(fmt,args...)	printk(KERN_INFO "[aml-pcm]" fmt,##args)
-- 
2.19.0

