From 1c7e3776ff2e208c5eac78f2795917999f226101 Mon Sep 17 00:00:00 2001
From: "shuai.cao" <shuai.cao@amlogic.com>
Date: Thu, 21 Nov 2013 14:06:37 +0800
Subject: [PATCH 2040/5965] reset ionvideo pts when steamon

---
 drivers/amlogic/ionvideo/ionvideo.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/amlogic/ionvideo/ionvideo.c b/drivers/amlogic/ionvideo/ionvideo.c
index 4943cd8a074c..8efe451fac43 100755
--- a/drivers/amlogic/ionvideo/ionvideo.c
+++ b/drivers/amlogic/ionvideo/ionvideo.c
@@ -352,6 +352,7 @@ static void buffer_queue(struct vb2_buffer *vb) {
 static int start_streaming(struct vb2_queue *vq, unsigned int count) {
     struct ionvideo_dev *dev = vb2_get_drv_priv(vq);
     is_actived = 1;
+    dev->receiver_register = 1;
     dprintk(dev, 2, "%s\n", __func__);
     return ionvideo_start_generating(dev);
 }
-- 
2.19.0

