From 4cd8999248b73fd19f27fc3c1beb4e268343e887 Mon Sep 17 00:00:00 2001
From: "xiaoyu.yuan" <xiaoyu.yuan@droid05.amlogic.com>
Date: Thu, 21 Nov 2013 14:45:15 +0800
Subject: [PATCH 2043/5965] add missing ae flag

---
 drivers/amlogic/camera/ov5647.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/amlogic/camera/ov5647.c b/drivers/amlogic/camera/ov5647.c
index 691252fccc66..de560e9e0d76 100755
--- a/drivers/amlogic/camera/ov5647.c
+++ b/drivers/amlogic/camera/ov5647.c
@@ -3478,6 +3478,7 @@ static int vidioc_streamoff(struct file *file, void *priv, enum v4l2_buf_type i)
         dev->vops->stop_tvin_service(0);
         fh->stream_on        = 0;
 	}
+	dev->ae_on = false;
 	return ret;
 }
 
-- 
2.19.0

