From 43d8b1ef8250705d07fadceefa44f91b32e08328 Mon Sep 17 00:00:00 2001
From: Greg Kroah-Hartman <gregkh@linuxfoundation.org>
Date: Fri, 29 Nov 2013 11:12:03 -0800
Subject: [PATCH 2173/5965] Linux 3.10.21

---
 Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index ba784b7b0f51..c0635fbab87f 100644
--- a/Makefile
+++ b/Makefile
@@ -1,6 +1,6 @@
 VERSION = 3
 PATCHLEVEL = 10
-SUBLEVEL = 20
+SUBLEVEL = 21
 EXTRAVERSION =
 NAME = TOSSUG Baby Fish
 
-- 
2.19.0

