From 4c017978c97d579dbe6e58af51a6f84e146f43bb Mon Sep 17 00:00:00 2001
From: "zhenfei.li" <zhenfei.li@amlogic>
Date: Fri, 29 Nov 2013 20:28:38 +0800
Subject: [PATCH 2176/5965] PD #83293: fixed camera flash light bugs

---
 drivers/amlogic/camera/ov5647.c | 9 ++++++++-
 1 file changed, 8 insertions(+), 1 deletion(-)

diff --git a/drivers/amlogic/camera/ov5647.c b/drivers/amlogic/camera/ov5647.c
index 4c07cf156142..37a5fa8a6c91 100755
--- a/drivers/amlogic/camera/ov5647.c
+++ b/drivers/amlogic/camera/ov5647.c
@@ -3935,11 +3935,18 @@ static int vidioc_s_input(struct file *file, void *priv, unsigned int i)
 static int vidioc_queryctrl(struct file *file, void *priv,
 			    struct v4l2_queryctrl *qc)
 {
+	struct ov5647_fh *fh = priv;
+	struct ov5647_device *dev = fh->dev;
 	int i;
 
 	for (i = 0; i < ARRAY_SIZE(ov5647_qctrl); i++)
 		if (qc->id && qc->id == ov5647_qctrl[i].id) {
-			memcpy(qc, &(ov5647_qctrl[i]),sizeof(*qc));
+			if (V4L2_CID_BACKLIGHT_COMPENSATION == ov5647_qctrl[i].id) {
+				if (dev->cam_info.flash_support) 
+					memcpy(qc, &(ov5647_qctrl[i]),sizeof(*qc));
+			} else {
+				memcpy(qc, &(ov5647_qctrl[i]),sizeof(*qc));
+			}
 			return (0);
 		}
 	return -EINVAL;
-- 
2.19.0

