From 5e8f05b878c43e1e976693901de924499aedf9bc Mon Sep 17 00:00:00 2001
From: "shi.liu" <shi.liu@amlogic.com>
Date: Tue, 3 Dec 2013 13:38:53 +0800
Subject: [PATCH 2189/5965] PD  #83354: cpufreq: Optimize performance

---
 drivers/cpufreq/cpufreq_hotplug.c | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/drivers/cpufreq/cpufreq_hotplug.c b/drivers/cpufreq/cpufreq_hotplug.c
index fa118c1f7023..8016df58fa42 100755
--- a/drivers/cpufreq/cpufreq_hotplug.c
+++ b/drivers/cpufreq/cpufreq_hotplug.c
@@ -694,8 +694,7 @@ static void hg_check_cpu(int cpu, unsigned int max_load)
 		unsigned int freq_next;
 
 		freq_next = max_load_freq /
-				(hg_tuners->up_threshold -
-				 hg_tuners->down_differential);
+				(hg_tuners->down_threshold);
 
 		if (freq_next < policy->min)
 			freq_next = policy->min;
-- 
2.19.0

