From 2ad7794072f78fdd35cb614bcc2b5313e326657f Mon Sep 17 00:00:00 2001
From: "zhi.zhou" <zhi.zhou@amlogic.com>
Date: Tue, 3 Dec 2013 17:17:29 +0800
Subject: [PATCH 2374/5965] pd#83596,fixed vpp_process_speed_check div0 crash
 bug.

---
 drivers/amlogic/amports/vpp.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/amlogic/amports/vpp.c b/drivers/amlogic/amports/vpp.c
index 55d36a4616c0..eb7d03e34949 100755
--- a/drivers/amlogic/amports/vpp.c
+++ b/drivers/amlogic/amports/vpp.c
@@ -285,7 +285,7 @@ vpp_process_speed_check(s32 width_in,
     }
 
     if (height_in > height_out) {
-        if (div_u64(VPP_SPEED_FACTOR * width_in * height_in * vinfo->sync_duration_num * height_screen,
+        if (height_out == 0 || div_u64(VPP_SPEED_FACTOR * width_in * height_in * vinfo->sync_duration_num * height_screen,
                     height_out * vinfo->sync_duration_den * 256) > get_vpu_clk()) {
             return SPEED_CHECK_VSKIP;
         } else {
-- 
2.19.0

