From 79338d23a96fbc604ea31d2016eafd73f03f2c7e Mon Sep 17 00:00:00 2001
From: timyao <tim.yao@amlogic.com>
Date: Sat, 7 Dec 2013 16:34:41 -0800
Subject: [PATCH 2381/5965] Use CLK_GATE_ON/OFF() to control VPU interrupt clk
 gates.

---
 drivers/amlogic/amports/amstream.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/drivers/amlogic/amports/amstream.c b/drivers/amlogic/amports/amstream.c
index e810c27fb1fe..cba4db3a1c01 100755
--- a/drivers/amlogic/amports/amstream.c
+++ b/drivers/amlogic/amports/amstream.c
@@ -1046,6 +1046,7 @@ static int amstream_open(struct inode *inode, struct file *file)
 
 #if MESON_CPU_TYPE >= MESON_CPU_TYPE_MESON8
     CLK_GATE_ON(HIU_PARSER_TOP);
+    CLK_GATE_ON(VPU_INTR);
 
     if (this->type & PORT_TYPE_VIDEO) {
         CLK_GATE_ON(DOS);
@@ -1108,6 +1109,8 @@ static int amstream_release(struct inode *inode, struct file *file)
     }
 
     CLK_GATE_OFF(HIU_PARSER_TOP);
+    CLK_GATE_OFF(VPU_INTR);
+
 #elif MESON_CPU_TYPE >= MESON_CPU_TYPE_MESON6
     switch_mod_gate_by_name("audio", 0);
     switch_mod_gate_by_name("vdec", 0);
-- 
2.19.0

