From 38c8082dbea3a52eaf152280de60309fa876d13d Mon Sep 17 00:00:00 2001
From: jiwei chen <jiwei.chen@amlogic.com>
Date: Thu, 12 Dec 2013 12:21:24 +0800
Subject: [PATCH 2461/5965] PD #83057: disable cm2 when stop dec.

---
 drivers/amlogic/tvin/vdin/vdin_ctl.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/drivers/amlogic/tvin/vdin/vdin_ctl.c b/drivers/amlogic/tvin/vdin/vdin_ctl.c
index dcc91242984c..a063c561d88a 100755
--- a/drivers/amlogic/tvin/vdin/vdin_ctl.c
+++ b/drivers/amlogic/tvin/vdin/vdin_ctl.c
@@ -1606,6 +1606,8 @@ inline void vdin_hw_enable(unsigned int offset)
 
 inline void vdin_hw_disable(unsigned int offset)
 {
+	/* disable cm2 */
+	WR_BITS(VDIN_CM_BRI_CON_CTRL,0,CM_TOP_EN_BIT,CM_TOP_EN_WID);
 	/* disable video data input */
 	// [    4]  top.datapath_en  = 0
 	WR_BITS(VDIN_COM_CTRL0, 0, 4, 1);
-- 
2.19.0

