From 05bcf8f867f4af11c93395d4a6dd1dd52d8904ea Mon Sep 17 00:00:00 2001
From: Greg Kroah-Hartman <gregkh@linuxfoundation.org>
Date: Wed, 11 Dec 2013 22:36:44 -0800
Subject: [PATCH 2515/5965] Linux 3.10.24

---
 Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index c6d2ec5e9580..ae6ccebf8af4 100644
--- a/Makefile
+++ b/Makefile
@@ -1,6 +1,6 @@
 VERSION = 3
 PATCHLEVEL = 10
-SUBLEVEL = 23
+SUBLEVEL = 24
 EXTRAVERSION =
 NAME = TOSSUG Baby Fish
 
-- 
2.19.0

