From 1bc2c1426b9e0a71deea5946f559b0326db82c28 Mon Sep 17 00:00:00 2001
From: "qi.duan" <qi.duan@amlogic.com>
Date: Fri, 13 Dec 2013 21:11:08 +0800
Subject: [PATCH 2524/5965] PD #82722 amlvideo2 num source for new reserve
 memory

---
 drivers/amlogic/video_dev/amlvideo2.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/drivers/amlogic/video_dev/amlvideo2.c b/drivers/amlogic/video_dev/amlvideo2.c
index ad4a52d52c35..ef4b0eefd878 100755
--- a/drivers/amlogic/video_dev/amlvideo2.c
+++ b/drivers/amlogic/video_dev/amlvideo2.c
@@ -2030,6 +2030,9 @@ static int amlvideo2_driver_probe(struct platform_device *pdev)
 {
 	int ret = 0;
 	struct amlvideo2_device *dev = NULL;
+	
+	if(of_get_property(pdev->dev.of_node, "reserve-memory", NULL))
+		pdev->num_resources = MAX_SUB_DEV_NODE;
 	if (pdev->num_resources == 0) {
 		dev_err(&pdev->dev, "probed for an unknown device\n");
 		return -ENODEV;
-- 
2.19.0

