From e35de33541373766afded39cb14cab0f4496c120 Mon Sep 17 00:00:00 2001
From: "zhenfei.li" <zhenfei.li@amlogic>
Date: Thu, 19 Dec 2013 12:29:48 +0800
Subject: [PATCH 2557/5965] PD #84692: add ov5647 with resolution 320X240.

---
 drivers/amlogic/camera/ov5647.c | 24 +++++++++++++++++++-----
 1 file changed, 19 insertions(+), 5 deletions(-)

diff --git a/drivers/amlogic/camera/ov5647.c b/drivers/amlogic/camera/ov5647.c
index 9f7d10b35bda..65755199ec4e 100755
--- a/drivers/amlogic/camera/ov5647.c
+++ b/drivers/amlogic/camera/ov5647.c
@@ -1978,10 +1978,17 @@ static resolution_param_t  debug_prev_resolution_array[] = {
 		.reg_script[0]			= OV5647_preview_960P_script,
 		.reg_script[1]			= OV5647_VGA_script_mipi,
 	}, {
+		.frmsize			= {320, 240},
+		.active_frmsize		= {1280, 960},
+		.active_fps			= 30,
+		.size_type			= SIZE_320X240,
+		.reg_script[0]			= OV5647_preview_960P_script,
+		.reg_script[1]			= OV5647_VGA_script_mipi,
+	},{
 		.frmsize			= {640, 480},
 		.active_frmsize		= {1280, 960},
 		.active_fps			= 30,
-		.size_type			= SIZE_640X480,
+		.size_type			= SIZE_1280X960,
 		.reg_script[0]			= OV5647_preview_960P_script,
 		.reg_script[1]			= OV5647_VGA_script_mipi,
 	}, {
@@ -2019,17 +2026,24 @@ static resolution_param_t  debug_prev_resolution_array[] = {
 static resolution_param_t  prev_resolution_array[] = {
 	{
 		.frmsize			= {352, 288},
-		.active_frmsize		= {640, 480},
+		.active_frmsize		= {1280, 960},
 		.active_fps			= 30,
 		.size_type			= SIZE_352X288,
-		.reg_script[0]			= OV5647_preview_VGA_script,
+		.reg_script[0]			= OV5647_preview_960P_script,
+		.reg_script[1]			= OV5647_VGA_script_mipi,
+	}, {
+		.frmsize			= {320, 240},
+		.active_frmsize		= {1280, 960},
+		.active_fps			= 30,
+		.size_type			= SIZE_320X240,
+		.reg_script[0]			= OV5647_preview_960P_script,
 		.reg_script[1]			= OV5647_VGA_script_mipi,
 	}, {
 		.frmsize			= {640, 480},
-		.active_frmsize		= {640, 480},
+		.active_frmsize		= {1280, 960},
 		.active_fps			= 30,
 		.size_type			= SIZE_640X480,
-		.reg_script[0]			= OV5647_preview_VGA_script,
+		.reg_script[0]			= OV5647_preview_960P_script,
 		.reg_script[1]			= OV5647_VGA_script_mipi,
 	}, {
 		.frmsize			= {1280, 720},
-- 
2.19.0

