From 656462c5ad0c181191ca3894a15c0da0d203c4e6 Mon Sep 17 00:00:00 2001
From: Mengchun Li <mengchun.li@amlogic.com>
Date: Thu, 19 Dec 2013 20:47:47 +0800
Subject: [PATCH 2650/5965] PD# 84265: improve rtl8188eu power control

---
 drivers/amlogic/wifi/wifi_power.c | 2 --
 1 file changed, 2 deletions(-)

diff --git a/drivers/amlogic/wifi/wifi_power.c b/drivers/amlogic/wifi/wifi_power.c
index 155c796995a3..0fa65f983f0f 100755
--- a/drivers/amlogic/wifi/wifi_power.c
+++ b/drivers/amlogic/wifi/wifi_power.c
@@ -83,8 +83,6 @@ static int wifi_power_ioctl(struct file *filp, unsigned int cmd, unsigned long a
 	switch (cmd) 
 	{
     	case POWER_UP:
-    		usb_wifi_power(0);
-    		mdelay(100);
     		usb_wifi_power(1);
     		mdelay(500);
    	        usb_wifi_power(0);
-- 
2.19.0

