From 4114d2b120a042b9849e7e168db05004786c79ac Mon Sep 17 00:00:00 2001
From: timyao <tim.yao@amlogic.com>
Date: Fri, 27 Dec 2013 23:43:04 -0800
Subject: [PATCH 2673/5965] PD #84571. Fix video layer 2 enabling bug from Bug
 #84027.

---
 drivers/amlogic/amports/video.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/amlogic/amports/video.c b/drivers/amlogic/amports/video.c
index e6eb91f3ae2a..bfceb596f9be 100755
--- a/drivers/amlogic/amports/video.c
+++ b/drivers/amlogic/amports/video.c
@@ -2250,7 +2250,7 @@ exit:
              */
             video_onoff_state = VIDEO_ENABLE_STATE_ON_PENDING;
         } else if (video_onoff_state == VIDEO_ENABLE_STATE_ON_PENDING) {
-            SET_VCBUS_REG_MASK(VPP_MISC + cur_dev->vpp_off, VPP_VD1_PREBLEND|VPP_VD2_PREBLEND|VPP_VD2_POSTBLEND|VPP_VD1_POSTBLEND|VPP_POSTBLEND_EN);
+            SET_VCBUS_REG_MASK(VPP_MISC + cur_dev->vpp_off, VPP_VD1_PREBLEND|VPP_VD1_POSTBLEND|VPP_POSTBLEND_EN);
 
             video_onoff_state = VIDEO_ENABLE_STATE_IDLE;
 
-- 
2.19.0

