From 7fe2798f16f115cfb1e514aca84df26d2458962a Mon Sep 17 00:00:00 2001
From: "baoqi.wang" <baoqi.wang@amlogic.com>
Date: Mon, 30 Dec 2013 14:33:35 +0800
Subject: [PATCH 2677/5965] PD#83746 fixed null pointer for ethernet probe

---
 drivers/amlogic/ethernet/am_net8218.c | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/drivers/amlogic/ethernet/am_net8218.c b/drivers/amlogic/ethernet/am_net8218.c
index e6e1c7ee6005..f7456165a823 100755
--- a/drivers/amlogic/ethernet/am_net8218.c
+++ b/drivers/amlogic/ethernet/am_net8218.c
@@ -1015,7 +1015,8 @@ static int aml_phy_init(struct net_device *dev)
         snprintf(phy_id, MII_BUS_ID_SIZE + 3, PHY_ID_FMT, bus_id,
                  priv->phy_addr);
         printk("aml_phy_init:  trying to attach to %s\n", phy_id);
-	 priv->phydev->drv->resume(priv->phydev);
+	if(priv->phydev)
+		 priv->phydev->drv->resume(priv->phydev);
         phydev = phy_connect(dev, phy_id, &aml_adjust_link, priv->phy_interface);
 
         if (IS_ERR(phydev)) {
@@ -2394,7 +2395,8 @@ static int ethernet_probe(struct platform_device *pdev)
 
 	eth_pdata = (struct aml_eth_platdata *)pdev->dev.platform_data;
 	struct am_net_private *np = netdev_priv(my_ndev);
-	np->phydev->drv->suspend(np->phydev);
+	if(np->phydev)
+		np->phydev->drv->suspend(np->phydev);
 	switch_mod_gate_by_name("ethernet",0);
 	if (!eth_pdata) {
 		printk("\nethernet pm ops resource undefined.\n");
-- 
2.19.0

