From e3f936183d2408a5211d3c1d518a68fc3f5e51db Mon Sep 17 00:00:00 2001
From: "Lujian.hu" <Lujian.hu@amlogic.com>
Date: Tue, 31 Dec 2013 16:25:19 +0800
Subject: [PATCH 2685/5965] pd#81327:fix compile error

---
 drivers/amlogic/amports/subtitle.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/amlogic/amports/subtitle.c b/drivers/amlogic/amports/subtitle.c
index 7745d0b5f396..069f0fa305a0 100755
--- a/drivers/amlogic/amports/subtitle.c
+++ b/drivers/amlogic/amports/subtitle.c
@@ -8,7 +8,7 @@
 #include <linux/cdev.h>
 #include <linux/fs.h>
 #include <linux/interrupt.h>
-#include <linux/amports/amstream.h>
+#include <linux/amlogic/amports/amstream.h>
 
 #include <linux/amlogic/amlog.h>
 MODULE_AMLOG(AMLOG_DEFAULT_LEVEL, 0, LOG_DEFAULT_LEVEL_DESC, LOG_DEFAULT_MASK_DESC);
-- 
2.19.0

