From be827cf2ed00948e29e875d9579f35d468e01714 Mon Sep 17 00:00:00 2001
From: "tao.zeng" <tao.zeng@amlogic.com>
Date: Thu, 2 Jan 2014 15:59:44 +0800
Subject: [PATCH 2694/5965] PD#85679:close vccx2/vccx3 before PMU power off

---
 drivers/amlogic/power/ricoh/rn5t618-sply.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/drivers/amlogic/power/ricoh/rn5t618-sply.c b/drivers/amlogic/power/ricoh/rn5t618-sply.c
index 3e1452234fc7..da2ccdcc9eb5 100755
--- a/drivers/amlogic/power/ricoh/rn5t618-sply.c
+++ b/drivers/amlogic/power/ricoh/rn5t618-sply.c
@@ -194,6 +194,9 @@ EXPORT_SYMBOL_GPL(rn5t618_get_gpio);
 
 void rn5t618_power_off()
 {
+    rn5t618_set_gpio(0, 1);
+    rn5t618_set_gpio(1, 1);
+    msleep(100);
     RICOH_DBG("%s, send power off command\n", __func__);
     rn5t618_set_bits(0x000f, 0x00, 0x01);                       // do not re-power-on system
     rn5t618_set_bits(0x000E, 0x01, 0x01);                       // software power off PMU
-- 
2.19.0

