From 1be6098022dcd8a695878be7d24419a7c82392d8 Mon Sep 17 00:00:00 2001
From: Mengchun Li <mengchun.li@amlogic.com>
Date: Thu, 2 Jan 2014 19:29:26 +0800
Subject: [PATCH 2699/5965] PD# 85616: Fix wifi can't open issue

---
 drivers/amlogic/wifi/wifi_power.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/amlogic/wifi/wifi_power.c b/drivers/amlogic/wifi/wifi_power.c
index b28a1beec739..f0e898b955aa 100755
--- a/drivers/amlogic/wifi/wifi_power.c
+++ b/drivers/amlogic/wifi/wifi_power.c
@@ -97,6 +97,7 @@ static int wifi_power_ioctl(struct file *filp, unsigned int cmd, unsigned long a
     		usb_wifi_power(1);
     		mdelay(500);
    	        usb_wifi_power(0);
+   	        mdelay(500);
     
     		printk(KERN_INFO "Set usb wifi power up!\n");
     		break;
-- 
2.19.0

