From 49173a0d55f9b41a19424fae1fdcecee930de289 Mon Sep 17 00:00:00 2001
From: Lawrence Mok <lawrence.mok@amlogic.com>
Date: Fri, 3 Jan 2014 17:16:17 -0800
Subject: [PATCH 2700/5965] fix memset arguments

---
 drivers/amlogic/efuse/efuse_bch_8.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/amlogic/efuse/efuse_bch_8.c b/drivers/amlogic/efuse/efuse_bch_8.c
index 772754ca740c..32130da40f54 100755
--- a/drivers/amlogic/efuse/efuse_bch_8.c
+++ b/drivers/amlogic/efuse/efuse_bch_8.c
@@ -148,7 +148,7 @@ int efuse_bch_enc(const char *ibuf, int isize, char *obuf, int reverse)
 		printk("malloc buffer error for efuse\n");
 		return -1;
 	}
-	memset(c,sizeof(int) * 255,0);
+	memset(c, 0, sizeof(int) * 255);
 	
 
 	for (i = 0; i < isize; ++i) {
@@ -201,7 +201,7 @@ int efuse_bch_dec(const char *ibuf, int isize, char *obuf, int reverse)
 		printk("malloc buffer error for efuse\n");
 		return -1;
 	}
-	memset(c,sizeof(int) * 255,0);
+	memset(c, 0, sizeof(int) * 255);
 
 	for (i = 0; i < isize; ++i) {
 		info = ibuf[i];
-- 
2.19.0

