From d17a299d22ff86237c7ce61ce262e2c50a7ada8d Mon Sep 17 00:00:00 2001
From: "Lujian.hu" <Lujian.hu@amlogic.com>
Date: Fri, 3 Jan 2014 09:24:31 +0800
Subject: [PATCH 2705/5965] PD #83217: fix kernel class name error.

---
 drivers/amlogic/amports/subtitle.c | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/drivers/amlogic/amports/subtitle.c b/drivers/amlogic/amports/subtitle.c
index 069f0fa305a0..98ce8a65a4fa 100755
--- a/drivers/amlogic/amports/subtitle.c
+++ b/drivers/amlogic/amports/subtitle.c
@@ -13,6 +13,8 @@
 #include <linux/amlogic/amlog.h>
 MODULE_AMLOG(AMLOG_DEFAULT_LEVEL, 0, LOG_DEFAULT_LEVEL_DESC, LOG_DEFAULT_MASK_DESC);
 #define DEVICE_NAME "amsubtitle"	/* Dev name as it appears in /proc/devices   */
+#define DEVICE_CLASS_NAME "subtitle"
+
 static int subdevice_open = 0;
 
 #define MAX_SUBTITLE_PACKET 10
@@ -644,7 +646,7 @@ static int __init subtitle_init(void)
         return ret;
     }
 
-    amsub_clsp = class_create(THIS_MODULE, DEVICE_NAME);
+    amsub_clsp = class_create(THIS_MODULE, DEVICE_CLASS_NAME);
     if(IS_ERR(amsub_clsp)){
         ret = PTR_ERR(amsub_clsp);
         goto err1;
-- 
2.19.0

