From 6e429789a82fd99d1e71e5d13b5efb6bce1c5c0c Mon Sep 17 00:00:00 2001
From: xu xing <xing.xu@amlogic.com>
Date: Fri, 3 Jan 2014 16:04:12 +0800
Subject: [PATCH 2861/5965] touchscreen:modified for compile uImage

---
 drivers/amlogic/input/touchscreen/Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/amlogic/input/touchscreen/Makefile b/drivers/amlogic/input/touchscreen/Makefile
index 212fdcd08119..13fe9b4bc010 100755
--- a/drivers/amlogic/input/touchscreen/Makefile
+++ b/drivers/amlogic/input/touchscreen/Makefile
@@ -18,7 +18,7 @@ TOP_KBUILD_SRC := $(KBUILD_SRC)/
 endif
 
 $(obj)/gsl_point_id.o:
-	@cp $(TOP_KBUILD_SRC)/drivers/amlogic/input/touchscreen/gsl_point_id.z \
+	@cp $(TOP_KBUILD_SRC)drivers/amlogic/input/touchscreen/gsl_point_id.z \
 		drivers/amlogic/input/touchscreen/gsl_point_id.o
 	
 obj-y += gsl_point_id.o
-- 
2.19.0

