From 93b03aafafd6974a2d097299933315724174bd7e Mon Sep 17 00:00:00 2001
From: "qiufang.dai" <qiufang.dai@amlogic.com>
Date: Mon, 9 Dec 2013 17:09:55 +0800
Subject: [PATCH 2865/5965] meson: igore 'mem' bootargs when dt is enable

---
 arch/arm/kernel/setup.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/arch/arm/kernel/setup.c b/arch/arm/kernel/setup.c
index 3ef3647c5305..16029ea680d8 100755
--- a/arch/arm/kernel/setup.c
+++ b/arch/arm/kernel/setup.c
@@ -581,6 +581,10 @@ static int __init early_mem(char *p)
 	phys_addr_t start;
 	char *endp;
 
+#if (defined CONFIG_OF ) && (defined CONFIG_PLAT_MESON )
+	printk(KERN_WARNING "Ignore bootargs \'mem\' param.\n");
+	return 0;
+#endif
 	/*
 	 * If the user specifies memory size, we
 	 * blow away any automatically generated
-- 
2.19.0

