From a7d22e8af4b8555246c6604cd6e7307e3493bafe Mon Sep 17 00:00:00 2001
From: "tao.zeng" <tao.zeng@amlogic.com>
Date: Tue, 10 Dec 2013 20:18:40 +0800
Subject: [PATCH 2871/5965] PD #00002: Fix compile bug of PMU lib

---
 drivers/amlogic/power/Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/amlogic/power/Makefile b/drivers/amlogic/power/Makefile
index ba25ff06be89..70588324b58c 100755
--- a/drivers/amlogic/power/Makefile
+++ b/drivers/amlogic/power/Makefile
@@ -13,7 +13,7 @@ ifneq ($(KBUILD_SRC),)
 TOP_KBUILD_SRC := $(KBUILD_SRC)/
 endif
 
-MAKE_FOR_LIB := y
+MAKE_FOR_LIB := n
 ifeq ($(MAKE_FOR_LIB), n)
 $(obj)/aml_pmu_algorithm_lib:
 	@cp $(TOP_KBUILD_SRC)drivers/amlogic/power/aml_pmu_algorithm_lib.z \
-- 
2.19.0

