From f658e86478b5a92989f9770db3333ab2118c6a65 Mon Sep 17 00:00:00 2001
From: xu xing <xing.xu@amlogic.com>
Date: Fri, 3 Jan 2014 20:10:27 +0800
Subject: [PATCH 2881/5965] saradc:get cpu temp api only for M8 thermal driver

---
 drivers/amlogic/input/saradc/saradc.c | 9 ++++++++-
 1 file changed, 8 insertions(+), 1 deletion(-)

diff --git a/drivers/amlogic/input/saradc/saradc.c b/drivers/amlogic/input/saradc/saradc.c
index 50a1e6187587..2a789059a6f2 100755
--- a/drivers/amlogic/input/saradc/saradc.c
+++ b/drivers/amlogic/input/saradc/saradc.c
@@ -420,7 +420,7 @@ static struct class saradc_class = {
     .name = "saradc",
     .class_attrs = saradc_class_attrs,
 };
-
+#ifdef CONFIG_ARCH_MESON8
 int get_cpu_temp()
 {
 	int ret=-1,tempa;
@@ -436,6 +436,13 @@ int get_cpu_temp()
 	}
 	return ret;
 }
+#else
+int get_cpu_temp()
+{
+
+	return NOT_WRITE_EFUSE;
+}
+#endif
 static int saradc_probe(struct platform_device *pdev)
 {
 	int err;
-- 
2.19.0

