From 1daa81651067a84740904e29aa6b860bc68cffa4 Mon Sep 17 00:00:00 2001
From: "sheng.luo" <sheng.luo@amlogic.com>
Date: Fri, 13 Dec 2013 13:29:23 +0800
Subject: [PATCH 2889/5965] fix 3.10 write boot crash bug

---
 drivers/amlogic/nand/m3_nand.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/amlogic/nand/m3_nand.c b/drivers/amlogic/nand/m3_nand.c
index 02adb7df3159..dfd2b53e3ae0 100755
--- a/drivers/amlogic/nand/m3_nand.c
+++ b/drivers/amlogic/nand/m3_nand.c
@@ -869,7 +869,7 @@ exit:
 	return;
 }
 
- int m3_nand_boot_write_page(struct mtd_info *mtd, struct nand_chip *chip, const uint8_t *buf,int oob_required, int page, int cached, int raw)
+ int m3_nand_boot_write_page(struct mtd_info *mtd, struct nand_chip *chip,uint32_t offset, int data_len, const uint8_t *buf,int oob_required, int page, int cached, int raw)
 {
 	struct aml_nand_chip *aml_chip = mtd_to_nand_chip(mtd);
 	int status, i, write_page, configure_data, pages_per_blk, ran_mode;
-- 
2.19.0

