From ef174e0ad4b38d1facfd55efe668141263467e46 Mon Sep 17 00:00:00 2001
From: "qiufang.dai" <qiufang.dai@amlogic.com>
Date: Mon, 16 Dec 2013 20:38:38 +0800
Subject: [PATCH 2906/5965] m6: dvfs: m6/m8 use unified cpufreq driver

---
 drivers/amlogic/cpufreq/Kconfig | 14 +++++++-------
 1 file changed, 7 insertions(+), 7 deletions(-)

diff --git a/drivers/amlogic/cpufreq/Kconfig b/drivers/amlogic/cpufreq/Kconfig
index 2e29f4f6b6b7..46f2fa3cf240 100644
--- a/drivers/amlogic/cpufreq/Kconfig
+++ b/drivers/amlogic/cpufreq/Kconfig
@@ -5,10 +5,10 @@ config AMLOGIC_MESON_CPUFREQ
 	help
 	  This adds the CPUFreq driver for Meson series CPU
 
-config ARM_MESON_CPUFREQ
-	bool "[obsoleted] Meson6 CPU Frequency scaling support"
-	depends on ARCH_MESON6
-	default y
-	help
-	   This adds the CPUFreq driver for Meson CPU.
-	   This driver is not clean, it's obsolete.
+#config ARM_MESON_CPUFREQ
+#	bool "[obsoleted] Meson6 CPU Frequency scaling support"
+#	depends on ARCH_MESON6
+#	default y
+#	help
+#	   This adds the CPUFreq driver for Meson CPU.
+#	   This driver is not clean, it's obsolete.
-- 
2.19.0

