From 065160b20e62249064d31d71f51dfb6f67f81204 Mon Sep 17 00:00:00 2001
From: "qiufang.dai" <qiufang.dai@amlogic.com>
Date: Tue, 17 Dec 2013 10:38:17 +0800
Subject: [PATCH 2907/5965] m6: DVFS: change cpufreq driver default enable

---
 drivers/amlogic/cpufreq/Kconfig | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/amlogic/cpufreq/Kconfig b/drivers/amlogic/cpufreq/Kconfig
index 46f2fa3cf240..984334a23ba1 100644
--- a/drivers/amlogic/cpufreq/Kconfig
+++ b/drivers/amlogic/cpufreq/Kconfig
@@ -1,7 +1,7 @@
 config AMLOGIC_MESON_CPUFREQ
 	bool "Meson CPU Frequency scaling support"
 	depends on PLAT_MESON
-	default n
+	default y
 	help
 	  This adds the CPUFreq driver for Meson series CPU
 
-- 
2.19.0

