From edbaa6aaa209efda7a775a6b32e128a48a3e6ec2 Mon Sep 17 00:00:00 2001
From: "bo.yang" <bo.yang@amlogic.com>
Date: Fri, 20 Dec 2013 12:04:38 +0800
Subject: [PATCH 2915/5965] add file header

---
 include/linux/amlogic/gpio-amlogic.h | 24 ++++++++++++++++++++++--
 1 file changed, 22 insertions(+), 2 deletions(-)

diff --git a/include/linux/amlogic/gpio-amlogic.h b/include/linux/amlogic/gpio-amlogic.h
index 99bb889b6682..0227f81a60e6 100755
--- a/include/linux/amlogic/gpio-amlogic.h
+++ b/include/linux/amlogic/gpio-amlogic.h
@@ -1,9 +1,26 @@
+/*
+ * include/linux/amlogic/gpio-amlogic.h
+ *
+ * Copyright (C) 2013 Amlogic, Inc.
+ *
+ * This software is licensed under the terms of the GNU General Public
+ * License version 2, as published by the Free Software Foundation, and
+ * may be copied, distributed, and modified under those terms.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ */
+
+#ifndef __LINUX_AMLOGIC_GPIO_AMLOGIC_H
+#define __LINUX_AMLOGIC_GPIO_AMLOGIC_H
 
 #define GPIO_REG_BIT(reg,bit) ((reg<<5)|bit)
 #define GPIO_REG(value) ((value>>5))
 #define GPIO_BIT(value) ((value&0x1F))
 
-struct amlogic_gpio_desc{
+struct amlogic_gpio_desc {
 	unsigned num;
 	char *name;
 	unsigned int out_en_reg_bit;
@@ -13,6 +30,9 @@ struct amlogic_gpio_desc{
 	unsigned int pull_up_reg_bit;
 	const char *gpio_owner;
 };
-struct amlogic_set_pullup{
+
+struct amlogic_set_pullup {
 	 int (*meson_set_pullup)(unsigned int,unsigned int,unsigned int);
 };
+
+#endif // __LINUX_AMLOGIC_GPIO_AMLOGIC_H
\ No newline at end of file
-- 
2.19.0

