From b2bbe86968386955bf609d336f44a5fa893f7bfe Mon Sep 17 00:00:00 2001
From: "yun.cai" <yun.cai@amlogic.com>
Date: Tue, 24 Dec 2013 17:15:33 +0800
Subject: [PATCH 2918/5965] M6: remove warning--section mismatch

---
 drivers/amlogic/bluetooth/bt_device.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/amlogic/bluetooth/bt_device.c b/drivers/amlogic/bluetooth/bt_device.c
index b0e054444621..4ba840847ec8 100755
--- a/drivers/amlogic/bluetooth/bt_device.c
+++ b/drivers/amlogic/bluetooth/bt_device.c
@@ -130,7 +130,7 @@ static int bt_resume(struct platform_device *pdev)
     return 0;
 }
 
-static int __init bt_probe(struct platform_device *pdev)
+static int bt_probe(struct platform_device *pdev)
 {
 	int ret = 0;
 	struct rfkill *bt_rfk;
-- 
2.19.0

