From 8621922687f3edc058063115cc98ea34c4706f13 Mon Sep 17 00:00:00 2001
From: xu xing <xing.xu@amlogic.com>
Date: Tue, 7 Jan 2014 11:01:24 +0800
Subject: [PATCH 2919/5965] remove warning "config dependencies" and  "section
 mismatch"

---
 drivers/cpufreq/cpufreq_hotplug.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/cpufreq/cpufreq_hotplug.c b/drivers/cpufreq/cpufreq_hotplug.c
index 8016df58fa42..fe053d51e4d9 100755
--- a/drivers/cpufreq/cpufreq_hotplug.c
+++ b/drivers/cpufreq/cpufreq_hotplug.c
@@ -510,7 +510,7 @@ wait_next_event:
 		set_current_state(TASK_RUNNING);
 	}
 }
-static void cpu_hotplug_thread(int *hotplug_flag)
+static void __cpuinit cpu_hotplug_thread(int *hotplug_flag)
 {
 	int i, j,target_cpu = 1;
 	unsigned long flags, cpu_down_num;
-- 
2.19.0

