From c2fe23f72eebed8f6cba6a8767923d10c307885a Mon Sep 17 00:00:00 2001
From: "qi.duan" <qi.duan@amlogic.com>
Date: Thu, 9 Jan 2014 23:13:30 -0800
Subject: [PATCH 2963/5965] remove audio_data.o for new reserve memory

---
 drivers/amlogic/amaudio/Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/amlogic/amaudio/Makefile b/drivers/amlogic/amaudio/Makefile
index df67286ff3e5..30c47a1d129f 100755
--- a/drivers/amlogic/amaudio/Makefile
+++ b/drivers/amlogic/amaudio/Makefile
@@ -13,7 +13,7 @@ endif
 $(obj)/audio_data.o:
 	@cp $(TOP_KBUILD_SRC)drivers/amlogic/amaudio/audio_data.z \
         drivers/amlogic/amaudio/audio_data.o
-amlogic_amaudio-objs += audio_data.o
+#amlogic_amaudio-objs += audio_data.o
 endif
 
 obj-$(CONFIG_AMAUDIO) += amlogic_amaudio.o
-- 
2.19.0

