From ad3d54ba94fd31839bd8b8c1972074d086359ff3 Mon Sep 17 00:00:00 2001
From: timyao <tim.yao@amlogic.com>
Date: Fri, 3 Jan 2014 15:45:21 -0800
Subject: [PATCH 2965/5965] PD #85802. Apply new play rate for HLS switching.

---
 drivers/amlogic/amports/vh264.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/drivers/amlogic/amports/vh264.c b/drivers/amlogic/amports/vh264.c
index b21e5880de61..e8e1ab83a643 100755
--- a/drivers/amlogic/amports/vh264.c
+++ b/drivers/amlogic/amports/vh264.c
@@ -701,8 +701,9 @@ static void vh264_set_params(void)
 
     if (timing_info_present_flag) {
         if (((num_units_in_tick * 120) >= time_scale && (!sync_outside)) && num_units_in_tick && time_scale) {
-			if(use_idr_framerate || !frame_dur || !duration_from_pts_done )
+            if (use_idr_framerate || !frame_dur || !duration_from_pts_done || vh264_running) {
             	frame_dur = div_u64(96000ULL * 2 * num_units_in_tick, time_scale);
+            }
         }
     }
 
-- 
2.19.0

