From 5a644c3d987a07a9f79a5446bccaad28f5a974d6 Mon Sep 17 00:00:00 2001
From: "zhenfei.li" <zhenfei.li@amlogic>
Date: Tue, 7 Jan 2014 15:36:50 +0800
Subject: [PATCH 2969/5965] PD #85846: reexcute focus command when camera is
 stream on.

---
 drivers/amlogic/camera/ov5647.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/amlogic/camera/ov5647.c b/drivers/amlogic/camera/ov5647.c
index b2f617f7ab38..a4f108e93fc2 100755
--- a/drivers/amlogic/camera/ov5647.c
+++ b/drivers/amlogic/camera/ov5647.c
@@ -3845,6 +3845,7 @@ static int vidioc_streamon(struct file *file, void *priv, enum v4l2_buf_type i)
     OV5647_set_param_wb(fh->dev,ov5647_qctrl[4].default_value);
     OV5647_set_param_exposure(fh->dev,ov5647_qctrl[5].default_value);
     OV5647_set_param_effect(fh->dev,ov5647_qctrl[6].default_value);
+    OV5647_AutoFocus(fh->dev, ov5647_qctrl[8].default_value);
     return ret;
 }
 
-- 
2.19.0

