From 2ac149c531846fcec75f620280baeaeecdcceafe Mon Sep 17 00:00:00 2001
From: "zhenfei.li" <zhenfei.li@amlogic>
Date: Wed, 15 Jan 2014 19:46:55 +0800
Subject: [PATCH 3058/5965] PD #86508: increase null frame waiting time.

---
 drivers/amlogic/camera/common/vm.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/amlogic/camera/common/vm.c b/drivers/amlogic/camera/common/vm.c
index 267dfa99917b..12de0dcd0a9f 100755
--- a/drivers/amlogic/camera/common/vm.c
+++ b/drivers/amlogic/camera/common/vm.c
@@ -1390,7 +1390,7 @@ static int vm_task(void *data) {
 
 		/* wait for frame from 656 provider until 500ms runs out */
 		//vf = local_vf_peek();
-		wait_event_interruptible_timeout(frame_ready, is_vf_available(),msecs_to_jiffies(1000));
+		wait_event_interruptible_timeout(frame_ready, is_vf_available(),msecs_to_jiffies(5000));
                   
 		if(!task_running){
 			up(&vb_done_sema);
-- 
2.19.0

