From f695f39dc7f241ce3fc209995561374f91694c38 Mon Sep 17 00:00:00 2001
From: "tao.dong" <tao.dong@amlogic.com>
Date: Thu, 16 Jan 2014 20:26:03 +0800
Subject: [PATCH 3064/5965] PD # 86569 : fix suspend crash by Wangbaoqi

---
 drivers/amlogic/ethernet/am_net8218.c | 6 ------
 1 file changed, 6 deletions(-)

diff --git a/drivers/amlogic/ethernet/am_net8218.c b/drivers/amlogic/ethernet/am_net8218.c
index 5cf6c07561ba..847fb26ad59b 100755
--- a/drivers/amlogic/ethernet/am_net8218.c
+++ b/drivers/amlogic/ethernet/am_net8218.c
@@ -1184,10 +1184,6 @@ static int netdev_close(struct net_device *dev)
 	if (g_debug > 0) {
 		printk(KERN_DEBUG "%s: closed\n", dev->name);
 	}
-	np->refcnt--;
-	if(np->refcnt == 0){
-		switch_mod_gate_by_name("ethernet",0);
-	}
 	return 0;
 }
 
@@ -2332,7 +2328,6 @@ static int ethernet_early_suspend(struct early_suspend *dev)
 {
 	printk("ethernet_early_suspend!\n");
 	netdev_close(my_ndev);
-	switch_mod_gate_by_name("ethernet",0);
 	return 0;
 }
 static int ethernet_late_resume(struct early_suspend *dev)
@@ -2446,7 +2441,6 @@ static int ethernet_suspend(struct platform_device *dev, pm_message_t event)
 {
 	printk("ethernet_suspend!\n");
 	netdev_close(my_ndev);	
-	switch_mod_gate_by_name("ethernet",0);
 	return 0;
 }
 
-- 
2.19.0

