From 081a242ba71945c0b1c14a2dd38f0ee9348a6b46 Mon Sep 17 00:00:00 2001
From: "chunhai.guo" <chunhai.guo@amlogic.com>
Date: Wed, 22 Jan 2014 16:06:33 +0800
Subject: [PATCH 3099/5965] PD #85897: fix MX compile error

---
 drivers/amlogic/mmc/amlsd.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/drivers/amlogic/mmc/amlsd.c b/drivers/amlogic/mmc/amlsd.c
index cc2c4fc9d245..a83855a0bca2 100755
--- a/drivers/amlogic/mmc/amlsd.c
+++ b/drivers/amlogic/mmc/amlsd.c
@@ -1230,6 +1230,7 @@ int aml_check_unsupport_cmd(struct mmc_host* mmc, struct mmc_request* mrq)
 
 int aml_sd_voltage_switch (struct amlsd_platform* pdata, char signal_voltage)
 {
+#ifdef CONFIG_ARCH_MESON8
 #ifdef CONFIG_AMLOGIC_BOARD_HAS_PMU 
     int vol = LDO4DAC_REG_3_3_V;
     int delay_ms = 0;
@@ -1279,6 +1280,7 @@ int aml_sd_voltage_switch (struct amlsd_platform* pdata, char signal_voltage)
         // sdhc_err("delay %dms.\n", delay_ms); 
     }
 #endif 
+#endif
 
     return 0;
 }
-- 
2.19.0

