From 680beb1c08d5f1c17d9fb78bb1e3e5f2612a3e5c Mon Sep 17 00:00:00 2001
From: "shuai.cao" <shuai.cao@amlogic.com>
Date: Sun, 26 Jan 2014 14:44:34 +0800
Subject: [PATCH 3140/5965] PD# 86823:fix ionvideo av synchronization bug

---
 drivers/amlogic/ionvideo/ionvideo.c | 7 ++++---
 1 file changed, 4 insertions(+), 3 deletions(-)

diff --git a/drivers/amlogic/ionvideo/ionvideo.c b/drivers/amlogic/ionvideo/ionvideo.c
index f650e0873b88..9a576bb9cede 100755
--- a/drivers/amlogic/ionvideo/ionvideo.c
+++ b/drivers/amlogic/ionvideo/ionvideo.c
@@ -150,7 +150,7 @@ static int ionvideo_av_synchronization(struct ionvideo_dev *dev, struct ionvideo
     int ret = 0;
     int d = timestamp_vpts_get() - timestamp_pcrscr_get();
 
-    if (d > 1350 && d < 450000) {
+    if (d > 1350) {
         vf = vf_get(RECEIVER_NAME);
         if (vf) {
             ret = ppmgr2_process(vf, &dev->ppmgr2_dev, vb->v4l2_buf.index);
@@ -188,7 +188,8 @@ static int ionvideo_av_synchronization(struct ionvideo_dev *dev, struct ionvideo
         } else {
             ret = -EAGAIN;
         }
-    } else if (d > 450000) {
+    }
+    if (d > 450000) {
         printk("ionvideo_av_synchronization error\n");
     }
 
@@ -618,7 +619,7 @@ static int vidioc_dqbuf(struct file *file, void *priv, struct v4l2_buffer *p){
         } else {
             p = timestamp_pcrscr_get();
         }
-        if (d > 1350 && d < 450000) {
+        if (d > 1350) {
             return -EAGAIN;
         }
     }
-- 
2.19.0

