From 1569265782ef26ed77ce45ebeb0676f11d4c114a Mon Sep 17 00:00:00 2001
From: Greg Kroah-Hartman <gregkh@linuxfoundation.org>
Date: Thu, 6 Feb 2014 11:08:34 -0800
Subject: [PATCH 3253/5965] Linux 3.10.29

---
 Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index addf1b007fe3..db7691ddd67c 100644
--- a/Makefile
+++ b/Makefile
@@ -1,6 +1,6 @@
 VERSION = 3
 PATCHLEVEL = 10
-SUBLEVEL = 28
+SUBLEVEL = 29
 EXTRAVERSION =
 NAME = TOSSUG Baby Fish
 
-- 
2.19.0

