From ffaa6023afc9ee01b29593e9d1f138aec8e0d453 Mon Sep 17 00:00:00 2001
From: "sunny.luo" <sunny.luo@amlogic.com>
Date: Fri, 7 Feb 2014 17:24:45 +0800
Subject: [PATCH 3254/5965] pd#86604: m6 compile error

---
 drivers/amlogic/input/saradc/saradc.c | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/drivers/amlogic/input/saradc/saradc.c b/drivers/amlogic/input/saradc/saradc.c
index d90025fb6991..6965ef43a96b 100755
--- a/drivers/amlogic/input/saradc/saradc.c
+++ b/drivers/amlogic/input/saradc/saradc.c
@@ -48,7 +48,9 @@ static void saradc_reset(void)
 	set_clock_divider(20);
 	enable_clock();
 	enable_adc();
+#ifdef CONFIG_ARCH_MESON8
 	enable_bandgap();
+#endif
 	set_sample_mode(DIFF_MODE);
 	set_tempsen(0);
 	disable_fifo_irq();
@@ -417,6 +419,7 @@ static struct class saradc_class = {
 
 int get_cpu_temp()
 {
+#ifdef CONFIG_ARCH_MESON8
 	int ret=-1,tempa;
 	if(gp_saradc->flag){
 		ret=get_adc_sample(6);
@@ -429,6 +432,9 @@ int get_cpu_temp()
 		ret=NOT_WRITE_EFUSE;
 	}
 	return ret;
+#else
+	return NOT_WRITE_EFUSE;
+#endif
 }
 static int saradc_probe(struct platform_device *pdev)
 {
-- 
2.19.0

