From 2ae2f2578f471d3be83f2440e4963712952ad30a Mon Sep 17 00:00:00 2001
From: "tao.dong" <tao.dong@amlogic.com>
Date: Mon, 10 Feb 2014 15:37:09 +0800
Subject: [PATCH 3257/5965] PD: #85481: delete unused file

---
 debug | 1 -
 1 file changed, 1 deletion(-)
 delete mode 100644 debug

diff --git a/debug b/debug
deleted file mode 100644
index b6623585d2c6..000000000000
--- a/debug
+++ /dev/null
@@ -1 +0,0 @@
-rc10c8
-- 
2.19.0

