From 9d01164f5f85baf22322b9fdcb8173b513b80a82 Mon Sep 17 00:00:00 2001
From: "dezhi.kong" <dezhi.kong@amlogic.com>
Date: Tue, 14 Jan 2014 16:22:55 +0800
Subject: [PATCH 3261/5965] PD#86081;isp,modify ov5647 for mwb

---
 drivers/amlogic/camera/ov5647.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/amlogic/camera/ov5647.c b/drivers/amlogic/camera/ov5647.c
index ce9b4515acf7..6d7339aa951d 100755
--- a/drivers/amlogic/camera/ov5647.c
+++ b/drivers/amlogic/camera/ov5647.c
@@ -2660,8 +2660,8 @@ void OV5647_set_param_wb(struct ov5647_device *dev,enum  camera_wb_flip_e para)/
             dev->cam_para->cam_command = CAM_COMMAND_AWB;
         }else{
             dev->cam_para->cam_command = CAM_COMMAND_MWB;
-            memcpy(dev->cam_para->xml_wb_manual->reg_map,dev->configure->wb.wb[index].export,WB_MAX * sizeof(int));
         }
+	memcpy(dev->cam_para->xml_wb_manual->reg_map,dev->configure->wb.wb[index].export,WB_MAX * sizeof(int));
         printk("set wb :%d\n",index);
         dev->fe_arg.port = TVIN_PORT_ISP;
         dev->fe_arg.index = 0;
-- 
2.19.0

