From 460083134a3092e4a5dda8eb24c5e327eeaa386d Mon Sep 17 00:00:00 2001
From: "chunxue.yu" <chunxue.yu@amlogic.com>
Date: Mon, 17 Feb 2014 19:43:40 +0800
Subject: [PATCH 3354/5965] update k102 touch driver

---
 drivers/amlogic/input/touchscreen/gslx680B.c | 12 ++++++------
 1 file changed, 6 insertions(+), 6 deletions(-)

diff --git a/drivers/amlogic/input/touchscreen/gslx680B.c b/drivers/amlogic/input/touchscreen/gslx680B.c
index a4d8907830cc..ad0ba0f818dc 100755
--- a/drivers/amlogic/input/touchscreen/gslx680B.c
+++ b/drivers/amlogic/input/touchscreen/gslx680B.c
@@ -731,15 +731,15 @@ static void report_data(struct gsl_ts *ts, u16 x, u16 y, u8 pressure, u8 id)
 	
 #ifdef REPORT_DATA_ANDROID_4_0
 	input_mt_slot(ts->input, id);		
-	// int x1 = (SCREEN_MAX_X-x);
-	// if (x1 < 0) x1 =0;
-	int y1 = (SCREEN_MAX_Y-y);
-	if (y1 < 0) y1 =0;
+	 int x1 = (SCREEN_MAX_X-x);
+	 if (x1 < 0) x1 =0;
+	// int y1 = (SCREEN_MAX_Y-y);
+	// if (y1 < 0) y1 =0;
 	// d
 	input_report_abs(ts->input, ABS_MT_TRACKING_ID, id);
 	input_report_abs(ts->input, ABS_MT_TOUCH_MAJOR, pressure);
-	input_report_abs(ts->input, ABS_MT_POSITION_X, y1);
-	input_report_abs(ts->input, ABS_MT_POSITION_Y, x);	
+	input_report_abs(ts->input, ABS_MT_POSITION_X, y);
+	input_report_abs(ts->input, ABS_MT_POSITION_Y, x1);	
 	input_report_abs(ts->input, ABS_MT_WIDTH_MAJOR, 1);
 #else
 	//input_report_key(ts->input, BTN_TOUCH, 1);
-- 
2.19.0

