From 62952f6ea4ddff82641bf46e1045e0281afcc2db Mon Sep 17 00:00:00 2001
From: "shuai.cao" <shuai.cao@amlogic.com>
Date: Thu, 13 Feb 2014 11:26:16 +0800
Subject: [PATCH 3357/5965] PD# 87019:fix ge2d rotation bug

---
 drivers/amlogic/amports/video.c | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/drivers/amlogic/amports/video.c b/drivers/amlogic/amports/video.c
index e8f24c156a35..261521abb837 100755
--- a/drivers/amlogic/amports/video.c
+++ b/drivers/amlogic/amports/video.c
@@ -1067,7 +1067,8 @@ static void vsync_toggle_frame(vframe_t *vf)
         (cur_dispbuf->height != vf->height) ||
         (cur_dispbuf->ratio_control != vf->ratio_control) ||
         ((cur_dispbuf->type_backup & VIDTYPE_INTERLACE) !=
-         (vf->type_backup & VIDTYPE_INTERLACE))) {
+         (vf->type_backup & VIDTYPE_INTERLACE)) ||
+         (cur_dispbuf->type != vf->type)) {
         amlog_mask(LOG_MASK_FRAMEINFO,
                    "%s %dx%d ar=0x%x\n",
                    ((vf->type & VIDTYPE_TYPEMASK) == VIDTYPE_INTERLACE_TOP) ?
@@ -2190,7 +2191,7 @@ static irqreturn_t vsync_isr(int irq, void *dev_id)
             field_matching_count = 0;
             // adjust system time to get one more field toggle
             // at next vsync to match field
-            timestamp_pcrscr_inc(vsync_pts_inc); 
+            timestamp_pcrscr_inc(vsync_pts_inc);
         }
     } else {
         field_matching_count = 0;
-- 
2.19.0

