From 85d1f9c5160c3bd734849eff9817d45a58cc1e8b Mon Sep 17 00:00:00 2001
From: FFT <zh.cao@amlogic.com>
Date: Tue, 18 Feb 2014 11:38:36 +0800
Subject: [PATCH 3364/5965] correct last_angle_orientation

---
 drivers/amlogic/amports/video.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/amlogic/amports/video.c b/drivers/amlogic/amports/video.c
index 261521abb837..a8819d4dd622 100755
--- a/drivers/amlogic/amports/video.c
+++ b/drivers/amlogic/amports/video.c
@@ -1904,7 +1904,7 @@ static irqreturn_t vsync_isr(int irq, void *dev_id)
             video_prot_set_angle(&video_prot, angle_orientation);
         }
         u32 angle_orientation = (video_angle + video_prot.src_vframe_orientation) % 4;
-        u32 last_angle_orientation = (video_prot.angle + video_prot.src_vframe_orientation) % 4;
+        u32 last_angle_orientation = video_prot.angle;
         if (video_prot.enable_layer || get_vpu_mem_pd_vmod(VPU_PIC_ROT2) == VPU_MEM_POWER_DOWN) {
             if (angle_orientation % 2) {
                 if (video_prot.angle_changed & 0x1) {
-- 
2.19.0

