From 0d8613eca9c96972cab59a2041a5104805e303a9 Mon Sep 17 00:00:00 2001
From: "shuai.cao" <shuai.cao@amlogic.com>
Date: Tue, 11 Feb 2014 20:01:41 +0800
Subject: [PATCH 3381/5965] PD# 86915: fix ionvideo orientation bug

---
 drivers/amlogic/ionvideo/ppmgr2.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/drivers/amlogic/ionvideo/ppmgr2.c b/drivers/amlogic/ionvideo/ppmgr2.c
index e57387843d04..bb4617b17ef5 100755
--- a/drivers/amlogic/ionvideo/ppmgr2.c
+++ b/drivers/amlogic/ionvideo/ppmgr2.c
@@ -317,6 +317,7 @@ int ppmgr2_process(struct vframe_s* vf, struct ppmgr2_device *ppd, int index) {
     int dst_pixel_format = ppd->ge2d_fmt;
     ge2d_context_t *context = ppd->context;
     config_para_ex_t* ge2d_config = &(ppd->ge2d_config);
+    int angle = (ppd->angle + src_vf->orientation) % 4;
 
     src_position[0] = 0;
     src_position[1] = 0;
@@ -334,7 +335,7 @@ int ppmgr2_process(struct vframe_s* vf, struct ppmgr2_device *ppd, int index) {
     ge2d_src_config(src_vf, ge2d_config);
 
     ge2d_mirror_config(ppd->mirror, ge2d_config);
-    ge2d_angle_config(ppd->angle, ge2d_config);
+    ge2d_angle_config(angle, ge2d_config);
     paint_mode_convert(ppd->paint_mode, src_position, dst_paint_position, dst_plane_position);
 
     if(src_vf->type & VIDTYPE_INTERLACE) {
-- 
2.19.0

