From 88bcb422f3bc71395c1cace417ac5f9f3f5d7ae7 Mon Sep 17 00:00:00 2001
From: "qiufang.dai" <qiufang.dai@amlogic.com>
Date: Thu, 20 Feb 2014 21:42:58 +0800
Subject: [PATCH 3387/5965] secure-os: support m8 suspend/resume

---
 arch/arm/mach-meson8/pm.c | 8 +++++++-
 1 file changed, 7 insertions(+), 1 deletion(-)

diff --git a/arch/arm/mach-meson8/pm.c b/arch/arm/mach-meson8/pm.c
index 5f5b7d7abe5b..6a8b926e704f 100755
--- a/arch/arm/mach-meson8/pm.c
+++ b/arch/arm/mach-meson8/pm.c
@@ -30,7 +30,9 @@
 #include <mach/clock.h>
 #include <plat/regops.h>
 #include <plat/io.h>
-
+#ifdef CONFIG_MESON_TRUSTZONE
+#include <mach/meson-secure.h>
+#endif
 
 #ifdef CONFIG_SUSPEND_WATCHDOG
 #include <mach/watchdog.h>
@@ -267,7 +269,11 @@ static void meson_pm_suspend(void)
 		WRITE_AOBUS_REG(AO_RTI_STATUS_REG2, 0x1234abcd);
 	}else{
 #ifdef CONFIG_MESON_SUSPEND
+#ifdef CONFIG_MESON_TRUSTZONE
+		meson_suspend_firmware();
+#else
 		meson_power_suspend();
+#endif
 #else
 #if 0
 		//k101 power key
-- 
2.19.0

