From ce32ef7b497353bfe1140821dbf0242e5fa0c79e Mon Sep 17 00:00:00 2001
From: "yun.cai" <yun.cai@amlogic.com>
Date: Wed, 25 Dec 2013 17:22:36 +0800
Subject: [PATCH 3428/5965] m6tv: new reserve memory way include M6TV

---
 drivers/of/fdt.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/of/fdt.c b/drivers/of/fdt.c
index 3fc23c8a92a4..270d99944f1c 100755
--- a/drivers/of/fdt.c
+++ b/drivers/of/fdt.c
@@ -861,7 +861,7 @@ int __init early_init_dt_scan_memory(unsigned long node, const char *uname,
 
 	set_memory_start_addr(phys_offset);
 	set_memory_total_size(total);
-#if MESON_CPU_TYPE >= MESON_CPU_TYPE_MESON8
+#if MESON_CPU_TYPE >= MESON_CPU_TYPE_MESON6TV
 	early_init_dt_add_memory_arch(aml_reserved_end+1,phys_offset+total-aml_reserved_end-1-high_reserve_size);
 #else
 	if(aml_reserved_end+1 > FIRMWARE_ADDR)
-- 
2.19.0

