From 6dbd1846a813523d1807a8257a134cefb0fc8f72 Mon Sep 17 00:00:00 2001
From: "qiufang.dai" <qiufang.dai@amlogic.com>
Date: Fri, 21 Feb 2014 18:33:21 +0800
Subject: [PATCH 3449/5965] fix m6tvd compile error

---
 drivers/amlogic/display/vout/tvoutc.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/drivers/amlogic/display/vout/tvoutc.c b/drivers/amlogic/display/vout/tvoutc.c
index c6e1ee44d0c4..f588fafa8e49 100755
--- a/drivers/amlogic/display/vout/tvoutc.c
+++ b/drivers/amlogic/display/vout/tvoutc.c
@@ -372,6 +372,7 @@ printk(" clk_util_clk_msr 27 = %d\n", clk_util_clk_msr(27));
 printk(" clk_util_clk_msr 29 = %d\n", clk_util_clk_msr(29));
 #endif
 
+#ifdef CONFIG_ARCH_MESON8
 	if( (mode==TVMODE_480CVBS) || (mode==TVMODE_576CVBS) )
 	{
 		msleep(1000);
@@ -379,6 +380,7 @@ printk(" clk_util_clk_msr 29 = %d\n", clk_util_clk_msr(29));
 		aml_write_reg32(P_HHI_VDAC_CNTL0,0x650001);
 		aml_write_reg32(P_HHI_VDAC_CNTL1,0x1);
 	}
+#endif
 //while(1);
 
 
-- 
2.19.0

