From 9d2512e09aaff00f9e8381240f7db3e2df39e133 Mon Sep 17 00:00:00 2001
From: "zhancai.yang" <zhancai.yang@amlogic.com>
Date: Tue, 25 Feb 2014 14:33:20 +0800
Subject: [PATCH 3531/5965] add sdhc controller keep on support

---
 drivers/amlogic/mmc/aml_sdhc_m8.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/drivers/amlogic/mmc/aml_sdhc_m8.c b/drivers/amlogic/mmc/aml_sdhc_m8.c
index 82cdc17c2868..b521f8a325b9 100755
--- a/drivers/amlogic/mmc/aml_sdhc_m8.c
+++ b/drivers/amlogic/mmc/aml_sdhc_m8.c
@@ -1626,7 +1626,8 @@ static int aml_sdhc_probe(struct platform_device *pdev)
         if (pdata->caps & MMC_CAP_NONREMOVABLE) {
             pdata->is_in = true;
         }
-
+        if (pdata->caps & MMC_PM_KEEP_POWER)
+            mmc->pm_caps |= MMC_PM_KEEP_POWER; 
         pdata->host = host;
         pdata->mmc = mmc;
         pdata->is_fir_init = true;
-- 
2.19.0

