From b4bfa962f848ec7bdc22ef97457d262089302c0a Mon Sep 17 00:00:00 2001
From: "chunxue.yu" <chunxue.yu@amlogic.com>
Date: Thu, 20 Feb 2014 14:23:02 +0800
Subject: [PATCH 3534/5965] pd #87258: fix don't support 5 points touch for g33

---
 drivers/amlogic/input/touchscreen/focaltech_5x06.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/drivers/amlogic/input/touchscreen/focaltech_5x06.c b/drivers/amlogic/input/touchscreen/focaltech_5x06.c
index 1838ead4a7e9..d424761fd6f6 100755
--- a/drivers/amlogic/input/touchscreen/focaltech_5x06.c
+++ b/drivers/amlogic/input/touchscreen/focaltech_5x06.c
@@ -1312,7 +1312,8 @@ ft5x0x_ts_probe(struct i2c_client *client, const struct i2c_device_id *id)
 			     ABS_MT_TOUCH_MAJOR, 0, PRESS_MAX, 0, 0);
 	input_set_abs_params(input_dev,
 			     ABS_MT_WIDTH_MAJOR, 0, 200, 0, 0);
-
+	input_set_abs_params(input_dev, 
+			     ABS_MT_TRACKING_ID, 0, FT5X0X_EVENT_MAX, 0, 0);
 	set_bit(EV_ABS, input_dev->evbit);
 	set_bit(EV_KEY, input_dev->evbit);
 
-- 
2.19.0

