From f4c5b0b6aa6d4e105c2eda522aaa4c76f98f89d8 Mon Sep 17 00:00:00 2001
From: "qi.duan" <qi.duan@amlogic.com>
Date: Tue, 25 Feb 2014 18:00:48 +0800
Subject: [PATCH 3540/5965] PD#87324:reset all ep driver_data before gadget
 bind

---
 drivers/usb/gadget/composite.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/usb/gadget/composite.c b/drivers/usb/gadget/composite.c
index 7d27bc1f4664..81cc3fd2db74 100755
--- a/drivers/usb/gadget/composite.c
+++ b/drivers/usb/gadget/composite.c
@@ -758,6 +758,7 @@ int usb_add_config(struct usb_composite_dev *cdev,
 	if (status)
 		goto done;
 
+	usb_ep_autoconfig_reset(cdev->gadget);
 	status = bind(config);
 	if (status < 0) {
 		while (!list_empty(&config->functions)) {
-- 
2.19.0

