From 58f7b20f6e41c41c74d9d3518582537ae8a0fc09 Mon Sep 17 00:00:00 2001
From: timyao <tim.yao@amlogic.com>
Date: Wed, 26 Feb 2014 00:53:23 -0800
Subject: [PATCH 3550/5965] PD #87734. Fix H.264 prog/interlaced type.

---
 drivers/amlogic/amports/vh264.c | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/drivers/amlogic/amports/vh264.c b/drivers/amlogic/amports/vh264.c
index 30b72e0f77a9..e39bb96aa045 100755
--- a/drivers/amlogic/amports/vh264.c
+++ b/drivers/amlogic/amports/vh264.c
@@ -1021,6 +1021,12 @@ static void vh264_isr(void)
                 force_interlaced_frame = true;
             }
 
+            if (pic_struct_present) {
+                if ((pic_struct == PIC_TOP_BOT) || (pic_struct == PIC_BOT_TOP)) {
+                    prog_frame = 0;
+                }
+            }
+
             if ((!force_interlaced_frame) && (frame_mb_only || prog_frame || (pic_struct_present && pic_struct <= PIC_TRIPLE_FRAME))) {
                 if (pic_struct_present) {
                     if (pic_struct == PIC_TOP_BOT_TOP || pic_struct == PIC_BOT_TOP_BOT) {
-- 
2.19.0

