From 12350a286b837f5b7b52a7c91ccbd2b22a7e3480 Mon Sep 17 00:00:00 2001
From: "shuai.cao" <shuai.cao@amlogic.com>
Date: Wed, 26 Feb 2014 19:10:01 +0800
Subject: [PATCH 3557/5965] PD# 86871:add ionvideo size changed response

---
 drivers/amlogic/ionvideo/ionvideo.c | 27 ++++++++++++++++-----------
 1 file changed, 16 insertions(+), 11 deletions(-)

diff --git a/drivers/amlogic/ionvideo/ionvideo.c b/drivers/amlogic/ionvideo/ionvideo.c
index affd80b80fef..ce82bf27f2a6 100755
--- a/drivers/amlogic/ionvideo/ionvideo.c
+++ b/drivers/amlogic/ionvideo/ionvideo.c
@@ -191,7 +191,7 @@ static void ionvideo_thread_tick(struct ionvideo_dev *dev) {
         msleep(5);
         return;
     }
-    if (ionvideo_size_changed(dev, vf)) {
+    if (freerun_mode == 0 && ionvideo_size_changed(dev, vf)) {
         msleep(10);
         return;
     }
@@ -477,11 +477,22 @@ static int vidioc_g_fmt_vid_cap(struct file *file, void *priv, struct v4l2_forma
     int ret = 0;
     unsigned long flags;
 
-    if (dev->c_width == 0 || dev->c_height == 0) {
-        return -EINVAL;
+    if (freerun_mode == 0) {
+        if (dev->c_width == 0 || dev->c_height == 0) {
+            return -EINVAL;
+        }
+        f->fmt.pix.width = dev->c_width;
+        f->fmt.pix.height = dev->c_height;
+        spin_lock_irqsave(&q->done_lock, flags);
+        ret = list_empty(&q->done_list);
+        spin_unlock_irqrestore(&q->done_lock, flags);
+        if (!ret) {
+            return -EAGAIN;
+        }
+    } else {
+        f->fmt.pix.width = dev->width;
+        f->fmt.pix.height = dev->height;
     }
-    f->fmt.pix.width = dev->c_width;
-    f->fmt.pix.height = dev->c_height;
     f->fmt.pix.field = V4L2_FIELD_INTERLACED;
     f->fmt.pix.pixelformat = dev->fmt->fourcc;
     f->fmt.pix.bytesperline = (f->fmt.pix.width * dev->fmt->depth) >> 3;
@@ -491,12 +502,6 @@ static int vidioc_g_fmt_vid_cap(struct file *file, void *priv, struct v4l2_forma
     else
         f->fmt.pix.colorspace = V4L2_COLORSPACE_SRGB;
 
-    spin_lock_irqsave(&q->done_lock, flags);
-    ret = list_empty(&q->done_list);
-    spin_unlock_irqrestore(&q->done_lock, flags);
-    if (!ret) {
-        return -EAGAIN;
-    }
     return 0;
 }
 
-- 
2.19.0

