From 0737417ce7fe711a4e310b672c1f08e6b8741ca3 Mon Sep 17 00:00:00 2001
From: "shuai.cao" <shuai.cao@amlogic.com>
Date: Thu, 27 Feb 2014 16:51:09 +0800
Subject: [PATCH 3561/5965] PD# 86871:add ionvideo size changed response

---
 drivers/amlogic/ionvideo/ionvideo.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/amlogic/ionvideo/ionvideo.c b/drivers/amlogic/ionvideo/ionvideo.c
index ce82bf27f2a6..2bd630310240 100755
--- a/drivers/amlogic/ionvideo/ionvideo.c
+++ b/drivers/amlogic/ionvideo/ionvideo.c
@@ -167,7 +167,7 @@ static int ionvideo_size_changed(struct ionvideo_dev *dev, struct vframe_s* vf)
     int aw = vf->width;
     int ah = vf->height;
 
-    v4l_bound_align_image(&aw, 48, MAX_WIDTH, 4, &ah, 32, MAX_HEIGHT, 0, 0);
+    v4l_bound_align_image(&aw, 48, MAX_WIDTH, 5, &ah, 32, MAX_HEIGHT, 0, 0);
     dev->c_width = aw;
     dev->c_height = ah;
     if (aw != dev->width || ah != dev->height) {
-- 
2.19.0

