From 9d2f9ace8c7dbd47e3308cee31b6cb050ce58ddb Mon Sep 17 00:00:00 2001
From: "wenbiao.zhang" <wenbiao.zhang@amlogic.com>
Date: Thu, 27 Feb 2014 21:39:23 +0800
Subject: [PATCH 3562/5965] PD #87610: not clear VPP_VD1_POSTBLEND for M6

---
 drivers/amlogic/amports/video.c | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/drivers/amlogic/amports/video.c b/drivers/amlogic/amports/video.c
index 187422751bf2..5953aef2fa2c 100755
--- a/drivers/amlogic/amports/video.c
+++ b/drivers/amlogic/amports/video.c
@@ -2296,8 +2296,11 @@ exit:
                 printk("VsyncEnableVideoLayer\n");
             }
         } else if (video_onoff_state == VIDEO_ENABLE_STATE_OFF_REQ) {
+        #if MESON_CPU_TYPE >= MESON_CPU_TYPE_MESON8
             CLEAR_VCBUS_REG_MASK(VPP_MISC + cur_dev->vpp_off, VPP_VD1_PREBLEND|VPP_VD2_PREBLEND|VPP_VD2_POSTBLEND|VPP_VD1_POSTBLEND);
-
+        #else
+            CLEAR_VCBUS_REG_MASK(VPP_MISC + cur_dev->vpp_off, VPP_VD1_PREBLEND|VPP_VD2_PREBLEND|VPP_VD2_POSTBLEND);
+        #endif
             video_onoff_state = VIDEO_ENABLE_STATE_IDLE;
 
             if(debug_flag& DEBUG_FLAG_BLACKOUT){
-- 
2.19.0

