From 1712e25bc1f3293e53721e9a4a593b0d525ac957 Mon Sep 17 00:00:00 2001
From: "shuai.cao" <shuai.cao@amlogic.com>
Date: Fri, 28 Feb 2014 10:19:17 +0800
Subject: [PATCH 3564/5965] fix 87004 commit cause M6 compile error

---
 drivers/amlogic/amports/video.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/amlogic/amports/video.c b/drivers/amlogic/amports/video.c
index 5953aef2fa2c..89f719e009cf 100755
--- a/drivers/amlogic/amports/video.c
+++ b/drivers/amlogic/amports/video.c
@@ -195,6 +195,7 @@ static int video_onoff_state = VIDEO_ENABLE_STATE_IDLE;
 #define PROT_MEM_POWER_OFF() \
     do { \
         unsigned long flags; \
+        video_prot_gate_off(); \
         spin_lock_irqsave(&delay_work_lock, flags); \
         vpu_delay_work_flag |= VPU_DELAYWORK_MEM_POWER_OFF_PROT; \
         vpu_mem_power_off_count = VPU_MEM_POWEROFF_DELAY; \
@@ -263,7 +264,6 @@ static int video_onoff_state = VIDEO_ENABLE_STATE_IDLE;
     do { \
          CLEAR_VCBUS_REG_MASK(VPP_MISC + cur_dev->vpp_off, \
            VPP_VD1_PREBLEND|VPP_VD2_PREBLEND|VPP_VD2_POSTBLEND|VPP_VD1_POSTBLEND ); \
-         video_prot_gate_off(); \
          VIDEO_LAYER_OFF(); \
          VD1_MEM_POWER_OFF(); \
          PROT_MEM_POWER_OFF(); \
-- 
2.19.0

