From ebbf748b55d3daf0043a293f0c2f250c533aef3e Mon Sep 17 00:00:00 2001
From: "spike.liao" <spike.liao@amlogic.com>
Date: Fri, 21 Feb 2014 15:18:06 +0800
Subject: [PATCH 3566/5965] PD# 87285 fix ionvideo first pts compute wrong

---
 drivers/amlogic/ionvideo/ionvideo.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/amlogic/ionvideo/ionvideo.c b/drivers/amlogic/ionvideo/ionvideo.c
index 2bd630310240..9956113ac51d 100755
--- a/drivers/amlogic/ionvideo/ionvideo.c
+++ b/drivers/amlogic/ionvideo/ionvideo.c
@@ -120,7 +120,7 @@ static void videoc_omx_compute_pts(struct ionvideo_dev *dev, struct vframe_s* vf
         dev->receiver_register = 0;
         dev->pts = vf->pts_us64;
     } else if (dev->receiver_register){
-        timestamp_vpts_set(timestamp_pcrscr_get());
+        timestamp_vpts_set(0);
         dev->receiver_register = 0;
         dev->pts = timestamp_vpts_get();
     } else {
-- 
2.19.0

