From b0277f6bfd7ff36a4fca1f463b74b6c96236c241 Mon Sep 17 00:00:00 2001
From: "zhenfei.li" <zhenfei.li@amlogic>
Date: Thu, 27 Feb 2014 15:53:36 +0800
Subject: [PATCH 3567/5965] PD #87290: fixed a m6 ge2d block problem.

---
 drivers/amlogic/display/ge2d/ge2d_hw.c | 27 ++++++++++++++++++++++++++
 1 file changed, 27 insertions(+)

diff --git a/drivers/amlogic/display/ge2d/ge2d_hw.c b/drivers/amlogic/display/ge2d/ge2d_hw.c
index 4cc25c2f707c..f42ddea6393f 100755
--- a/drivers/amlogic/display/ge2d/ge2d_hw.c
+++ b/drivers/amlogic/display/ge2d/ge2d_hw.c
@@ -605,6 +605,31 @@ void ge2d_set_cmd (ge2d_cmd_t *cfg)
     x_yc_ratio = READ_MPEG_REG_BITS(GE2D_GEN_CTRL0, 11, 1);
     y_yc_ratio = READ_MPEG_REG_BITS(GE2D_GEN_CTRL0, 10, 1);
 
+#if MESON_CPU_TYPE == MESON_CPU_TYPE_MESON6	
+    if (x_yc_ratio) {
+        if (cfg->src1_x_rev) {
+            x_extra_bit_start = 0;
+            x_extra_bit_end   = 3;
+            x_chr_phase = 0x80;
+        } else {
+            x_extra_bit_start = 3;
+            x_extra_bit_end   = 0;
+            x_chr_phase = 0x08;
+        }
+    }
+    
+    if (y_yc_ratio) {
+        if (cfg->src1_y_rev) {
+            y_extra_bit_start = 2;
+            y_extra_bit_end   = 3;
+            y_chr_phase = 0xc4;
+        } else {
+            y_extra_bit_start = 3;
+            y_extra_bit_end   = 2;
+            y_chr_phase = 0x4c;
+        }
+    }
+#else
     if (x_yc_ratio) {
         if( (cfg->src1_x_rev+cfg->dst_x_rev) == 1) {
             x_extra_bit_start = 3;
@@ -629,6 +654,8 @@ void ge2d_set_cmd (ge2d_cmd_t *cfg)
         }
     }
 
+#endif
+
     WRITE_MPEG_REG(GE2D_SRC1_X_START_END, 
                          (x_extra_bit_start << 30) |  //x start extra
                          ((cfg->src1_x_start & 0x3fff) << 16) | // Limit the range in case of the sign bit extending to the top
-- 
2.19.0

