From 86489eca3a356bb5574fbef7262d203ba0f7b923 Mon Sep 17 00:00:00 2001
From: xu xing <xing.xu@amlogic.com>
Date: Thu, 6 Mar 2014 16:38:51 +0800
Subject: [PATCH 3597/5965] PD#87250:clean warning for M6 kernel

---
 drivers/amlogic/audiodsp/dsp_mailbox.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/amlogic/audiodsp/dsp_mailbox.c b/drivers/amlogic/audiodsp/dsp_mailbox.c
index 7beaeeec9aa3..a8bff072cddd 100755
--- a/drivers/amlogic/audiodsp/dsp_mailbox.c
+++ b/drivers/amlogic/audiodsp/dsp_mailbox.c
@@ -107,7 +107,7 @@ int get_mailbox_data(struct audiodsp_priv *priv,int num,struct mail_msg *msg)
 static irqreturn_t audiodsp_mailbox_irq(int irq, void *data)
 {
 	struct audiodsp_priv *priv=(struct audiodsp_priv *)data;
-	unsigned long status/*,fiq_mask*/;
+	unsigned long status,fiq_mask;
 	struct mail_msg msg;
 	int i = 0;
 	status=READ_VREG(MB1_REG);
-- 
2.19.0

