From ead03fdbaef03cd6e487c98472f13c31cb328635 Mon Sep 17 00:00:00 2001
From: "ke.gong" <ke.gong@amlogic.com>
Date: Fri, 14 Mar 2014 08:43:02 +0800
Subject: [PATCH 3729/5965] PD #88622: Add CPU type check in demux CSI setting.

---
 drivers/amlogic/dvb_tv/aml_dvb.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/drivers/amlogic/dvb_tv/aml_dvb.c b/drivers/amlogic/dvb_tv/aml_dvb.c
index f5b9b2f52dfa..8b10ce949630 100755
--- a/drivers/amlogic/dvb_tv/aml_dvb.c
+++ b/drivers/amlogic/dvb_tv/aml_dvb.c
@@ -92,6 +92,7 @@ static struct tsdemux_ops aml_tsdemux_ops = {
 
 static int control_ts_on_csi_port(int tsin, int enable)
 {
+#if MESON_CPU_TYPE >= MESON_CPU_TYPE_MESON8
 	unsigned int temp_data;
 	if(tsin==2 && enable) {
 		//TS2 is on CSI port.
@@ -104,6 +105,7 @@ static int control_ts_on_csi_port(int tsin, int enable)
 		temp_data |= 0x80000fc0;
 		WRITE_CBUS_REG(HHI_CSI_PHY_CNTL2,temp_data);
 	}
+#endif
 	return 0;
 }
 
-- 
2.19.0

