From 933184c8299a92d6b31d056fc432cd10b1f9a1d1 Mon Sep 17 00:00:00 2001
From: "hongwu.su" <hongwu.su@amlogic.com>
Date: Tue, 18 Mar 2014 19:38:58 +0800
Subject: [PATCH 3762/5965] pd#88605 : add wipe ioctl for emmc platform in
 android 4.4

---
 drivers/mmc/card/block.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/mmc/card/block.c b/drivers/mmc/card/block.c
index 8c1a4fe7430c..4d480488fe0e 100755
--- a/drivers/mmc/card/block.c
+++ b/drivers/mmc/card/block.c
@@ -602,6 +602,7 @@ static int mmc_blk_ioctl(struct block_device *bdev, fmode_t mode,
        case MEMLOCK:
        case MEMUNLOCK:
        case MEMGETBADBLOCK:
+	   case BLKWIPEPART:
            return 0;
        default:
            ret = -EINVAL;
-- 
2.19.0

