From 2436bb6d98539e555cbf7494e587a3dd92687e4a Mon Sep 17 00:00:00 2001
From: "hongwu.su" <hongwu.su@amlogic.com>
Date: Wed, 19 Mar 2014 19:10:51 +0800
Subject: [PATCH 3767/5965] PD#88605: fix wipe part problem for m6

---
 drivers/mtd/mtd_blkdevs.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/drivers/mtd/mtd_blkdevs.c b/drivers/mtd/mtd_blkdevs.c
index 025ba24e915d..a63d5d92d14f 100755
--- a/drivers/mtd/mtd_blkdevs.c
+++ b/drivers/mtd/mtd_blkdevs.c
@@ -300,6 +300,10 @@ static int blktrans_ioctl(struct block_device *bdev, fmode_t mode,
 	case BLKFLSBUF:
 		ret = dev->tr->flush ? dev->tr->flush(dev) : 0;
 		break;
+		//add this for m6 old nand driver
+	case BLKWIPEPART:
+	ret =0;	
+        break;
 	default:
 		ret = -ENOTTY;
 	}
-- 
2.19.0

