From e9499c45072eafa318a13dd42ec218f9e459fcd5 Mon Sep 17 00:00:00 2001
From: "sheng.luo" <sheng.luo@amlogic.com>
Date: Wed, 19 Mar 2014 10:34:06 +0800
Subject: [PATCH 3774/5965] PD#88838:fix 3.10 kernel capacity error in the M6
 plat

---
 drivers/mtd/mtd_blkdevs.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/mtd/mtd_blkdevs.c b/drivers/mtd/mtd_blkdevs.c
index a63d5d92d14f..284c73d73200 100755
--- a/drivers/mtd/mtd_blkdevs.c
+++ b/drivers/mtd/mtd_blkdevs.c
@@ -409,7 +409,7 @@ int add_mtd_blktrans_dev(struct mtd_blktrans_dev *new)
 		snprintf(gd->disk_name, sizeof(gd->disk_name),
 			 "%s%d", tr->name, new->devnum);
 
-	set_capacity(gd, (new->size * tr->blksize) >> 9);
+	set_capacity(gd, new->size *(tr->blksize >> 9));
 
 	/* Create the request queue */
 	spin_lock_init(&new->queue_lock);
-- 
2.19.0

