From ae442e2028c94e0192ca7603f73c4c63fd502794 Mon Sep 17 00:00:00 2001
From: "Matthew.Shyu" <matthew.shyu@amlogic.com>
Date: Fri, 21 Mar 2014 17:42:44 +0800
Subject: [PATCH 3786/5965] correct wifi driver path

---
 drivers/amlogic/Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/amlogic/Makefile b/drivers/amlogic/Makefile
index 839e0333bffd..5cc9e6527c73 100755
--- a/drivers/amlogic/Makefile
+++ b/drivers/amlogic/Makefile
@@ -93,7 +93,7 @@ obj-$(CONFIG_AM_DVB) += dvb_tv/
 obj-$(CONFIG_AM_SMARTCARD) += smartcard/
 
 obj-$(CONFIG_AM_WIFI) += wifi/
-ifneq ($(wildcard $(srctree)/../hardware/amlogic/wifi),)
+ifneq ($(wildcard $(srctree)/../hardware/wifi),)
 	obj-$(CONFIG_SDIO_DHD_CDC_WIFI_40181_MODULE) += ../../../hardware/wifi/broadcom/drivers/ap6xxx/broadcm_40181/
 	obj-$(CONFIG_RTL8188EU) += ../../../hardware/wifi/realtek/drivers/8188eu/rtl8xxx_EU/
 	obj-$(CONFIG_RTL8188EU_MP) += ../../../hardware/wifi/realtek/drivers/8188eu/rtl8xxx_EU_MP/
-- 
2.19.0

