From 9240e2637d43677c4dcfb090be47cbe0a96ad86e Mon Sep 17 00:00:00 2001
From: Victor Wan <victor.wan@amlogic.com>
Date: Mon, 24 Mar 2014 12:28:17 +0800
Subject: [PATCH 3795/5965] meson: plat: fix compile

---
 arch/arm/plat-meson/Kconfig | 5 +----
 1 file changed, 1 insertion(+), 4 deletions(-)

diff --git a/arch/arm/plat-meson/Kconfig b/arch/arm/plat-meson/Kconfig
index 99fe9e4a22c1..06b7b9330e39 100755
--- a/arch/arm/plat-meson/Kconfig
+++ b/arch/arm/plat-meson/Kconfig
@@ -46,7 +46,6 @@ source "arch/arm/mach-meson8b/Kconfig.boards"
 endmenu # "Meson reference boards"
 
 choice 
->>>>>>> 3c525bb... meson8b: compile ok
 	prompt "Interrupt Controller Select"
 	default MESON_ARM_GIC
 config MESON_IRQ
@@ -118,13 +117,13 @@ config MESON_CUSTOM_BOARD_SUPPORT
 	help
 	  This enables board definition from external customer directory
 
-<<<<<<< HEAD
 menu "Meson development boards"
 
 source "arch/arm/mach-meson6/Kconfig.boards"
 source "arch/arm/mach-meson6tv/Kconfig.boards"
 source "arch/arm/mach-meson6tvd/Kconfig.boards"
 source "arch/arm/mach-meson8/Kconfig.boards"
+source "arch/arm/mach-meson8b/Kconfig.boards"
 
 #if MESON_CUSTOM_BOARD_SUPPORT
 #isource "customer/meson/dt/Kconfig"
@@ -132,8 +131,6 @@ source "arch/arm/mach-meson8/Kconfig.boards"
 
 endmenu # "Meson reference boards"
 
-=======
->>>>>>> 3c525bb... meson8b: compile ok
 menu "Meson customer board drivers"
 depends on MESON_CUSTOM_BOARD_SUPPORT
 #isource "customer/drivers/Kconfig"
-- 
2.19.0

