From 5e764b0ae699c2328d2af23ec69c328e65f59ee4 Mon Sep 17 00:00:00 2001
From: "tao.zeng" <tao.zeng@amlogic.com>
Date: Wed, 26 Mar 2014 10:15:54 +0800
Subject: [PATCH 3816/5965] PD #81802: update new thermal parameters for M8
 project

---
 include/linux/cpufreq.h | 31 -------------------------------
 1 file changed, 31 deletions(-)

diff --git a/include/linux/cpufreq.h b/include/linux/cpufreq.h
index 11da55c0d93e..0e2805da4070 100755
--- a/include/linux/cpufreq.h
+++ b/include/linux/cpufreq.h
@@ -442,39 +442,8 @@ void cpufreq_frequency_table_put_attr(unsigned int cpu);
 #ifdef CONFIG_CPU_FREQ_GOV_HOTPLUG
 void cpufreq_set_max_cpu_num(unsigned int cpu_num);
 #else
-unsigned int max_cpu_num=NR_CPUS;
-unsigned int last_max_cpu_num=max_cpu_num;
-static inline void cpu_up_num(unsigned int num)
-{
-	for(i = 0; i < num; i++){
-		if(cpu_online(i))
-			continue;
-		cpu_up(i);
-		cpumask_set_cpu(i, tsk_cpus_allowed(NULL_task));
-	}
-}
-
 static inline void cpufreq_set_max_cpu_num(unsigned int cpu_num)
 {
-	int i=0;
-	if(cpu_num>=NR_CPUS)
-	{
-		cpu_up_num(NR_CPUS);
-	}else
-	{
-		if(cpu_num>last_max_cpu_num){
-			cpu_up_num(cpu_num);
-		}else{
-			for(i=NR_CPUS-1;i>=cpu_num;i--)
-			{
-				if(!cpu_active(i))
-					continue;
-				cpu_down(i);
-			}
-		}
-		
-	}
-	
 }
 #endif
 #endif /* _LINUX_CPUFREQ_H */
-- 
2.19.0

