From aeedc457c8056587d509e7a618feb333ee5f3d4b Mon Sep 17 00:00:00 2001
From: "zhenfei.li" <zhenfei.li@amlogic>
Date: Tue, 1 Apr 2014 15:14:54 +0800
Subject: [PATCH 3852/5965] PD #89818: Reset OV5647 before close it.

---
 drivers/amlogic/camera/ov5647.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/drivers/amlogic/camera/ov5647.c b/drivers/amlogic/camera/ov5647.c
index 2f012c137869..ab0e328e15a0 100755
--- a/drivers/amlogic/camera/ov5647.c
+++ b/drivers/amlogic/camera/ov5647.c
@@ -2478,7 +2478,8 @@ static CLASS_ATTR(camera_debug, 0664, i2c_debug_show, i2c_debug_store);
 static void power_down_ov5647(struct ov5647_device *dev)
 {
 	struct i2c_client *client = v4l2_get_subdevdata(&dev->sd);
-	i2c_put_byte(client,0x0104, 0x00);
+	i2c_put_byte(client,0x0103, 0x01);
+	msleep(10);
 	i2c_put_byte(client,0x0100, 0x00);
 }
 
-- 
2.19.0

