From 484c832ed32d4e46c17a635fd271db777360cd3f Mon Sep 17 00:00:00 2001
From: "jiyu.yang" <jiyu.yang@amlogic.com>
Date: Wed, 2 Apr 2014 11:07:10 +0800
Subject: [PATCH 3856/5965] PD#88382 canvas align is 16

---
 drivers/amlogic/camera/ov5647.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/drivers/amlogic/camera/ov5647.c b/drivers/amlogic/camera/ov5647.c
index d723b491b89c..bd03da8024d8 100755
--- a/drivers/amlogic/camera/ov5647.c
+++ b/drivers/amlogic/camera/ov5647.c
@@ -3560,6 +3560,9 @@ static int queue_setup(struct vb2_queue *vq, const struct v4l2_format *fmt,
         unsigned long size;
         int width = dev->width;
         int height = dev->height;
+
+        if (1080 == height)
+                height = 1088;
 				
         if (fmt)
                 size = fmt->fmt.pix.sizeimage;
-- 
2.19.0

