From b25da4db66cca3edfd42e328ee01b55995d4c673 Mon Sep 17 00:00:00 2001
From: Robin Zhu <robin.zhu@amlogic.com>
Date: Tue, 8 Apr 2014 19:02:52 -0700
Subject: [PATCH 3926/5965] m6tvc: fix h264 decoding after mpg2 by adding pic
 dc reset.

---
 drivers/amlogic/amports/amvdec.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/amlogic/amports/amvdec.c b/drivers/amlogic/amports/amvdec.c
index 4b6739507409..c4906b1a0ce8 100755
--- a/drivers/amlogic/amports/amvdec.c
+++ b/drivers/amlogic/amports/amvdec.c
@@ -74,6 +74,7 @@ static void amvdec_pg_enable(bool enable)
 #if MESON_CPU_TYPE >= MESON_CPU_TYPE_MESON6TVD
         WRITE_VREG(GCLK_EN, 0x3ff);
 #endif
+	CLEAR_VREG_MASK(MDEC_PIC_DC_CTRL, 1<<31);
     } else {
         AMVDEC_CLK_GATE_OFF(AMRISC);
 
-- 
2.19.0

