From b16e14e74af697e6e846a55daaf5b5473efc23fe Mon Sep 17 00:00:00 2001
From: "sunny.luo" <sunny.luo@amlogic.com>
Date: Thu, 24 Apr 2014 17:13:34 +0800
Subject: [PATCH 3951/5965] pd#88150: optimize i2c driver for suspend/resume
 issue

---
 drivers/amlogic/i2c/aml_i2c.c | 9 +++++++--
 1 file changed, 7 insertions(+), 2 deletions(-)

diff --git a/drivers/amlogic/i2c/aml_i2c.c b/drivers/amlogic/i2c/aml_i2c.c
index 84e51c1b7cbc..8df5bb46df7d 100755
--- a/drivers/amlogic/i2c/aml_i2c.c
+++ b/drivers/amlogic/i2c/aml_i2c.c
@@ -1195,11 +1195,14 @@ static	int aml_i2c_suspend(struct device *dev)
 	BUG_ON(!adapter);
 	i2c = i2c_get_adapdata(adapter);
 	BUG_ON(!i2c);
-	i2c->state = I2C_STATE_SUSPEND;
 	if (i2c->mode == I2C_INTERRUPT_MODE) {
+  	mutex_lock(i2c->lock);
+  	i2c->state = I2C_STATE_SUSPEND;
 		disable_irq(i2c->irq);
+	  mutex_unlock(i2c->lock);
 	  printk("%s: disable #%d irq\n", __func__, i2c->irq);
 	}
+
 	return 0;
 }
 
@@ -1215,9 +1218,11 @@ static	int aml_i2c_resume(struct device *dev)
 	BUG_ON(!adapter);
 	i2c = i2c_get_adapdata(adapter);
 	BUG_ON(!i2c);
-	i2c->state = I2C_STATE_IDLE;
 	if (i2c->mode == I2C_INTERRUPT_MODE) {
+    mutex_lock(i2c->lock);
+    i2c->state = I2C_STATE_IDLE;
 		enable_irq(i2c->irq);
+	  mutex_unlock(i2c->lock);
 	  printk("%s: enable #%d irq\n", __func__, i2c->irq);
 	}
 	return 0;
-- 
2.19.0

