From 8608c994668c66d4d2419eaed3585ac1011554ff Mon Sep 17 00:00:00 2001
From: "junliang.zhou" <junliang.zhou@amlogic.com>
Date: Mon, 14 Apr 2014 14:35:57 +0800
Subject: [PATCH 3960/5965] PD #89268: drop video frame after seek

---
 drivers/amlogic/ionvideo/ionvideo.c | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/drivers/amlogic/ionvideo/ionvideo.c b/drivers/amlogic/ionvideo/ionvideo.c
index f046e5b8c1f3..f690480f821e 100755
--- a/drivers/amlogic/ionvideo/ionvideo.c
+++ b/drivers/amlogic/ionvideo/ionvideo.c
@@ -611,6 +611,13 @@ static int vidioc_synchronization_dqbuf(struct file *file, void *priv, struct v4
         tsync_avevent_locked(VIDEO_START, buf->pts ? buf->pts : timestamp_vpts_get());
         dev->receiver_register = 0;
         d = 0;
+
+		while(!list_empty(&q->done_list)) {
+			   ret = vb2_ioctl_dqbuf(file, priv, p);
+			   if (ret) {  return ret; }
+			   ret = vb2_ioctl_qbuf(file, priv, p);
+			   if (ret) { return ret; }
+		}
     } else if (buf->pts) {
         if (abs(timestamp_pcrscr_get() - buf->pts) > tsync_vpts_discontinuity_margin()) {
             tsync_avevent_locked(VIDEO_TSTAMP_DISCONTINUITY, buf->pts);
-- 
2.19.0

