From 0764b6c7e5cc9764b90ddb0a91c7734c537fccfb Mon Sep 17 00:00:00 2001
From: "yongcheng.chen" <yongcheng.chen@amlogic.com>
Date: Mon, 28 Apr 2014 20:19:52 +0800
Subject: [PATCH 3961/5965] PD #91205, support GD25Q16 spi flash

---
 drivers/amlogic/spi_nor/spi_nor.c | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/drivers/amlogic/spi_nor/spi_nor.c b/drivers/amlogic/spi_nor/spi_nor.c
index 1ed489cf0d3b..095007059a08 100755
--- a/drivers/amlogic/spi_nor/spi_nor.c
+++ b/drivers/amlogic/spi_nor/spi_nor.c
@@ -471,6 +471,12 @@ static struct flash_info spi_nor_data [] = {
 	{ "mx25l3205d", 0xC22016, 64 * 1024, 64, BLOCK_64K, },
 	{ "mx25l6445e", 0xC22017, 64 * 1024, 128, SECT_4K, },
 	{ "mx25l2805d", 0xC22018, 64 * 1024, 256, SECT_4K, },
+
+    /*GigaByte----*/
+    { "GD25Q16", 0xC84015, 64 * 1024, 32, SECT_4K, },
+    { "GD25Q32", 0xC84016, 64 * 1024, 64, SECT_4K, },
+    { "GD25Q40", 0xC84013, 64 * 1024, 8, SECT_4K, },
+
 	 /*PM spi nor*/
 	{"pm25lq032c", 0x7f9d46, 64*1024, 64, SECT_4K, },
 };
-- 
2.19.0

