From 59b939407f3dec8b6d62651cbd8b8f073deef51c Mon Sep 17 00:00:00 2001
From: "hui.zhang" <hui.zhang@amlogic.com>
Date: Wed, 30 Apr 2014 09:19:40 +0800
Subject: [PATCH 3987/5965] pd# 90741: clear alloc_work workqueue before 4k2k
 remove

---
 drivers/amlogic/amports/vh264_4k2k.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/amlogic/amports/vh264_4k2k.c b/drivers/amlogic/amports/vh264_4k2k.c
index 668f5ffe8f1b..2a10c98b8221 100755
--- a/drivers/amlogic/amports/vh264_4k2k.c
+++ b/drivers/amlogic/amports/vh264_4k2k.c
@@ -1361,8 +1361,7 @@ static int amvdec_h264_4k2k_probe(struct platform_device *pdev)
 
 static int amvdec_h264_4k2k_remove(struct platform_device *pdev)
 {
-    printk("amvdec_h264_4k2k_remove\n");
-
+    cancel_work_sync(&alloc_work);
     mutex_lock(&vh264_4k2k_mutex);
 
     vh264_4k2k_stop();
@@ -1380,6 +1379,7 @@ static int amvdec_h264_4k2k_remove(struct platform_device *pdev)
 
     mutex_unlock(&vh264_4k2k_mutex);
     
+    printk("amvdec_h264_4k2k_remove\n");
     return 0;
 }
 
-- 
2.19.0

