From 27560cd279c45a0c2ea19ba7e831b4c105aafe8b Mon Sep 17 00:00:00 2001
From: "zhe.wang" <zhe.wang@amlogic.com>
Date: Tue, 6 May 2014 14:10:24 +0800
Subject: [PATCH 4000/5965] PD[90997]: alsa ptr check only in I2SIN source on
 M8 platform

---
 sound/soc/aml/m8/aml_i2s.c | 14 ++++++++------
 1 file changed, 8 insertions(+), 6 deletions(-)

diff --git a/sound/soc/aml/m8/aml_i2s.c b/sound/soc/aml/m8/aml_i2s.c
index 3bcf8c835d2d..eaca8d4f639e 100755
--- a/sound/soc/aml/m8/aml_i2s.c
+++ b/sound/soc/aml/m8/aml_i2s.c
@@ -682,11 +682,13 @@ static int aml_i2s_copy_capture(struct snd_pcm_runtime *runtime, int channel,
     unsigned int t1, t2;
 	struct aml_runtime_data *prtd = runtime->private_data;
 	audio_stream_t *s = &prtd->s;
-    char *hwbuf = runtime->dma_area + frames_to_bytes(runtime, pos)*2;
-	unsigned int buffersize = (unsigned int)runtime->buffer_size*8;  //512*4*4*2
-	unsigned int hw_ptr = aml_get_in_wr_ptr();
-	unsigned int alsa_read_ptr = frames_to_bytes(runtime, pos)*2;
-	int size = (buffersize + hw_ptr - alsa_read_ptr)%buffersize;
+	if(s->device_type == AML_AUDIO_I2SIN){
+		char *hwbuf = runtime->dma_area + frames_to_bytes(runtime, pos)*2;
+		unsigned int buffersize = (unsigned int)runtime->buffer_size*8;  //512*4*4*2
+		unsigned int hw_ptr = aml_get_in_wr_ptr();
+		unsigned int alsa_read_ptr = frames_to_bytes(runtime, pos)*2;
+		int size = (buffersize + hw_ptr - alsa_read_ptr)%buffersize;
+	}
     unsigned char r_shift = 8;
 	if(s->device_type == AML_AUDIO_SPDIFIN) //spdif in
     {
@@ -699,7 +701,7 @@ static int aml_i2s_copy_capture(struct snd_pcm_runtime *runtime, int channel,
 		printk("Too many datas to read,please enlarge the snd_i2s_tmp buffer size\n");
       return -EINVAL;
     }
-	if(size < 2*n){
+	if(size < 2*n && s->device_type == AML_AUDIO_I2SIN){
 		printk(KERN_DEBUG "~~~Reset ALSA!~~~\n");
 		return -EPIPE;
 	}
-- 
2.19.0

