From 3a50e5e5b2d2544408bb52c5fb446a076b88a16d Mon Sep 17 00:00:00 2001
From: "zhe.wang" <zhe.wang@amlogic.com>
Date: Tue, 6 May 2014 14:49:17 +0800
Subject: [PATCH 4001/5965] PD[90997]: correct compile error in last commit

---
 sound/soc/aml/m8/aml_i2s.c | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/sound/soc/aml/m8/aml_i2s.c b/sound/soc/aml/m8/aml_i2s.c
index eaca8d4f639e..eee1921497f1 100755
--- a/sound/soc/aml/m8/aml_i2s.c
+++ b/sound/soc/aml/m8/aml_i2s.c
@@ -682,12 +682,13 @@ static int aml_i2s_copy_capture(struct snd_pcm_runtime *runtime, int channel,
     unsigned int t1, t2;
 	struct aml_runtime_data *prtd = runtime->private_data;
 	audio_stream_t *s = &prtd->s;
+	char *hwbuf = runtime->dma_area + frames_to_bytes(runtime, pos)*2;
+	int size = 0;
 	if(s->device_type == AML_AUDIO_I2SIN){
-		char *hwbuf = runtime->dma_area + frames_to_bytes(runtime, pos)*2;
 		unsigned int buffersize = (unsigned int)runtime->buffer_size*8;  //512*4*4*2
 		unsigned int hw_ptr = aml_get_in_wr_ptr();
 		unsigned int alsa_read_ptr = frames_to_bytes(runtime, pos)*2;
-		int size = (buffersize + hw_ptr - alsa_read_ptr)%buffersize;
+		size = (buffersize + hw_ptr - alsa_read_ptr)%buffersize;
 	}
     unsigned char r_shift = 8;
 	if(s->device_type == AML_AUDIO_SPDIFIN) //spdif in
-- 
2.19.0

