From 3ff8fda095912ecd61d16e3a95d7e8afa8c5e738 Mon Sep 17 00:00:00 2001
From: "hui.zhang" <hui.zhang@amlogic.com>
Date: Wed, 7 May 2014 11:20:08 +0800
Subject: [PATCH 4012/5965] pd# 91139: do cancel_work lists before 264 dev
 remove

---
 drivers/amlogic/amports/vh264.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/drivers/amlogic/amports/vh264.c b/drivers/amlogic/amports/vh264.c
index f936954c9f66..f0f468ace545 100755
--- a/drivers/amlogic/amports/vh264.c
+++ b/drivers/amlogic/amports/vh264.c
@@ -1859,6 +1859,8 @@ static int amvdec_h264_probe(struct platform_device *pdev)
 
 static int amvdec_h264_remove(struct platform_device *pdev)
 {
+     cancel_work_sync(&error_wd_work);
+     cancel_work_sync(&stream_switching_work);
     mutex_lock(&vh264_mutex);
     vh264_stop();
 
-- 
2.19.0

