From f2183947fd5130a0473bde2d5d3582ff56b8c803 Mon Sep 17 00:00:00 2001
From: Evoke Zhang <evoke.zhang@amlogic.com>
Date: Thu, 8 May 2014 09:55:38 +0800
Subject: [PATCH 4015/5965] PD#91098: fix m6 compile mistake

---
 drivers/amlogic/display/vout/lcdoutc.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/amlogic/display/vout/lcdoutc.c b/drivers/amlogic/display/vout/lcdoutc.c
index da8b3da52d95..1a5c5c01e9f2 100755
--- a/drivers/amlogic/display/vout/lcdoutc.c
+++ b/drivers/amlogic/display/vout/lcdoutc.c
@@ -4317,7 +4317,7 @@ static inline int _get_lcd_model_timing(struct platform_device *pdev)
 	int ret=0;
 	const char *str;
 	unsigned int val;
-	unsigned int lcd_para[DSI_INIT_ON_MAX];
+	unsigned int lcd_para[100];
 	int i, j;
 	struct device_node *lcd_model_node;
 	phandle fhandle;
-- 
2.19.0

