From fb581524e60f68dac8b962e850ceba2a13ee375a Mon Sep 17 00:00:00 2001
From: "spike.liao" <spike.liao@amlogic.com>
Date: Thu, 8 May 2014 10:28:29 +0800
Subject: [PATCH 4016/5965] PD #91166 fix vh264 driver array bound writeZ

---
 drivers/amlogic/amports/vh264.c | 8 +++++++-
 1 file changed, 7 insertions(+), 1 deletion(-)

diff --git a/drivers/amlogic/amports/vh264.c b/drivers/amlogic/amports/vh264.c
index f0f468ace545..27325742f514 100755
--- a/drivers/amlogic/amports/vh264.c
+++ b/drivers/amlogic/amports/vh264.c
@@ -572,15 +572,21 @@ static void vh264_set_params(void)
     if (max_dpb_size > 5) {
         if (actual_dpb_size < max_dpb_size + 3) {
             actual_dpb_size = max_dpb_size + 3;
+        if (actual_dpb_size > 24) {    
+            actual_dpb_size = 24;
+        }            
             max_reference_size = (frame_buffer_size - mb_total * 384 * actual_dpb_size) / (mb_total * mb_mv_byte);
         }
     } else {
         if (actual_dpb_size < max_dpb_size + 4) {
             actual_dpb_size = max_dpb_size + 4;
+        if (actual_dpb_size > 24) {
+            actual_dpb_size = 24;
+        }            
             max_reference_size = (frame_buffer_size - mb_total * 384 * actual_dpb_size) / (mb_total * mb_mv_byte);
         }
     }
-
+ 
     if (!(READ_VREG(AV_SCRATCH_F) & 0x1)) {
         addr = buf_start;
 
-- 
2.19.0

