From 18e473083de4ef453949def1b48ffac24ce5c4ab Mon Sep 17 00:00:00 2001
From: xu xing <xing.xu@amlogic.com>
Date: Mon, 12 May 2014 19:09:48 +0800
Subject: [PATCH 4051/5965] meson8b: gpio: add gpio_bsd_en and test_n pin

---
 arch/arm/mach-meson8b/pinctrl.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/arch/arm/mach-meson8b/pinctrl.c b/arch/arm/mach-meson8b/pinctrl.c
index df71dbbe88e3..72c070527ce5 100755
--- a/arch/arm/mach-meson8b/pinctrl.c
+++ b/arch/arm/mach-meson8b/pinctrl.c
@@ -191,6 +191,8 @@ const static struct pinctrl_pin_desc m8b_pads[] = {
 	PINCTRL_PIN(HDMI_TTL_2_N,"HDMI_TTL_2_N"),
 	PINCTRL_PIN(HDMI_TTL_CK_P,"HDMI_TTL_CK_P"),
 	PINCTRL_PIN(HDMI_TTL_CK_N,"HDMI_TTL_CK_N"),
+	PINCTRL_PIN(GPIO_BSD_EN,"GPIO_BSD_EN"),
+	PINCTRL_PIN(GPIO_TEST_N,"GPIO_TEST_N"),
 
 };
 int m8b_pin_to_pullup(unsigned int pin ,unsigned int *reg,unsigned int *bit,unsigned int *bit_en)
-- 
2.19.0

