From 780cff9767ea4d9f9f57cb35161a6bfd234c1d94 Mon Sep 17 00:00:00 2001
From: "larson.jiang" <larson.jiang@amlogic.com>
Date: Tue, 13 May 2014 20:59:23 +0800
Subject: [PATCH 4074/5965] release lcd RGB low tow bit

---
 drivers/amlogic/display/vout/lcdoutc.c | 1 -
 1 file changed, 1 deletion(-)

diff --git a/drivers/amlogic/display/vout/lcdoutc.c b/drivers/amlogic/display/vout/lcdoutc.c
index 400dca7bd330..272366db5baf 100755
--- a/drivers/amlogic/display/vout/lcdoutc.c
+++ b/drivers/amlogic/display/vout/lcdoutc.c
@@ -323,7 +323,6 @@ static void lcd_ports_ctrl_edp(Bool_t status)
 #endif
 static unsigned pinmux_data_set[][2]={
     {PERIPHS_PIN_MUX_0,0x03c00000},//RGB[7:2]
-    {PERIPHS_PIN_MUX_8,0x00070000},//RGB[1:0]
 };
 static unsigned pinmux_data_clr[][2]={
     {PERIPHS_PIN_MUX_6,0x0000077f},//RGB[7:2]
-- 
2.19.0

