From 2476baad6722b042e2e498bcffc6d81ce364c918 Mon Sep 17 00:00:00 2001
From: "alex.deng" <alex.deng@amlogic.com>
Date: Tue, 13 May 2014 21:35:04 +0800
Subject: [PATCH 4075/5965] m101:g-sensor mc3230 can work

---
 .../boot/dts/amlogic/meson8b_m101_512M.dtd    | 21 +++++++++++++++----
 1 file changed, 17 insertions(+), 4 deletions(-)

diff --git a/arch/arm/boot/dts/amlogic/meson8b_m101_512M.dtd b/arch/arm/boot/dts/amlogic/meson8b_m101_512M.dtd
index ebff3712308a..de0ec0a21838 100755
--- a/arch/arm/boot/dts/amlogic/meson8b_m101_512M.dtd
+++ b/arch/arm/boot/dts/amlogic/meson8b_m101_512M.dtd
@@ -1041,16 +1041,29 @@ void root_func(){
 		compatible = "amlogic,aml_sensor";
 		status = "okay";
 
-//$$ DEVICE="stk831x"
+//$$ DEVICE="mc32x0"
 //$$ L2 PROP_STR = "status"
 //$$ L2 PROP_U32 = "address"
 //$$ L2 PROP_STR = "i2c_bus"
-        stk831x{
-            dev_name = "stk831x";
+        mc32x0{
+            dev_name = "mc32x0";
             status = "okay";
-            address = <0x3d>;
+            address = <0x4c>;
             i2c_bus = "i2c_bus_b";
+            acc_dir = <0x0010>;
         };
+//$$ DEVICE="mma7660"
+//$$ L2 PROP_STR = "status"
+//$$ L2 PROP_U32 = "address"
+//$$ L2 PROP_STR = "i2c_bus"
+//        mma7660{
+//            dev_name = "mma7660";
+//            status = "okay";
+//            address = <0x4c>;
+//            i2c_bus = "i2c_bus_b";
+//       };
+
+
 	};
 
 
-- 
2.19.0

