From 9ed54d4dc6b6f44af8ada077b440b4ebb7e22b42 Mon Sep 17 00:00:00 2001
From: "kele.bai" <kele.bai@amlogic.com>
Date: Wed, 14 May 2014 15:37:07 +0800
Subject: [PATCH 4091/5965] PD #91784:add protect for normal di post mtnp.

---
 drivers/amlogic/deinterlace/deinterlace.c | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/drivers/amlogic/deinterlace/deinterlace.c b/drivers/amlogic/deinterlace/deinterlace.c
index 10b367f54576..f994fa0ddbed 100755
--- a/drivers/amlogic/deinterlace/deinterlace.c
+++ b/drivers/amlogic/deinterlace/deinterlace.c
@@ -4751,7 +4751,8 @@ static int de_post_process(void* arg, unsigned zoom_start_x_lines,
 	    else
 	        di_post_stru.di_buf1_mif.canvas0_addr0 = di_buf->di_buf_dup_p[0]->nr_canvas_idx;
 	    di_post_stru.di_mtncrd_mif.canvas_num = di_buf->di_buf_dup_p[1]->mtn_canvas_idx;
-      di_post_stru.di_mtnprd_mif.canvas_num = di_buf->di_buf_dup_p[2]->mtn_canvas_idx;
+    if(di_buf->di_buf_dup_p[2])
+        di_post_stru.di_mtnprd_mif.canvas_num = di_buf->di_buf_dup_p[2]->mtn_canvas_idx;
 
 
     	post_blend_en = 1;
@@ -4925,7 +4926,8 @@ static int de_post_process_pd(void* arg, unsigned zoom_start_x_lines,
 	    else
 	        di_post_stru.di_buf1_mif.canvas0_addr0 = di_buf->di_buf_dup_p[0]->nr_canvas_idx;
 	    di_post_stru.di_mtncrd_mif.canvas_num = di_buf->di_buf_dup_p[1]->mtn_canvas_idx;
-      di_post_stru.di_mtnprd_mif.canvas_num = di_buf->di_buf_dup_p[2]->mtn_canvas_idx;
+      if(di_buf->di_buf_dup_p[2])
+          di_post_stru.di_mtnprd_mif.canvas_num = di_buf->di_buf_dup_p[2]->mtn_canvas_idx;
 
       if((di_post_stru.update_post_reg_flag)&&((force_update_post_reg&0x80)==0))
   	    enable_di_post_pd (
-- 
2.19.0

