From d26312248e28a3369a474ae3328d99364658483e Mon Sep 17 00:00:00 2001
From: Evoke Zhang <evoke.zhang@amlogic.com>
Date: Thu, 15 May 2014 19:18:04 +0800
Subject: [PATCH 4104/5965] m8 baby:update backlight control

---
 drivers/amlogic/display/vout/lcdoutc.c | 9 +++++----
 1 file changed, 5 insertions(+), 4 deletions(-)

diff --git a/drivers/amlogic/display/vout/lcdoutc.c b/drivers/amlogic/display/vout/lcdoutc.c
index 780041f7c858..3cc202e82046 100755
--- a/drivers/amlogic/display/vout/lcdoutc.c
+++ b/drivers/amlogic/display/vout/lcdoutc.c
@@ -100,7 +100,7 @@ static lcd_dev_t *pDev = NULL;
 static spinlock_t gamma_write_lock;
 static spinlock_t lcd_clk_lock;
 static Bool_t data_status = ON;
-static int bl_status = 0;
+static int bl_status = ON;
 
 static inline void lcd_mdelay(int n)
 {
@@ -3355,20 +3355,21 @@ static int lcd_set_current_vmode(vmode_t mode)
 		mutex_unlock(&lcd_vout_mutex);
 		return -EINVAL;
 	}
-	_disable_backlight();
-	vpp2_sel = 0;
 
+	vpp2_sel = 0;
 	WRITE_LCD_REG(VPP_POSTBLEND_H_SIZE, pDev->lcd_info.width);
 
 	if( !(mode&VMODE_LOGO_BIT_MASK) ){
+		_disable_backlight();
 #if ((MESON_CPU_TYPE == MESON_CPU_TYPE_MESON8) || (MESON_CPU_TYPE == MESON_CPU_TYPE_MESON8B))
 		request_vpu_clk_vmod(pDev->lcd_info.video_clk, pDev->lcd_info.mode);
 #endif
 		_lcd_module_enable();
+		_enable_backlight();
 	}
 	if (VMODE_INIT_NULL == pDev->lcd_info.mode)
 		pDev->lcd_info.mode = VMODE_LCD;
-	_enable_backlight();
+	
 	mutex_unlock(&lcd_vout_mutex);
 	return 0;
 }
-- 
2.19.0

