From ee210f15c96833053aaf15130a56c863875c007b Mon Sep 17 00:00:00 2001
From: timyao <tim.yao@amlogic.com>
Date: Fri, 16 May 2014 14:11:50 -0700
Subject: [PATCH 4148/5965] PD #91886. Fix HEVC streaming buffer input setting.

---
 drivers/amlogic/amports/esparser.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/amlogic/amports/esparser.c b/drivers/amlogic/amports/esparser.c
index 0e253f67aef6..9bba4ce0b6bf 100755
--- a/drivers/amlogic/amports/esparser.c
+++ b/drivers/amlogic/amports/esparser.c
@@ -345,7 +345,7 @@ s32 esparser_init(struct stream_buf_s *buf)
         WRITE_MPEG_REG(PARSER_VIDEO_START_PTR,
                        READ_VREG(HEVC_STREAM_START_ADDR));
         WRITE_MPEG_REG(PARSER_VIDEO_END_PTR,
-                       READ_VREG(HEVC_STREAM_END_ADDR));
+                       READ_VREG(HEVC_STREAM_END_ADDR) - 8);
 
         CLEAR_MPEG_REG_MASK(PARSER_ES_CONTROL, ES_VID_MAN_RD_PTR);
 
-- 
2.19.0

