From 01ed36142d8a756574bd8b14a159e9ff7038c43c Mon Sep 17 00:00:00 2001
From: Evoke Zhang <evoke.zhang@amlogic.com>
Date: Wed, 21 May 2014 10:08:00 +0800
Subject: [PATCH 4163/5965] fix m8 compile error for lcd driver

---
 drivers/amlogic/display/vout/lcdoutc.c | 2 --
 1 file changed, 2 deletions(-)

diff --git a/drivers/amlogic/display/vout/lcdoutc.c b/drivers/amlogic/display/vout/lcdoutc.c
index 3cc202e82046..e168a4f7c353 100755
--- a/drivers/amlogic/display/vout/lcdoutc.c
+++ b/drivers/amlogic/display/vout/lcdoutc.c
@@ -2473,10 +2473,8 @@ static void generate_clk_parameter(Lcd_Config_t *pConf)
     unsigned od_sel, pre_div_sel;
     unsigned div_pre_sel_max, crt_xd_max;
     unsigned f_ref, pll_vco, fout_pll, div_pre_out, div_post_out, final_freq, iflogic_vid_clk_in_max;
-#if (MESON_CPU_TYPE == MESON_CPU_TYPE_MESON6)
     unsigned min_error = MAX_ERROR;
     unsigned error = MAX_ERROR;
-#endif
 #if ((MESON_CPU_TYPE == MESON_CPU_TYPE_MESON8) || (MESON_CPU_TYPE == MESON_CPU_TYPE_MESON8B))
     unsigned od_fb=0;
     unsigned int dsi_bit_rate_min=0, dsi_bit_rate_max=0;
-- 
2.19.0

