From 4332439a14296c4395f0c39a5c887a53bb2e27f3 Mon Sep 17 00:00:00 2001
From: Larson Jiang <larson.jiang@amlogic.com>
Date: Thu, 22 May 2014 15:19:24 +0800
Subject: [PATCH 4175/5965] #92095, fix sd debug board problem

---
 drivers/amlogic/mmc/amlsd.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/drivers/amlogic/mmc/amlsd.c b/drivers/amlogic/mmc/amlsd.c
index 0db0d874b385..eab1367cedcc 100755
--- a/drivers/amlogic/mmc/amlsd.c
+++ b/drivers/amlogic/mmc/amlsd.c
@@ -922,7 +922,7 @@ static int aml_is_card_insert (struct amlsd_platform * pdata)
 	return ret;
 }
 
-#ifdef CONFIG_ARCH_MESON8
+#if MESON_CPU_TYPE >= MESON_CPU_TYPE_MESON8
 static int aml_is_sdjtag(struct amlsd_platform * pdata)
 {
     int card0;
@@ -1071,7 +1071,7 @@ void aml_sd_uart_detect_clr (struct amlsd_platform* pdata)
 
 void aml_sd_uart_detect (struct amlsd_platform* pdata)
 {
-#ifdef CONFIG_ARCH_MESON8
+#if MESON_CPU_TYPE >= MESON_CPU_TYPE_MESON8
     static bool is_jtag = false;
 
     if (aml_is_card_insert(pdata)){
@@ -1169,7 +1169,7 @@ irqreturn_t aml_sd_irq_cd(int irq, void *dev_id)
 void aml_sduart_pre (struct amlsd_platform* pdata)
 {
     if (((pdata->port == MESON_SDIO_PORT_B) || (pdata->port == MESON_SDIO_PORT_XC_B))) {
-#ifdef CONFIG_ARCH_MESON8
+#if MESON_CPU_TYPE >= MESON_CPU_TYPE_MESON8
         // clear pinmux of CARD_0 and CARD_4 to make them used as gpio
         CLEAR_CBUS_REG_MASK(PERIPHS_PIN_MUX_2, 0x00005040);
         CLEAR_CBUS_REG_MASK(PERIPHS_PIN_MUX_8, 0x00000400);
-- 
2.19.0

