From b3189b134ff05f37191af42b812639dae5307697 Mon Sep 17 00:00:00 2001
From: "zhe.wang" <zhe.wang@amlogic.com>
Date: Fri, 23 May 2014 17:28:16 +0800
Subject: [PATCH 4233/5965] PD[91683]: correct init error

---
 sound/soc/aml/m6/aml_pcm.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/sound/soc/aml/m6/aml_pcm.c b/sound/soc/aml/m6/aml_pcm.c
index b3f7ab4d1a8c..4a8e287f6c1e 100755
--- a/sound/soc/aml/m6/aml_pcm.c
+++ b/sound/soc/aml/m6/aml_pcm.c
@@ -963,7 +963,6 @@ static int aml_pcm_open(struct snd_pcm_substream *substream)
 {
 	struct snd_pcm_runtime *runtime = substream->runtime;
 	struct aml_runtime_data *prtd;
-	audio_stream_t *s = &prtd->s;
 	int ret = 0;
 
 	if (substream->stream == SNDRV_PCM_STREAM_PLAYBACK){
@@ -1013,6 +1012,7 @@ static int aml_pcm_open(struct snd_pcm_substream *substream)
 	runtime->private_data = prtd;
 
 	spin_lock_init(&prtd->s.lock);
+	audio_stream_t *s = &prtd->s;
 	s->xrun_num = 0;
  out:
 	return ret;
-- 
2.19.0

