From bde02906bd247add7fb9ab4f6af86355d04e0876 Mon Sep 17 00:00:00 2001
From: Mengchun Li <mengchun.li@amlogic.com>
Date: Wed, 28 May 2014 19:10:28 +0800
Subject: [PATCH 4241/5965] PD#92490:wifi:m102 wifi cann't connect

---
 drivers/amlogic/wifi/wifi_power.c | 12 ------------
 1 file changed, 12 deletions(-)

diff --git a/drivers/amlogic/wifi/wifi_power.c b/drivers/amlogic/wifi/wifi_power.c
index 87a7cafe54bf..d2e8a7ffa57c 100755
--- a/drivers/amlogic/wifi/wifi_power.c
+++ b/drivers/amlogic/wifi/wifi_power.c
@@ -105,9 +105,6 @@ static int wifi_power_ioctl(struct file *filp, unsigned int cmd, unsigned long a
      		mdelay(500);
     	  usb_wifi_power(0);   
      		printk(KERN_INFO "Meson8 set usb wifi power up!\n");
-     		#elif (MESON_CPU_TYPE == MESON_CPU_TYPE_MESON8B)
-     		usb_wifi_power(0);   
-     		printk(KERN_INFO "Meson8 Baby set usb wifi power up!\n");
    		  #elif (MESON_CPU_TYPE == MESON_CPU_TYPE_MESON6)
     		usb_wifi_power(0);
     		mdelay(500);
@@ -281,13 +278,6 @@ static void usb_wifi_power(int is_power)
 //    CLEAR_CBUS_REG_MASK(PERIPHS_PIN_MUX_1,(1<<11));
 //    CLEAR_CBUS_REG_MASK(PERIPHS_PIN_MUX_0,(1<<18));
 #ifdef CONFIG_OF
-#if ( MESON_CPU_TYPE == MESON_CPU_TYPE_MESON8B )	
-	printk(KERN_INFO "mcli: -- usb_wifi_power m8 baby! --\n");	
-	    if(is_power)
-            amlogic_gpio_direction_output(pdata->power_gpio, 1, WIFI_POWER_MODULE_NAME); 	  
-	    else
-	        amlogic_gpio_direction_output(pdata->power_gpio, 0, WIFI_POWER_MODULE_NAME); 
-#else
 	if(pdata->power_gpio > 0)
 	{
 	    if(is_power)
@@ -295,8 +285,6 @@ static void usb_wifi_power(int is_power)
 	    else
 	        amlogic_gpio_direction_output(pdata->power_gpio, 1, WIFI_POWER_MODULE_NAME); 
     }
-#endif
-    
 	if(wifi_power_on_pin2){
 	    	if(pdata->power_gpio2 > 0)
 		{
-- 
2.19.0

