From f04e40e29f740cfe506941bc140215312315e397 Mon Sep 17 00:00:00 2001
From: Sandy lUo <sandy.luo@amlogic.com>
Date: Thu, 29 May 2014 15:23:33 +0800
Subject: [PATCH 4250/5965] [m101]:PMU1218:change battery para

---
 .../boot/dts/amlogic/meson8b_m101_512M.dtd    | 26 +++++++++----------
 1 file changed, 13 insertions(+), 13 deletions(-)

diff --git a/arch/arm/boot/dts/amlogic/meson8b_m101_512M.dtd b/arch/arm/boot/dts/amlogic/meson8b_m101_512M.dtd
index 8bcee3aaa01a..8fa59f24810c 100755
--- a/arch/arm/boot/dts/amlogic/meson8b_m101_512M.dtd
+++ b/arch/arm/boot/dts/amlogic/meson8b_m101_512M.dtd
@@ -784,8 +784,8 @@ void root_func(){
 	    pmu_twi_id = <2>;                   /** PMU i2c bus                          */
         pmu_irq_id = <0>;                   /** INT_WATCHDOG                         */
         pmu_twi_addr = <0x35>;              /** PMU i2c addr                         */
-        pmu_battery_rdc = <100>;            /** battery internal resistor            */
-        pmu_battery_cap = <3670>;           /** battery capacity                     */
+        pmu_battery_rdc = <89>;            /** battery internal resistor            */
+        pmu_battery_cap = <2834>;           /** battery capacity                     */
         pmu_battery_technology = <4>;       /** POWER_SUPPLY_TECHNOLOGY_LiFe         */
         pmu_battery_name = "PTI PL336078";  /** battery name                         */
         pmu_init_chgvol = <4200000>;        /** target charge voltage                */
@@ -812,7 +812,7 @@ void root_func(){
         pmu_pwrnoe_time = <2000>;           /**                                      */
         pmu_intotp_en = <1>;                /** over temperature protection enable   */
         pmu_pekon_time = <128>;             /** powerkey hold time for power on      */
-        pmu_charge_efficiency = <99>;       /** efficient of charging                */
+        pmu_charge_efficiency = <89>;       /** efficient of charging                */
         pmu_bat_curve = <
          /* ocv, charge, discharge */
             3132      0       0
@@ -820,16 +820,16 @@ void root_func(){
             3414      0       0
             3585      0       0  
             3625      1       2  
-            3660      2       5  
-            3696      3      12 
-            3731      8      19 
-            3766     13      32 
-            3801     19      45 
-            3836     36      54
-            3872     48      61
-            3942     63      71
-            4012     77      81
-            4083     87      90
+            3660      2       6  
+            3696      3      13 
+            3731      9      19 
+            3766     15      35 
+            3801     21      48 
+            3836     39      57
+            3872     50      63
+            3942     62      73
+            4012     74      83
+            4083     85      92
             4153    100     100>;
 	};
 
-- 
2.19.0

