From b2331fd221edacc877b1abda35c124fdfd563175 Mon Sep 17 00:00:00 2001
From: "hui.fang" <hui.fang@amlogic.com>
Date: Fri, 25 Apr 2014 11:31:01 +0800
Subject: [PATCH 4264/5965] video:fix the cat func of crop

---
 drivers/amlogic/amports/video.c | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/drivers/amlogic/amports/video.c b/drivers/amlogic/amports/video.c
index fb513ea4aaee..285b602751ca 100755
--- a/drivers/amlogic/amports/video.c
+++ b/drivers/amlogic/amports/video.c
@@ -3872,8 +3872,7 @@ static ssize_t video_crop_show(struct class *cla, struct class_attribute *attr,
 {
     u32 t, l, b, r;
 
-    vpp_get_video_layer_position(&t, &l, &b, &r);
-
+    vpp_get_video_source_crop(&t, &l, &b, &r);
     return snprintf(buf, 40, "%d %d %d %d\n", t, l, b, r);
 }
 
-- 
2.19.0

