From 9f769febbf8894dbb7f70d55ff1920e78520f409 Mon Sep 17 00:00:00 2001
From: Victor Wan <victor.wan@amlogic.com>
Date: Fri, 30 May 2014 20:55:22 +0800
Subject: [PATCH 4271/5965] meson8b: fix max cpufreq to 1488M

---
 arch/arm/mach-meson8b/clock.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/arch/arm/mach-meson8b/clock.c b/arch/arm/mach-meson8b/clock.c
index c70265eb3ead..d89c4f4291e3 100755
--- a/arch/arm/mach-meson8b/clock.c
+++ b/arch/arm/mach-meson8b/clock.c
@@ -60,7 +60,7 @@ static DEFINE_MUTEX(clock_ops_lock);
 #define SYS_PLL_TABLE_MIN	 24000000
 #define SYS_PLL_TABLE_MAX	2112000000
 
-#define CPU_FREQ_LIMIT 1608000000
+#define CPU_FREQ_LIMIT 1488000000
 
 struct sys_pll_s {
     unsigned int freq;
-- 
2.19.0

