From cde5bd8c85a4dfee95c4f84226c80f8ffb5c6e3c Mon Sep 17 00:00:00 2001
From: timyao <tim.yao@amlogic.com>
Date: Thu, 5 Jun 2014 17:54:05 -0700
Subject: [PATCH 4299/5965] PD #91368-2. Improve 4k2k decoder watchdog
 condition.

---
 drivers/amlogic/amports/vh264_4k2k.c | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/drivers/amlogic/amports/vh264_4k2k.c b/drivers/amlogic/amports/vh264_4k2k.c
index bafc6077ce3c..595ec67cb11a 100755
--- a/drivers/amlogic/amports/vh264_4k2k.c
+++ b/drivers/amlogic/amports/vh264_4k2k.c
@@ -54,7 +54,7 @@
 #define MODULE_NAME "amvdec_h264_4k2k"
 
 #define PUT_INTERVAL        (HZ/100)
-#define ERROR_RESET_COUNT   100
+#define ERROR_RESET_COUNT   500
 
 
 
@@ -807,6 +807,8 @@ static void vh264_4k2k_put_timer_func(unsigned long arg)
             printk("H264 4k2k decoder fatal error watchdog.\n");
             fatal_error = DECODER_FATAL_ERROR_UNKNOW;
         }
+    } else {
+      error_watchdog_count = 0;
     }
 
     if (READ_VREG(FATAL_ERROR) != 0) {
-- 
2.19.0

