From 1fdcd4c0607c6473badacf7dbe00a036186d6d76 Mon Sep 17 00:00:00 2001
From: Lawrence Mok <lawrence.mok@amlogic.com>
Date: Fri, 6 Jun 2014 16:22:24 -0700
Subject: [PATCH 4306/5965] ethernet: call SET_NETDEV_DEV()

to setup /sys/class/net/eth0/device/driver symlink
---
 drivers/amlogic/ethernet/am_net8218.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/amlogic/ethernet/am_net8218.c b/drivers/amlogic/ethernet/am_net8218.c
index 1572a8671c1a..34d3ede5ae9c 100755
--- a/drivers/amlogic/ethernet/am_net8218.c
+++ b/drivers/amlogic/ethernet/am_net8218.c
@@ -2919,6 +2919,7 @@ static int ethernet_probe(struct platform_device *pdev)
        early_suspend.resume = ethernet_late_resume;
        register_early_suspend(&early_suspend);
 #endif
+	SET_NETDEV_DEV(my_ndev, &pdev->dev);
 	res = probe_init(my_ndev);
 	if (res != 0)
 		free_netdev(my_ndev);
-- 
2.19.0

