From 40cdb7dc2f0f0b646c7d58a7e657dbf03b326a79 Mon Sep 17 00:00:00 2001
From: "kele.bai" <kele.bai@amlogic.com>
Date: Tue, 10 Jun 2014 10:56:53 +0800
Subject: [PATCH 4316/5965] PD #92489:enable vdin go field after enable di pre.

---
 drivers/amlogic/deinterlace/deinterlace.c | 18 +++++++++---------
 1 file changed, 9 insertions(+), 9 deletions(-)

diff --git a/drivers/amlogic/deinterlace/deinterlace.c b/drivers/amlogic/deinterlace/deinterlace.c
index 855b3781b442..7c04c6a8a9d2 100755
--- a/drivers/amlogic/deinterlace/deinterlace.c
+++ b/drivers/amlogic/deinterlace/deinterlace.c
@@ -3368,15 +3368,6 @@ static void pre_de_process(void)
 		   	);
 #endif
     //Wr(DI_PRE_CTRL, 0x3 << 30); // remove it for M6, can not disalbe it here
-#ifdef SUPPORT_MPEG_TO_VDIN
-	if(mpeg2vdin_en){
-	    vdin_arg_t vdin_arg;
-	    vdin_v4l2_ops_t *vdin_ops = get_vdin_v4l2_ops();
-	    vdin_arg.cmd = VDIN_CMD_FORCE_GO_FIELD;
-	    if(vdin_ops->tvin_vdin_func)
-	        vdin_ops->tvin_vdin_func(0,&vdin_arg);
-	}
-#endif
 
     enable_di_pre_aml (  &di_pre_stru.di_inp_mif,               // di_inp
                &di_pre_stru.di_mem_mif,               // di_mem
@@ -3429,6 +3420,15 @@ static void pre_de_process(void)
 	//Wr(DI_PRE_CTRL, Rd(DI_PRE_CTRL)|(1<<25));
     di_apply_reg_cfg(0);
 #endif
+#ifdef SUPPORT_MPEG_TO_VDIN
+	if(mpeg2vdin_en){
+	    vdin_arg_t vdin_arg;
+	    vdin_v4l2_ops_t *vdin_ops = get_vdin_v4l2_ops();
+	    vdin_arg.cmd = VDIN_CMD_FORCE_GO_FIELD;
+	    if(vdin_ops->tvin_vdin_func)
+	        vdin_ops->tvin_vdin_func(0,&vdin_arg);
+	}
+#endif
 
 }
 
-- 
2.19.0

