From 084333aabd688bccf9ad7faa0afec1e9aed23372 Mon Sep 17 00:00:00 2001
From: "tao.dong" <tao.dong@amlogic.com>
Date: Sat, 14 Jun 2014 14:20:35 +0800
Subject: [PATCH 4344/5965] m202:DVFS:pwmm vcck voltage change

---
 .../boot/dts/amlogic/meson8b_m202_512M.dtd    | 26 +++++++++----------
 1 file changed, 13 insertions(+), 13 deletions(-)

diff --git a/arch/arm/boot/dts/amlogic/meson8b_m202_512M.dtd b/arch/arm/boot/dts/amlogic/meson8b_m202_512M.dtd
index 3500ce0aaee1..db98a9ea4efe 100755
--- a/arch/arm/boot/dts/amlogic/meson8b_m202_512M.dtd
+++ b/arch/arm/boot/dts/amlogic/meson8b_m202_512M.dtd
@@ -726,22 +726,22 @@ void root_func(){
 //$$ L2 PROP_U32 16*3 = "dvfs_table"
         vcck_dvfs {
             dvfs_id     = <1>;                                  /** must be value of (1 << n)            */
-            table_count = <11>;                                 /** must be correct count for dvfs_table */
-            change-frequent-only;
+            table_count = <12>;                                 /** must be correct count for dvfs_table */
             dvfs_table  = <
             /* NOTE: frequent in this table must be ascending order */
             /* frequent(Khz)    min_uV      max_uV                  */
-                  96000         875000      875000
-                 192000         875000      875000
-                 312000         875000      875000
-                 408000         875000      875000
-                 504000         875000      875000
-                 600000         900000      900000
-                 720000         900000      900000
-                 816000         925000      925000
-                1008000         975000      975000
-                1200000        1025000     1025000
-                1488000        1075000     1075000
+                  96000         860000      860000
+                 192000         860000      860000
+                 312000         860000      860000
+                 408000         860000      860000
+                 504000         860000      860000
+                 600000         860000      860000
+                 720000         860000      860000
+                 816000         900000      900000
+                1008000         920000      920000
+                1200000         970000      970000
+                1320000        1050000     1050000
+                1488000        1100000     1100000
             >;
         };
     };
-- 
2.19.0

