From 50860477c429e3563201a62885fb8c5f3b0fe838 Mon Sep 17 00:00:00 2001
From: "alex.deng" <alex.deng@amlogic.com>
Date: Tue, 17 Jun 2014 19:44:52 +0800
Subject: [PATCH 4359/5965] PD 93285:stk8313 Gsensor CTS failed

---
 drivers/amlogic/input/sensor/stk8313.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/amlogic/input/sensor/stk8313.c b/drivers/amlogic/input/sensor/stk8313.c
index 882403a22e3a..9b102345b759 100755
--- a/drivers/amlogic/input/sensor/stk8313.c
+++ b/drivers/amlogic/input/sensor/stk8313.c
@@ -58,7 +58,7 @@
 #define STK_RESUME_RE_INIT	
 //#define STK_DEBUG_PRINT
 //#define STK_DEBUG_RAWDATA
-#define STK_FIR_LEN	1
+#define STK_FIR_LEN	8
 #define STK_ZG_FILTER
 
 #define STK_ZG_COUNT	4
-- 
2.19.0

