From d4235152a41df7dc8e0a604949e1cc5214109d7e Mon Sep 17 00:00:00 2001
From: "kele.bai" <kele.bai@amlogic.com>
Date: Thu, 19 Jun 2014 15:34:45 +0800
Subject: [PATCH 4369/5965] PD #92995:fix compile error in di.

---
 drivers/amlogic/deinterlace/deinterlace.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/amlogic/deinterlace/deinterlace.c b/drivers/amlogic/deinterlace/deinterlace.c
index 9ac23656ab14..8f8d4197e4ec 100755
--- a/drivers/amlogic/deinterlace/deinterlace.c
+++ b/drivers/amlogic/deinterlace/deinterlace.c
@@ -324,7 +324,7 @@ static int d2d3_enable = 1;
 #endif
 
 #ifdef DET3D
-static bool det3d_en = false;
+bool det3d_en = false;
 static unsigned int det3d_mode = 0;
 #endif
 
-- 
2.19.0

