From 0b5a827b7e2600b2a483c06c66d5f7f336f6830f Mon Sep 17 00:00:00 2001
From: "kele.bai" <kele.bai@amlogic.com>
Date: Fri, 20 Jun 2014 11:16:37 +0800
Subject: [PATCH 4382/5965] PD #93391: add missing light unreg even in d2d3

---
 drivers/amlogic/d2d3/d2d3_drv.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/amlogic/d2d3/d2d3_drv.c b/drivers/amlogic/d2d3/d2d3_drv.c
index 57cd90f74f4d..39edcc0e0071 100755
--- a/drivers/amlogic/d2d3/d2d3_drv.c
+++ b/drivers/amlogic/d2d3/d2d3_drv.c
@@ -310,7 +310,7 @@ static int d2d3_receiver_event_fun(int type, void* data, void* arg)
         int i, ret=0;
         d2d3_param_t *parm = &d2d3_devp->param;
         if((type == VFRAME_EVENT_PROVIDER_UNREG)||
-                        (type == VFRAME_EVENT_PROVIDER_LIGHT_UNREG_RETURN_VFRAME)){
+           (type == VFRAME_EVENT_PROVIDER_LIGHT_UNREG)){
                 prov = NULL;
 		if(d2d3_devp->flag & D2D3_BYPASS){
                 	d2d3_enable_hw(false);
-- 
2.19.0

