From fa4903bcf9d5c380600f82fbb9a5533cddceb92f Mon Sep 17 00:00:00 2001
From: "kele.bai" <kele.bai@amlogic.com>
Date: Fri, 20 Jun 2014 17:55:33 +0800
Subject: [PATCH 4383/5965] PD #93391: keep the buffer used by post only

---
 drivers/amlogic/deinterlace/deinterlace.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/amlogic/deinterlace/deinterlace.c b/drivers/amlogic/deinterlace/deinterlace.c
index 8f8d4197e4ec..1ed1dfa3e9de 100755
--- a/drivers/amlogic/deinterlace/deinterlace.c
+++ b/drivers/amlogic/deinterlace/deinterlace.c
@@ -2660,7 +2660,7 @@ static void di_uninit_buf(void)
     }
 
     queue_for_each_entry(p, ptmp, QUEUE_DISPLAY, list) {
-        if(p->di_buf[0]->type!=VFRAME_TYPE_IN){
+        if(p->di_buf[0]->type!=VFRAME_TYPE_IN && p->process_fun_index != PROCESS_FUN_NULL){
 #if 1
 	    if(p->index == di_post_stru.cur_disp_index){
 	        used_post_buf_index = p->index;
-- 
2.19.0

