From 2f33a533f0ac059dfac7d6eea8d6e0668fa3dc62 Mon Sep 17 00:00:00 2001
From: "benlong.zhou" <benlong.zhou@amlogic.com>
Date: Wed, 18 Jun 2014 19:01:12 +0800
Subject: [PATCH 4413/5965] PD #93184: rtc: add nandkey read for testing
 nandkey

---
 drivers/amlogic/rtc/aml_rtc.c | 19 +++++++++++++++++++
 1 file changed, 19 insertions(+)

diff --git a/drivers/amlogic/rtc/aml_rtc.c b/drivers/amlogic/rtc/aml_rtc.c
index b6254ace0cd7..0aba1110e933 100755
--- a/drivers/amlogic/rtc/aml_rtc.c
+++ b/drivers/amlogic/rtc/aml_rtc.c
@@ -692,11 +692,30 @@ static struct class aml_rtc_class = {
 };
 
 
+#ifdef CONFIG_SECURITYKEY
+extern int get_aml_key_kernel(const char* key_name, unsigned char* data, int ascii_flag);
+extern int extenal_api_key_set_version(char *devvesion);
+#endif
+
 static int aml_rtc_probe(struct platform_device *pdev)
 {
 	struct aml_rtc_priv *priv;
 	int ret;
 
+#ifdef CONFIG_SECURITYKEY
+	static char keyexamples[4096];
+#endif
+#ifdef CONFIG_SECURITYKEY
+	extenal_api_key_set_version("auto3");
+	ret = get_aml_key_kernel("keyexample", keyexamples, 1);
+	if(ret >= 0){
+		printk("key name:keyexample, key data:%s\n",keyexamples);
+	}
+	else{
+		printk("get keyexample fail, %s:%d\n",__func__,__LINE__);
+	}
+#endif
+
 	priv = (struct aml_rtc_priv *)kzalloc(sizeof(*priv), GFP_KERNEL);
 
 	if(!priv)
-- 
2.19.0

