From 05dfcd3d840f30e777cbaf47b247db2cfe37e9ec Mon Sep 17 00:00:00 2001
From: "jianqin.zhang" <jianqin.zhang@amlogic.com>
Date: Fri, 4 Jul 2014 15:15:04 +0800
Subject: [PATCH 4440/5965] pd#93885:wifi: open wifi power by Limengchun

---
 drivers/amlogic/wifi/wifi_power.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/amlogic/wifi/wifi_power.c b/drivers/amlogic/wifi/wifi_power.c
index 4c739252409e..e38d34529973 100755
--- a/drivers/amlogic/wifi/wifi_power.c
+++ b/drivers/amlogic/wifi/wifi_power.c
@@ -309,9 +309,9 @@ static int wifi_power_probe(struct platform_device *pdev)
 	     }else{
 	        pdata->power_gpio = amlogic_gpio_name_map_num(str);
 	        printk("wifi_power power_gpio is %d\n",pdata->power_gpio);
-	        //ret = amlogic_gpio_request(pdata->power_gpio,WIFI_POWER_MODULE_NAME);
+	        ret = amlogic_gpio_request(pdata->power_gpio,WIFI_POWER_MODULE_NAME);
 	        //mcli pdata->usb_set_power(0);    //power on   
-	        //pdata->usb_set_power(1);    //power on   
+	        pdata->usb_set_power(1);    //power on   
 	     }
 	     
 	    ret = of_property_read_string(pdev->dev.of_node, "valid", &str);
-- 
2.19.0

