From 0a7779b2fd6d6c036f796f03b5e003caa8c4b92f Mon Sep 17 00:00:00 2001
From: "yun.cai" <yun.cai@amlogic.com>
Date: Mon, 16 Jun 2014 16:14:04 +0800
Subject: [PATCH 4456/5965] PD#92679: fix M8 compile error

---
 drivers/amlogic/amports/Makefile | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/drivers/amlogic/amports/Makefile b/drivers/amlogic/amports/Makefile
index c9365039b04a..2770e02c2c07 100755
--- a/drivers/amlogic/amports/Makefile
+++ b/drivers/amlogic/amports/Makefile
@@ -131,7 +131,7 @@ obj-$(CONFIG_AM_VIDEO) += amvideo.o
 ifeq ($(CONFIG_ARCH_MESON8),y)
 amvideo-objs := video.o vpp.o video_prot.o
 amvideo-y    := video.o vpp.o video_prot.o
-endif
+else
 ifeq ($(CONFIG_ARCH_MESON8M2),y)
 amvideo-objs := video.o vpp.o video_prot.o
 amvideo-y    := video.o vpp.o video_prot.o
@@ -139,6 +139,7 @@ else
 amvideo-objs := video.o vpp.o
 amvideo-y    := video.o vpp.o
 endif
+endif
 
 ifeq ($(CONFIG_AM_VIDEOCAPTURE),y)
 amports-objs +=  amvideocap.o
-- 
2.19.0

