From 65c06bf5994db6899ceb4666e6ce145349701ac0 Mon Sep 17 00:00:00 2001
From: timyao <tim.yao@amlogic.com>
Date: Thu, 3 Jul 2014 20:17:17 -0700
Subject: [PATCH 4475/5965] Recover H264 4k2k decoder watchdog.

---
 drivers/amlogic/amports/vh264_4k2k.c | 2 --
 1 file changed, 2 deletions(-)

diff --git a/drivers/amlogic/amports/vh264_4k2k.c b/drivers/amlogic/amports/vh264_4k2k.c
index 341eb5cccaac..98ec5ada62e7 100755
--- a/drivers/amlogic/amports/vh264_4k2k.c
+++ b/drivers/amlogic/amports/vh264_4k2k.c
@@ -812,7 +812,6 @@ static void vh264_4k2k_put_timer_func(unsigned long arg)
         state = RECEIVER_INACTIVE;
     }
 
-#if 0
     // error watchdog
     if (((READ_VREG(VLD_MEM_VIFIFO_CONTROL) & 0x100) == 0) && // decoder has input
         (state == RECEIVER_INACTIVE) &&                       // receiver has no buffer to recycle
@@ -836,7 +835,6 @@ static void vh264_4k2k_put_timer_func(unsigned long arg)
         fatal_error = DECODER_FATAL_ERROR_UNKNOW;
         WRITE_VREG(FATAL_ERROR, 0);
     }
-#endif
 
     while (!kfifo_is_empty(&recycle_q) &&
            (READ_VREG(BUFFER_RECYCLE) == 0)) {
-- 
2.19.0

