From 33413935cc63d4fb22602ae79dcbc540236c27b1 Mon Sep 17 00:00:00 2001
From: Victor Wan <victor.wan@amlogic.com>
Date: Sat, 5 Jul 2014 15:12:54 +0800
Subject: [PATCH 4483/5965] printk: print full version info in dump stack or
 panic

---
 kernel/printk.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/kernel/printk.c b/kernel/printk.c
index 4577451cddb2..4b213412294b 100755
--- a/kernel/printk.c
+++ b/kernel/printk.c
@@ -45,7 +45,7 @@
 #include <linux/poll.h>
 #include <linux/irq_work.h>
 #include <linux/utsname.h>
-
+#include <generated/utsrelease.h>
 #include <asm/uaccess.h>
 
 #define CREATE_TRACE_POINTS
@@ -2910,7 +2910,7 @@ void dump_stack_print_info(const char *log_lvl)
 {
 	printk("%sCPU: %d PID: %d Comm: %.20s %s %s %.*s\n",
 	       log_lvl, raw_smp_processor_id(), current->pid, current->comm,
-	       print_tainted(), init_utsname()->release,
+	       print_tainted(), UTS_RELEASE_FULL /*init_utsname()->release*/,
 	       (int)strcspn(init_utsname()->version, " "),
 	       init_utsname()->version);
 
-- 
2.19.0

