From 0e7bb2331166757b3888f0eb21edc2dc249b5a7a Mon Sep 17 00:00:00 2001
From: "zhi.zhou" <zhi.zhou@amlogic.com>
Date: Sat, 5 Jul 2014 17:22:55 +0800
Subject: [PATCH 4485/5965] m8m2# for hevc 4k support feature .,

---
 drivers/amlogic/amports/vh265.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/drivers/amlogic/amports/vh265.c b/drivers/amlogic/amports/vh265.c
index b3c4263e44f0..f15f63386f49 100755
--- a/drivers/amlogic/amports/vh265.c
+++ b/drivers/amlogic/amports/vh265.c
@@ -3346,7 +3346,11 @@ static struct platform_driver amvdec_h265_driver = {
 
 static struct codec_profile_t amvdec_h265_profile = {
     .name = "hevc",
+#if MESON_CPU_TYPE >= MESON_CPU_TYPE_MESON8M2
+    .profile = "4k" //support 4k
+#else
     .profile = ""
+#endif
 };
 
 static int __init amvdec_h265_driver_init_module(void)
-- 
2.19.0

