From c5a483caf0b84aec3a05510afcebff78dee8c2f8 Mon Sep 17 00:00:00 2001
From: "yun.cai" <yun.cai@amlogic.com>
Date: Sat, 5 Jul 2014 20:06:11 +0800
Subject: [PATCH 4486/5965] M8M2: bt work

---
 drivers/amlogic/wifi/wifi_dt.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/drivers/amlogic/wifi/wifi_dt.c b/drivers/amlogic/wifi/wifi_dt.c
index 03f503723a25..0938e3a66022 100644
--- a/drivers/amlogic/wifi/wifi_dt.c
+++ b/drivers/amlogic/wifi/wifi_dt.c
@@ -194,7 +194,7 @@ int wifi_setup_dt()
 	//setup 32k clock
 	wifi_request_32k_clk(1, OWNER_NAME);
 	
-#if ((!(defined CONFIG_ARCH_MESON8)) && (!(defined CONFIG_ARCH_MESON8B)))
+#if ((!(defined CONFIG_ARCH_MESON8)) && (!(defined CONFIG_ARCH_MESON8B)) && (!(defined CONFIG_ARCH_MESON8M2)))
 	//setup sdio pullup
 	aml_clr_reg32_mask(P_PAD_PULL_UP_REG4,0xf|1<<8|1<<9|1<<11|1<<12);		
 	aml_clr_reg32_mask(P_PAD_PULL_UP_REG2,1<<7|1<<8|1<<9);	
@@ -294,7 +294,7 @@ void wifi_request_32k_clk(int is_on, const char *requestor)
 			amlogic_gpio_direction_output(wifi_info.clock_32k_pin, 0, OWNER_NAME);
 			CHECK_RET(ret);
 			SHOW_PIN_OWN("clock_32k_pin", wifi_info.clock_32k_pin);
-#if ((defined CONFIG_ARCH_MESON8) || (defined CONFIG_ARCH_MESON8B))
+#if ((defined CONFIG_ARCH_MESON8) || (defined CONFIG_ARCH_MESON8B) || (defined CONFIG_ARCH_MESON8M2))
 			aml_set_reg32_mask(P_PERIPHS_PIN_MUX_3,0x1<<22);//set mode GPIOX_10-->CLK_OUT3
 #else
             if(wifi_info.clock_32k_pin == 96) { // GPIOD_1, as PWM_D output
@@ -322,7 +322,7 @@ void wifi_request_32k_clk(int is_on, const char *requestor)
         if(clk_32k_on < 0)
             clk_32k_on = 0; 
 		if (clk_32k_on == 0) {
-#if ((defined CONFIG_ARCH_MESON8) || (defined CONFIG_ARCH_MESON8B))
+#if ((defined CONFIG_ARCH_MESON8) || (defined CONFIG_ARCH_MESON8B) || (defined CONFIG_ARCH_MESON8M2))
                         aml_clr_reg32_mask(P_PERIPHS_PIN_MUX_3,0x1<<22);
 #else
 			aml_clr_reg32_mask(P_PERIPHS_PIN_MUX_3,0x1<<21);
-- 
2.19.0

