From bf70e66128ed6caeb2daf009716e0895a71e7423 Mon Sep 17 00:00:00 2001
From: "kele.bai" <kele.bai@amlogic.com>
Date: Mon, 7 Jul 2014 09:44:13 +0800
Subject: [PATCH 4487/5965] [M8M2]: add default version information.

---
 drivers/amlogic/deinterlace/deinterlace.h | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/drivers/amlogic/deinterlace/deinterlace.h b/drivers/amlogic/deinterlace/deinterlace.h
index 94cf905fe340..2b53b2f7030d 100755
--- a/drivers/amlogic/deinterlace/deinterlace.h
+++ b/drivers/amlogic/deinterlace/deinterlace.h
@@ -35,6 +35,9 @@
 #elif (MESON_CPU_TYPE==MESON_CPU_TYPE_MESON8 || MESON_CPU_TYPE==MESON_CPU_TYPE_MESON8B || MESON_CPU_TYPE == MESON_CPU_TYPE_MESON8M2)
 #define NEW_DI_V1 //from m6tvc
 #define NEW_DI_V2 //from m6tvd(noise meter bug fix,improvement for 2:2 pull down)
+#else
+#define NEW_DI_V1 //from m6tvc
+#define NEW_DI_V2 //from m6tvd(noise meter bug fix,improvement for 2:2 pull down)
 #endif
 
 #ifndef CONFIG_VSYNC_RDMA
-- 
2.19.0

