From 07eb231e39de394d02d40c2e3ce87c423fe45d82 Mon Sep 17 00:00:00 2001
From: "kele.bai" <kele.bai@amlogic.com>
Date: Tue, 8 Jul 2014 10:24:25 +0800
Subject: [PATCH 4489/5965] [M8M2]:add m6 di version null.

---
 drivers/amlogic/deinterlace/deinterlace.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/amlogic/deinterlace/deinterlace.h b/drivers/amlogic/deinterlace/deinterlace.h
index 2b53b2f7030d..1564ab861f7f 100755
--- a/drivers/amlogic/deinterlace/deinterlace.h
+++ b/drivers/amlogic/deinterlace/deinterlace.h
@@ -35,7 +35,7 @@
 #elif (MESON_CPU_TYPE==MESON_CPU_TYPE_MESON8 || MESON_CPU_TYPE==MESON_CPU_TYPE_MESON8B || MESON_CPU_TYPE == MESON_CPU_TYPE_MESON8M2)
 #define NEW_DI_V1 //from m6tvc
 #define NEW_DI_V2 //from m6tvd(noise meter bug fix,improvement for 2:2 pull down)
-#else
+#elif (MESON_CPU_TYPE > MESON_CPU_TYPE_MESON8M2)
 #define NEW_DI_V1 //from m6tvc
 #define NEW_DI_V2 //from m6tvd(noise meter bug fix,improvement for 2:2 pull down)
 #endif
-- 
2.19.0

