From 973288b747336fff6c7f8da58a4b5a26ca9b5b30 Mon Sep 17 00:00:00 2001
From: "wenbiao.zhang" <wenbiao.zhang@amlogic.com>
Date: Thu, 15 May 2014 14:22:32 +0800
Subject: [PATCH 4493/5965] PD #91675: don't suspend ethernet in early suspend

---
 drivers/amlogic/ethernet/am_net8218.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/amlogic/ethernet/am_net8218.c b/drivers/amlogic/ethernet/am_net8218.c
index 2eb3da37bcf8..237f0312f452 100755
--- a/drivers/amlogic/ethernet/am_net8218.c
+++ b/drivers/amlogic/ethernet/am_net8218.c
@@ -55,6 +55,7 @@
 #define DRV_NAME	DRIVER_NAME
 #define DRV_VERSION	"v2.0.0"
 
+#undef CONFIG_HAS_EARLYSUSPEND
 #ifdef CONFIG_HAS_EARLYSUSPEND
 #include <linux/earlysuspend.h>
 static struct early_suspend early_suspend;
-- 
2.19.0

