From 4ade93919c90ed2595eb372ce39d0340437ea735 Mon Sep 17 00:00:00 2001
From: "yvonne.chen" <yvonne.chen@amlogic.com>
Date: Thu, 3 Jul 2014 19:29:02 +0800
Subject: [PATCH 4507/5965] pd#94021 add args control for drm mode setting

---
 drivers/amlogic/amports/amstream.c | 8 +++++++-
 1 file changed, 7 insertions(+), 1 deletion(-)

diff --git a/drivers/amlogic/amports/amstream.c b/drivers/amlogic/amports/amstream.c
index 875ab121111f..08dc39bd1e31 100755
--- a/drivers/amlogic/amports/amstream.c
+++ b/drivers/amlogic/amports/amstream.c
@@ -1704,7 +1704,13 @@ static long amstream_ioctl(struct file *file,
         break;		
 		}
 	case AMSTREAM_IOC_SET_DRMMODE:
-		this->flag |= PORT_FLAG_DRM;
+		if((u32)arg==1){
+            printk("set drmmode\n");
+			this->flag |= PORT_FLAG_DRM;
+		}else{
+		    this->flag &= (~PORT_FLAG_DRM);
+			printk("no drmmode\n");
+		}
 		break;
      case AMSTREAM_IOC_SET_APTS:
         {
-- 
2.19.0

