From 74319a3a607eeebf53ea3d97c289cc3070f0f7d7 Mon Sep 17 00:00:00 2001
From: Evoke Zhang <evoke.zhang@amlogic.com>
Date: Fri, 11 Jul 2014 17:54:18 +0800
Subject: [PATCH 4512/5965] PD#93958:VPU:delete VPU control when suspend/resume

---
 arch/arm/mach-meson8b/vpu.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/arch/arm/mach-meson8b/vpu.c b/arch/arm/mach-meson8b/vpu.c
index fee32a245f9c..97956058e803 100755
--- a/arch/arm/mach-meson8b/vpu.c
+++ b/arch/arm/mach-meson8b/vpu.c
@@ -660,13 +660,13 @@ static void vpu_driver_disable(void)
 #ifdef CONFIG_PM
 static int vpu_suspend(struct platform_device *pdev, pm_message_t state)
 {
-	vpu_driver_disable();
+	//vpu_driver_disable();
 	return 0;
 }
 
 static int vpu_resume(struct platform_device *pdev)
 {
-	vpu_driver_init();
+	//vpu_driver_init();
 	return 0;
 }
 #endif
-- 
2.19.0

