From 6c0c6f8cff456df9dddc6d00abac6572e6467d92 Mon Sep 17 00:00:00 2001
From: "bo.yang" <bo.yang@amlogic.com>
Date: Mon, 21 Jul 2014 17:07:33 +0800
Subject: [PATCH 4549/5965] regd: fix compile warning

---
 drivers/amlogic/regd/regd.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/drivers/amlogic/regd/regd.c b/drivers/amlogic/regd/regd.c
index aa807a689c50..cc70d3df85b2 100755
--- a/drivers/amlogic/regd/regd.c
+++ b/drivers/amlogic/regd/regd.c
@@ -776,7 +776,7 @@ static long regd_ioctl(struct file *file, unsigned int cmd, unsigned long arg)
 		sscanf(parm, "%x", gammaR);
 		printk("sgr:%s --> %d\n",parm, gammaR[0]);
 		if(gammaR[0] > 598){
-			gammaR[0] == 598;
+			gammaR[0] = 598;
 		}
 		set_lcd_gamma_table(gammaR, H_SEL_R);
 		break;
@@ -796,7 +796,7 @@ static long regd_ioctl(struct file *file, unsigned int cmd, unsigned long arg)
 		sscanf(parm, "%x", gammaR);
 		printk("sgg:%s --> %d\n",parm, gammaR[0]);
 		if(gammaR[0] > 598){
-			gammaR[0] == 598;
+			gammaR[0] = 598;
 		}
 		set_lcd_gamma_table(gammaR, H_SEL_G);
 		break;
@@ -816,7 +816,7 @@ static long regd_ioctl(struct file *file, unsigned int cmd, unsigned long arg)
 		sscanf(parm, "%x", gammaR);
 		printk("sgb:%s --> %d\n",parm, gammaR[0]);
 		if(gammaR[0] > 598){
-			gammaR[0] == 598;
+			gammaR[0] = 598;
 		}
 		set_lcd_gamma_table(gammaR, H_SEL_B);
 		break;
-- 
2.19.0

