From e8b654483b31ccc2ea12c4b7717405b5597cd57b Mon Sep 17 00:00:00 2001
From: "brian.zhu" <brian.zhu@amlogic.com>
Date: Wed, 30 Jul 2014 12:44:13 +0800
Subject: [PATCH 4597/5965] PD #89978: fix compile err for m6tvd

---
 drivers/amlogic/amports/encoder.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/drivers/amlogic/amports/encoder.c b/drivers/amlogic/amports/encoder.c
index 9386515989aa..cceefa250945 100755
--- a/drivers/amlogic/amports/encoder.c
+++ b/drivers/amlogic/amports/encoder.c
@@ -1701,6 +1701,7 @@ void amvenc_avc_start_cmd(int cmd, unsigned* input_info, int ucode_mode)
 		}
 		avc_init_ie_me_parameter();
 	}
+#if MESON_CPU_TYPE >= MESON_CPU_TYPE_MESON8
 	else{
 		if((dblk_buf_canvas&0xff)==ENC_CANVAS_OFFSET){
 			WRITE_HREG(CURRENT_Y_CANVAS_START, gAmvencbuff.buf_start + gAmvencbuff.bufspec->dec0_y.buf_start);
@@ -1766,6 +1767,7 @@ void amvenc_avc_start_cmd(int cmd, unsigned* input_info, int ucode_mode)
 		}
 #endif	    
 	}	    
+#endif
 	encoder_status = cmd;
 	WRITE_HREG(ENCODER_STATUS , cmd);
 	if((cmd == ENCODER_IDR)||(cmd == ENCODER_NON_IDR)){
-- 
2.19.0

