From 56f902b8a0f5775f72c0efa087910abc96e0247a Mon Sep 17 00:00:00 2001
From: "kele.bai" <kele.bai@amlogic.com>
Date: Tue, 22 Jul 2014 16:02:08 +0800
Subject: [PATCH 4614/5965] di: remove m8m2 pre-compile macro

---
 drivers/amlogic/deinterlace/deinterlace.h | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/amlogic/deinterlace/deinterlace.h b/drivers/amlogic/deinterlace/deinterlace.h
index 1564ab861f7f..9dd672fa3cf6 100755
--- a/drivers/amlogic/deinterlace/deinterlace.h
+++ b/drivers/amlogic/deinterlace/deinterlace.h
@@ -32,10 +32,10 @@
 #define NEW_DI_TV
 #define NEW_DI_V1 //from m6tvc
 #define NEW_DI_V2 //from m6tvd(noise meter bug fix,improvement for 2:2 pull down)
-#elif (MESON_CPU_TYPE==MESON_CPU_TYPE_MESON8 || MESON_CPU_TYPE==MESON_CPU_TYPE_MESON8B || MESON_CPU_TYPE == MESON_CPU_TYPE_MESON8M2)
+#elif (MESON_CPU_TYPE==MESON_CPU_TYPE_MESON8 || MESON_CPU_TYPE==MESON_CPU_TYPE_MESON8B)
 #define NEW_DI_V1 //from m6tvc
 #define NEW_DI_V2 //from m6tvd(noise meter bug fix,improvement for 2:2 pull down)
-#elif (MESON_CPU_TYPE > MESON_CPU_TYPE_MESON8M2)
+#elif (MESON_CPU_TYPE > MESON_CPU_TYPE_MESON8B)
 #define NEW_DI_V1 //from m6tvc
 #define NEW_DI_V2 //from m6tvd(noise meter bug fix,improvement for 2:2 pull down)
 #endif
-- 
2.19.0

