From ed8ee2d9bf75ecdd13d7fde02363f0a095f7e97d Mon Sep 17 00:00:00 2001
From: "baoqi.wang" <baoqi.wang@amlogic.com>
Date: Thu, 24 Jul 2014 17:10:31 +0800
Subject: [PATCH 4650/5965] PD#94858 Mask  Abnormal Interrupt for ethernet

---
 drivers/amlogic/ethernet/am_net8218.c | 2 --
 1 file changed, 2 deletions(-)

diff --git a/drivers/amlogic/ethernet/am_net8218.c b/drivers/amlogic/ethernet/am_net8218.c
index 08431ad282f1..99c3b5152d57 100755
--- a/drivers/amlogic/ethernet/am_net8218.c
+++ b/drivers/amlogic/ethernet/am_net8218.c
@@ -1866,9 +1866,7 @@ static int setup_net_device(struct net_device *dev)
 	               (1 << 6) |           //Receive Interrupt Enable
 	               (1 << 2) |           //Transmit Buffer Unavailable Enable
 	               (1 << 3) |           //TJT: Transmit Jabber Timeout
-	               (1 << 4) |           //OVF: Receive Overflow
 	               (1 << 5) |           //UNF: Transmit Underflow
-	               (1 << 7) |           //RU: Receive Buffer Unavailable
 	               (1 << 8) |           //RPS: Receive Process Stopped
 	               (1 << 13) |          //FBI: Fatal Bus Error Interrupt
 	               (1) | 		        //tx interrupt
-- 
2.19.0

