From 42ed78afcbc8ced40652f186624fb984cf644422 Mon Sep 17 00:00:00 2001
From: "spike.liao" <spike.liao@amlogic.com>
Date: Tue, 18 Mar 2014 14:30:15 +0800
Subject: [PATCH 4671/5965] PD #88891 mpeg4 support 1088x1920 video

---
 drivers/amlogic/amports/vmpeg4.c | 10 ++++++++--
 1 file changed, 8 insertions(+), 2 deletions(-)

diff --git a/drivers/amlogic/amports/vmpeg4.c b/drivers/amlogic/amports/vmpeg4.c
index baee1b7a5431..fc1fa93f88ec 100755
--- a/drivers/amlogic/amports/vmpeg4.c
+++ b/drivers/amlogic/amports/vmpeg4.c
@@ -605,8 +605,14 @@ static void vmpeg4_canvas_init(void)
         decbuf_size = 0x100000;
     } else {
         /* HD & SD */
-        canvas_width = 1920;
-        canvas_height = 1088;
+        if(vmpeg4_amstream_dec_info.height > vmpeg4_amstream_dec_info.width){
+            canvas_width = 1088;
+            canvas_height = 1920;            
+        }
+        else{
+            canvas_width = 1920;
+            canvas_height = 1088;
+        }
         decbuf_y_size = 0x200000;
         decbuf_uv_size = 0x80000;
         decbuf_size = 0x300000;
-- 
2.19.0

