From 83ce0bef4ae97405c63a9abaf4a86898ece4d22e Mon Sep 17 00:00:00 2001
From: "simon.zheng" <simon@ubuntu.(none)>
Date: Wed, 23 Jul 2014 14:18:18 +0800
Subject: [PATCH 4724/5965] pd 94497: frame recycle by wrong triggered

---
 drivers/amlogic/ppmgr/ppmgr_vpp.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/drivers/amlogic/ppmgr/ppmgr_vpp.c b/drivers/amlogic/ppmgr/ppmgr_vpp.c
index 75bb07176345..1ed62a7899d3 100755
--- a/drivers/amlogic/ppmgr/ppmgr_vpp.c
+++ b/drivers/amlogic/ppmgr/ppmgr_vpp.c
@@ -108,6 +108,7 @@ void vf_ppmgr_reset(int type);
 void ppmgr_vf_put_dec(vframe_t *vf);
 
 
+extern u32 timestamp_pcrscr_enable_state(void);
 
 #define is_valid_ppframe(pp_vf) \
 	((pp_vf >= &vfp_pool[0]) && (pp_vf <= &vfp_pool[VF_POOL_SIZE-1]))
@@ -317,6 +318,10 @@ static int ppmgr_event_cb(int type, void *data, void *private_data)
 #endif
     if(type & VFRAME_EVENT_RECEIVER_FRAME_WAIT){
         if(task_running && !ppmgr_device.use_prot){
+			
+			if(timestamp_pcrscr_enable_state()){
+				return 0;
+			}
             if(get_property_change()){
                 //printk("--ppmgr: get angle changed msg.\n");
                 set_property_change(0);
-- 
2.19.0

