From 5aea0cfeb21175c78a6489c810911165ba442b0f Mon Sep 17 00:00:00 2001
From: "brian.zhu" <brian.zhu@amlogic.com>
Date: Mon, 18 Aug 2014 13:58:59 +0800
Subject: [PATCH 4742/5965] PD#96021 fix wait 5s whe hdmi in

---
 drivers/amlogic/video_dev/amlvideo2.c | 31 +++++++++++++--------------
 1 file changed, 15 insertions(+), 16 deletions(-)

diff --git a/drivers/amlogic/video_dev/amlvideo2.c b/drivers/amlogic/video_dev/amlvideo2.c
index b69e226c4b13..dccc43df83fa 100755
--- a/drivers/amlogic/video_dev/amlvideo2.c
+++ b/drivers/amlogic/video_dev/amlvideo2.c
@@ -2131,7 +2131,7 @@ static int frame_num_out = 0;
 static struct timeval test_time;
 #endif
 
-static void amlvideo2_thread_tick(struct amlvideo2_fh *fh)
+static int  amlvideo2_thread_tick(struct amlvideo2_fh *fh)
 {
 	struct amlvideo2_node_buffer *buf = NULL;
 	struct amlvideo2_node *node = fh->node;
@@ -2145,18 +2145,11 @@ static void amlvideo2_thread_tick(struct amlvideo2_fh *fh)
 	dprintk(node->vid_dev, 1, "Thread tick\n");
 
 	if (kthread_should_stop())
-		return;
+		return 0;
 
 
 	if( (AML_RECEIVER_NONE != node->r_type) && vfq_full( &q_ready) ){
-		return;
-	}
-
-	wait_event_interruptible_timeout(dma_q->wq, ((vf_peek(node->recv.name)!= NULL)&&(fh->is_streamed_on)&&(node->provide_ready)) || (node->vidq.task_running==0),msecs_to_jiffies(5000));
-
-	if(!node->provide_ready){
-		dprintk(node->vid_dev, 1, "provide is not ready\n");
-		return ;
+		return -1;
 	}
 
 	if(!fh->is_streamed_on){
@@ -2165,7 +2158,14 @@ static void amlvideo2_thread_tick(struct amlvideo2_fh *fh)
 			vf = vf_get(node->recv.name);
 			vf_inqueue(vf, node->recv.name);
 		}
-		return ;
+		return -1;
+	}
+
+	wait_event_interruptible_timeout(dma_q->wq, ((vf_peek(node->recv.name)!= NULL)&&(fh->is_streamed_on)&&(node->provide_ready)) || (node->vidq.task_running==0),msecs_to_jiffies(5000));
+
+	if(!node->provide_ready){
+		dprintk(node->vid_dev, 1, "provide is not ready\n");
+		return -1;
 	}
 
 	spin_lock_irqsave(&node->slock, flags);
@@ -2319,11 +2319,11 @@ static void amlvideo2_thread_tick(struct amlvideo2_fh *fh)
 	}
 #endif
 	dprintk(node->vid_dev, 2, "[%p/%d] wakeup\n", buf, buf->vb.i);
-	return;
+	return 0;
 
 unlock:
 	spin_unlock_irqrestore(&node->slock, flags);
-	return;
+	return 0;
 }
 
 static void amlvideo2_sleep(struct amlvideo2_fh *fh)
@@ -2343,9 +2343,8 @@ static void amlvideo2_sleep(struct amlvideo2_fh *fh)
 	/* Calculate time to wake up */
 	//timeout = msecs_to_jiffies(frames_to_ms(1));
 
-	amlvideo2_thread_tick(fh);
-
-	//schedule_timeout_interruptible(1);
+	if(amlvideo2_thread_tick(fh)<0)
+		schedule_timeout_interruptible(1);
 
 //stop_task:
 	//remove_wait_queue(&dma_q->wq, &wait);
-- 
2.19.0

