From 9d028888ac6d77482935c53957509a35824c25d7 Mon Sep 17 00:00:00 2001
From: "brian.zhu" <brian.zhu@amlogic.com>
Date: Mon, 1 Sep 2014 16:44:07 +0800
Subject: [PATCH 4813/5965] PD#95557:fix amlvideo2 compile error

---
 drivers/amlogic/video_dev/amlvideo2.c | 13 +++++++------
 1 file changed, 7 insertions(+), 6 deletions(-)

diff --git a/drivers/amlogic/video_dev/amlvideo2.c b/drivers/amlogic/video_dev/amlvideo2.c
index bdf7cf9757b5..47109ba9ecd7 100755
--- a/drivers/amlogic/video_dev/amlvideo2.c
+++ b/drivers/amlogic/video_dev/amlvideo2.c
@@ -2142,8 +2142,8 @@ static int  amlvideo2_thread_tick(struct amlvideo2_fh *fh)
 	unsigned diff = 0;
 	bool no_frame = false;
 	vframe_t *vf = NULL;
-
 	unsigned long flags = 0;
+	int active_duration = 0;
 
 	dprintk(node->vid_dev, 1, "Thread tick\n");
 
@@ -2208,7 +2208,6 @@ static int  amlvideo2_thread_tick(struct amlvideo2_fh *fh)
 		}
 	}
 #else
-	int active_duration = 0;
 	if(frame_inittime == 1){
 	   	if(no_frame)
 			goto unlock;
@@ -2269,9 +2268,10 @@ static int  amlvideo2_thread_tick(struct amlvideo2_fh *fh)
 #ifdef TEST_LATENCY
 	if(print_latecny)
 	{
+		int timeNow64, timePts;
 		do_gettimeofday(&test_time);
-		int timeNow64 = ((test_time.tv_sec & 0xFFFFFFFF)*1000*1000) + (test_time.tv_usec);
-		int timePts =  ((vf->pts_us64 & 0xFFFFFFFF)*1000*1000) + (vf->pts);
+		timeNow64 = ((test_time.tv_sec & 0xFFFFFFFF)*1000*1000) + (test_time.tv_usec);
+		timePts =  ((vf->pts_us64 & 0xFFFFFFFF)*1000*1000) + (vf->pts);
 		//printk("amlvideo2 in  num:%d delay:%d\n", timePts, (timeNow64 - timePts)/1000);
 		if(cur_time == test_time.tv_sec){
 			total_latency += (int)((timeNow64 - timePts)/1000);
@@ -2310,9 +2310,10 @@ static int  amlvideo2_thread_tick(struct amlvideo2_fh *fh)
 #ifdef TEST_LATENCY
 	if(print_latecny)
 	{
+		int timeNow64_out, timePts_out;
 		do_gettimeofday(&test_time);
-		int timeNow64_out = ((test_time.tv_sec & 0xFFFFFFFF)*1000*1000) + (test_time.tv_usec);
-		int timePts_out =  ((buf->vb.ts.tv_sec & 0xFFFFFFFF)*1000*1000) + (buf->vb.ts.tv_usec);
+		timeNow64_out = ((test_time.tv_sec & 0xFFFFFFFF)*1000*1000) + (test_time.tv_usec);
+		timePts_out =  ((buf->vb.ts.tv_sec & 0xFFFFFFFF)*1000*1000) + (buf->vb.ts.tv_usec);
 		//printk("amlvideo2 out num:%d delay:%d\n", timePts_out, (timeNow64_out - timePts_out)/1000);
 		if(cur_time_out == test_time.tv_sec){
 			total_latency_out += (timeNow64_out - timePts_out)/1000;
-- 
2.19.0

