From 96b35c3bfcf77e9c701d3dd76d1e6bf2121fe8df Mon Sep 17 00:00:00 2001
From: "bo.yang" <bo.yang@amlogic.com>
Date: Thu, 4 Sep 2014 11:17:10 +0800
Subject: [PATCH 4832/5965] m6tvd: fix compile error of encoder

---
 drivers/amlogic/amports/encoder.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/drivers/amlogic/amports/encoder.c b/drivers/amlogic/amports/encoder.c
index d5c108606005..1809a82b5f34 100755
--- a/drivers/amlogic/amports/encoder.c
+++ b/drivers/amlogic/amports/encoder.c
@@ -2380,8 +2380,10 @@ static struct resource memobj;
 #endif
 static int amvenc_avc_probe(struct platform_device *pdev)
 {
+#ifndef CONFIG_CMA
     struct resource *mem;
     int idx;
+#endif
 
     amlog_level(LOG_LEVEL_INFO, "amvenc_avc probe start.\n");
 
-- 
2.19.0

