From 8e3a935de12ff4147685d92a5a2e784bd4f62b99 Mon Sep 17 00:00:00 2001
From: "bo.yang" <bo.yang@amlogic.com>
Date: Wed, 8 Oct 2014 11:19:09 +0800
Subject: [PATCH 4921/5965] m6tv: update tool of updatemodules.sh

---
 .../tools/upmod/updatemodules.sh              | 55 +++++++++++++++++++
 1 file changed, 55 insertions(+)

diff --git a/arch/arm/mach-meson6tv/tools/upmod/updatemodules.sh b/arch/arm/mach-meson6tv/tools/upmod/updatemodules.sh
index 35f8e471e4f5..393f4d8c500d 100755
--- a/arch/arm/mach-meson6tv/tools/upmod/updatemodules.sh
+++ b/arch/arm/mach-meson6tv/tools/upmod/updatemodules.sh
@@ -108,6 +108,26 @@ cd ../../../../../
 echo ""
 
 
+
+
+if [ ! -d "hardware/dvb/silabs/drivers/si2168" ]; then
+	mkdir -p hardware/dvb/silabs/drivers/
+	cd hardware/dvb/silabs/drivers/
+	git clone ssh://android@10.8.9.5/linux/dvb/silabs/drivers/si2168.git
+	cd si2168
+	git checkout -t origin/amlogic-3.10 -b amlogic-3.10
+else
+	cd hardware/dvb/silabs/drivers/si2168
+	echo "updating hardware/dvb/silabs/drivers/si2168"
+	git pull
+fi
+
+
+# cd <work-directory>
+cd ../../../../../
+echo ""
+
+
 if [ ! -d "hardware/dvb/availink/drivers/avl6211" ]; then
 	mkdir -p hardware/dvb/availink/drivers/
 	cd hardware/dvb/availink/drivers/
@@ -126,6 +146,23 @@ cd ../../../../../
 echo ""
 
 
+if [ ! -d "hardware/dvb/maxlinear/drivers/mxl101" ]; then
+	mkdir -p hardware/dvb/maxlinear/drivers/
+	cd hardware/dvb/maxlinear/drivers/
+	git clone ssh://android@10.8.9.5/linux/dvb/maxlinear/drivers/mxl101.git
+	cd mxl101/
+	git checkout -t origin/amlogic-3.10 -b amlogic-3.10
+else
+	cd hardware/dvb/maxlinear/drivers/mxl101
+	echo "updating hardware/dvb/maxlinear/drivers/mxl101"
+	git pull
+fi
+
+
+# cd <work-directory>
+cd ../../../../../
+echo ""
+
 
 ################################################################################
 #
@@ -363,6 +400,24 @@ else
 fi
 
 
+# cd <work-directory>
+cd ../../../../../
+echo ""
+
+
+if [ ! -d "hardware/wifi/realtek/drivers/8723bu" ]; then
+	mkdir -p hardware/wifi/realtek/drivers
+	cd hardware/wifi/realtek/drivers
+	git clone git://git-sc.amlogic.com/platform/hardware/wifi/realtek/drivers/8723bu.git
+	cd 8723bu
+	git checkout -t origin/8723bu -b 8723bu
+else
+	cd hardware/wifi/realtek/drivers/8723bu
+	echo "updating hardware/wifi/realtek/drivers/8723bu"
+	git pull
+fi
+
+
 
 # cd <work-directory>
 cd ../../../../../
-- 
2.19.0

