From 28d80ff3af524698c01f635ca8d400de402bb6c2 Mon Sep 17 00:00:00 2001
From: "ke.gong" <ke.gong@amlogic.com>
Date: Mon, 10 Nov 2014 10:04:23 +0800
Subject: [PATCH 5040/5965] PD #99252: Fixed bug when store fatal_error_flag in
 vh264.c.

---
 drivers/amlogic/amports/vh264.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/amlogic/amports/vh264.c b/drivers/amlogic/amports/vh264.c
index 9440da68cc4a..6a931afb838c 100755
--- a/drivers/amlogic/amports/vh264.c
+++ b/drivers/amlogic/amports/vh264.c
@@ -1487,7 +1487,7 @@ int vh264_dec_status(struct vdec_status *vstatus)
     vstatus->error_count = READ_VREG(AV_SCRATCH_D);
     vstatus->status = stat;
     if (fatal_error_reset) {
-        vstatus->status |= fatal_error_flag << 16;
+        vstatus->status |= fatal_error_flag;
     }
     return 0;
 }
-- 
2.19.0

