From 537223e9a558a64e3be2a9943dba735be418a927 Mon Sep 17 00:00:00 2001
From: "shi.liu" <shi.liu@amlogic.com>
Date: Wed, 19 Nov 2014 16:59:10 +0800
Subject: [PATCH 5065/5965] pd#99063:mm: optimize for 512M s805 platform

Change-Id: I37130084c8c0c7780656b3bd371f47b6b4f74b4b
---
 mm/vmscan.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/mm/vmscan.c b/mm/vmscan.c
index f57d483d3d67..6f9093183cbe 100755
--- a/mm/vmscan.c
+++ b/mm/vmscan.c
@@ -1818,6 +1818,9 @@ static void get_scan_count(struct lruvec *lruvec, struct scan_control *sc,
 	 * This scanning priority is essentially the inverse of IO cost.
 	 */
 	anon_prio = vmscan_swappiness(sc);
+	if((totalram_pages <= 0x20000) && vm_swap_full()){
+		anon_prio >>= 1;
+	}
 	file_prio = 200 - anon_prio;
 
 	/*
-- 
2.19.0

