From 9192c1190bf20e3ab34ea846701d73235a351631 Mon Sep 17 00:00:00 2001
From: "bo.yang" <bo.yang@amlogic.com>
Date: Wed, 19 Nov 2014 13:55:08 +0800
Subject: [PATCH 5068/5965] g9tv: add support of canvas

---
 drivers/amlogic/canvas/canvas.c | 4 +---
 1 file changed, 1 insertion(+), 3 deletions(-)

diff --git a/drivers/amlogic/canvas/canvas.c b/drivers/amlogic/canvas/canvas.c
index 82031111a5d2..74338ddda5e4 100755
--- a/drivers/amlogic/canvas/canvas.c
+++ b/drivers/amlogic/canvas/canvas.c
@@ -44,9 +44,7 @@
 #define CANVAS_NUM	192
 #endif
 
-#if MESON_CPU_TYPE == MESON_CPU_TYPE_MESON8B
-#define GM8_CANVAS_REG 0
-#elif MESON_CPU_TYPE >= MESON_CPU_TYPE_MESON8
+#if MESON_CPU_TYPE == MESON_CPU_TYPE_MESON8
 #define GM8_CANVAS_REG 1
 #else
 #define GM8_CANVAS_REG 0
-- 
2.19.0

