From 2a09feb022eecfd98155da040720e487cc6892d2 Mon Sep 17 00:00:00 2001
From: "bo.yang" <bo.yang@amlogic.com>
Date: Wed, 19 Nov 2014 17:34:14 +0800
Subject: [PATCH 5070/5965] PD #97124: add workaround to avoid compile error

---
 drivers/amlogic/display/vout/vout_serve.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/drivers/amlogic/display/vout/vout_serve.c b/drivers/amlogic/display/vout/vout_serve.c
index 240767f95e1f..71aa0007f3f5 100755
--- a/drivers/amlogic/display/vout/vout_serve.c
+++ b/drivers/amlogic/display/vout/vout_serve.c
@@ -150,7 +150,9 @@ static  void  set_vout_mode(char * name)
 		return ; 
 	}
 
+#ifdef CONFIG_AML_VOUT_FRAMERATE_AUTOMATION
 	mode_by_user = mode;
+#endif
 
 	if(mode==get_current_vmode())
 	{
-- 
2.19.0

