From f62938e5460830f8dc8a4b81fd0b07f52a4746bf Mon Sep 17 00:00:00 2001
From: "dezhi.kong" <dezhi.kong@amlogic.com>
Date: Sat, 22 Nov 2014 10:59:14 +0800
Subject: [PATCH 5101/5965] g9tv: fix compile for m6tv error in amcm.

---
 drivers/amlogic/amvecm/amcm.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/drivers/amlogic/amvecm/amcm.c b/drivers/amlogic/amvecm/amcm.c
index 11be4073c011..2a83b7042231 100755
--- a/drivers/amlogic/amvecm/amcm.c
+++ b/drivers/amlogic/amvecm/amcm.c
@@ -145,12 +145,14 @@ void am_set_regmap(struct am_regs_s *p)
 				WRITE_CBUS_REG(VPP_CHROMA_DATA_PORT, p->am_reg[i].val);
 #endif
 			break;
+#if (MESON_CPU_TYPE >= MESON_CPU_TYPE_MESONG9TV)
 		case REG_TYPE_VCBUS:
 			if (p->am_reg[i].mask == 0xffffffff)
 				WRITE_VCBUS_REG(p->am_reg[i].addr, p->am_reg[i].val);
 			else
 				WRITE_VCBUS_REG(p->am_reg[i].addr, (READ_VCBUS_REG(p->am_reg[i].addr) & (~(p->am_reg[i].mask))) | (p->am_reg[i].val & p->am_reg[i].mask));
 			break;
+#endif
 		default:
 			break;
 		}
-- 
2.19.0

