From 802d729a6d060531a5c8d6c85bfd9be7694c7e51 Mon Sep 17 00:00:00 2001
From: "bo.yang" <bo.yang@amlogic.com>
Date: Sat, 22 Nov 2014 14:51:04 +0800
Subject: [PATCH 5103/5965] video: fix compile error

---
 drivers/amlogic/amports/video2.c | 12 ++++++------
 1 file changed, 6 insertions(+), 6 deletions(-)

diff --git a/drivers/amlogic/amports/video2.c b/drivers/amlogic/amports/video2.c
index 36c763569811..76d464a8152b 100755
--- a/drivers/amlogic/amports/video2.c
+++ b/drivers/amlogic/amports/video2.c
@@ -75,13 +75,13 @@ MODULE_AMLOG(LOG_LEVEL_ERROR, 0, LOG_DEFAULT_LEVEL_DESC, LOG_MASK_DESC);
 
 #undef CONFIG_AM_DEINTERLACE
 
-#include "linux/amlogic/amports/ve.h"
-#include "linux/amlogic/amports/cm.h"
+#include "linux/amlogic/ve.h"
+#include "linux/amlogic/cm.h"
 
-#include "ve_regs.h"
-#include "amve.h"
-#include "cm_regs.h"
-#include "amcm.h"
+//#include "ve_regs.h"
+//#include "amve.h"
+//#include "cm_regs.h"
+//#include "amcm.h"
 
 
 //*************************************************
-- 
2.19.0

