From db99b23de17cfac1d6c7bfd5a9f73c3cd6b575e1 Mon Sep 17 00:00:00 2001
From: Larson Jiang <larson.jiang@amlogic.com>
Date: Tue, 25 Nov 2014 14:33:38 +0800
Subject: [PATCH 5108/5965] fix uard board detect problem

---
 drivers/amlogic/mmc/amlsd.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/amlogic/mmc/amlsd.c b/drivers/amlogic/mmc/amlsd.c
index f5d9e1d35c7b..8990eb0b4db5 100755
--- a/drivers/amlogic/mmc/amlsd.c
+++ b/drivers/amlogic/mmc/amlsd.c
@@ -970,7 +970,7 @@ static int aml_is_sduart(struct amlsd_platform * pdata)
 #else
        dat3 = aml_get_reg32_bits(P_PREG_PAD_GPIO0_I,26,1);
 #endif
-         return 0;
+
         if(dat3 == 1){
             // if (cnt)
                 // sdhc_err("cnt=%d\n", cnt);
-- 
2.19.0

