From 8b95984416625de6b502a75bd4e30cf1087ce0ee Mon Sep 17 00:00:00 2001
From: Peifu Jiang <peifu.jiang@amlogic.com>
Date: Wed, 26 Nov 2014 17:17:54 +0800
Subject: [PATCH 5123/5965] PD #99773: fix vpp vscale skip error when preblend
 enabled

---
 drivers/amlogic/amports/vpp.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/drivers/amlogic/amports/vpp.c b/drivers/amlogic/amports/vpp.c
index 5443c6b30434..763d1fb42fbd 100755
--- a/drivers/amlogic/amports/vpp.c
+++ b/drivers/amlogic/amports/vpp.c
@@ -385,6 +385,11 @@ vpp_process_speed_check(s32 width_in,
 
     return SPEED_CHECK_DONE;
 #else
+    /* return okay if vpp preblend enabled */
+    if (aml_read_reg32(P_VPP_MISC) & VPP_PREBLEND_EN) {
+        return SPEED_CHECK_DONE;
+    }
+
     /* if (video_speed_check_width * video_speed_check_height * height_out > height_screen * width_in * height_in) { */
     if ((height_out+1) > height_in) {
         return SPEED_CHECK_DONE;
-- 
2.19.0

