From 8b050771c31758a23f7bd9013e7109dfef54f709 Mon Sep 17 00:00:00 2001
From: "bo.yang" <bo.yang@amlogic.com>
Date: Mon, 5 Jan 2015 16:01:20 +0800
Subject: [PATCH 5293/5965] g9tv: lcd: update makefile

---
 drivers/amlogic/display/Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/amlogic/display/Makefile b/drivers/amlogic/display/Makefile
index d005b35d72ef..eacbb5b6fd4b 100755
--- a/drivers/amlogic/display/Makefile
+++ b/drivers/amlogic/display/Makefile
@@ -5,7 +5,7 @@ obj-$(CONFIG_AM_TCON_OUTPUT)	+= vout/
 obj-$(CONFIG_AM_GE2D)		+= ge2d/
 obj-$(CONFIG_FB_AM)		+= osd/
 obj-$(CONFIG_AM_FB_EXT)		+= osd_ext/
-obj-$(CONFIG_AML_TV_LCD)	+= lcd/
+obj-y						+= lcd/
 obj-$(CONFIG_AM_LOGO)		+= aml_logo/
 
 obj-y				+= backlight/
-- 
2.19.0

