From 636b535a8512122ddac487788df3c8b90a5e5f50 Mon Sep 17 00:00:00 2001
From: "simon.zheng" <simon@ubuntu.(none)>
Date: Tue, 13 Jan 2015 16:40:37 +0800
Subject: [PATCH 5338/5965] PD#101898 : detect orientation change in DI module

---
 drivers/amlogic/deinterlace/deinterlace.c | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/drivers/amlogic/deinterlace/deinterlace.c b/drivers/amlogic/deinterlace/deinterlace.c
index 0c8de69060c4..fd1f38fb069c 100755
--- a/drivers/amlogic/deinterlace/deinterlace.c
+++ b/drivers/amlogic/deinterlace/deinterlace.c
@@ -2309,8 +2309,9 @@ static unsigned char is_bypass(vframe_t *vf_in)
 #endif
 #if MESON_CPU_TYPE >= MESON_CPU_TYPE_MESON8
 /*prot is conflict with di post*/
-    if(di_pre_stru.orientation)
-	return 1;
+	if(vf_in && vf_in->video_angle){
+	    return 1;
+	}
 #endif
     if((di_vscale_skip_enable & 0x4)&& vf_in){
 	di_vscale_skip_count = get_current_vscale_skip_count(vf_in);
-- 
2.19.0

