From eaedfe55c67afd7d1a84ad3af38598f7bb62e509 Mon Sep 17 00:00:00 2001
From: Shi Liu <shi.liu@amlogic.com>
Date: Fri, 13 Mar 2015 14:45:01 +0800
Subject: [PATCH 5463/5965] PD#101972: mm: fix compile error

Change-Id: I056674abd68728b0262b37b02b428d84ba82e7f0
---
 mm/migrate.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/mm/migrate.c b/mm/migrate.c
index 8b4abcbb8f54..d55e19d9a233 100644
--- a/mm/migrate.c
+++ b/mm/migrate.c
@@ -89,6 +89,8 @@ EXPORT_SYMBOL(has_cma_page);
 #else
 void wakeup_wq(bool has_cma) {}
 bool has_cma_page(struct page *page){return false;}
+EXPORT_SYMBOL(has_cma_page);
+EXPORT_SYMBOL(wakeup_wq);
 #endif
 /*
  * migrate_prep() needs to be called before we start compiling a list of pages
-- 
2.19.0

